diff --git a/src/codegen/x64/register-x64.h b/src/codegen/x64/register-x64.h
index 261be62dac..a9c563317e 100644
--- a/src/codegen/x64/register-x64.h
+++ b/src/codegen/x64/register-x64.h
@@ -6,8 +6,6 @@
 #define V8_CODEGEN_X64_REGISTER_X64_H_
 
 #include "src/codegen/register-base.h"
-#include "src/codegen/register-configuration.h"
-#include "src/codegen/reglist.h"
 
 namespace v8 {
 namespace internal {
@@ -87,20 +85,6 @@ constexpr Register no_reg = Register::no_reg();
 
 constexpr int kNumRegs = 16;
 
-constexpr RegList kJSCallerSaved =
-    Register::ListOf(rax, rcx, rdx,
-                     rbx,  // used as a caller-saved register in JavaScript code
-                     rdi);  // callee function
-
-constexpr RegList kCallerSaved =
-#ifdef V8_TARGET_OS_WIN
-    Register::ListOf(rax, rcx, rdx, r8, r9, r10, r11);
-#else
-    Register::ListOf(rax, rcx, rdx, rdi, rsi, r8, r9, r10, r11);
-#endif  // V8_TARGET_OS_WIN
-
-constexpr int kNumJSCallerSaved = 5;
-
 #ifdef V8_TARGET_OS_WIN
 // Windows calling convention
 constexpr Register arg_reg_1 = rcx;
diff --git a/src/codegen/x64/reglist-x64.h b/src/codegen/x64/reglist-x64.h
new file mode 100644
index 0000000000..6c7be82440
