diff --git a/test/cctest/test-assembler-arm64.cc b/test/cctest/test-assembler-arm64.cc
index efcc5ebad1a..bd5621fa268 100644
--- a/test/cctest/test-assembler-arm64.cc
+++ b/test/cctest/test-assembler-arm64.cc
@@ -131,7 +131,7 @@ static void InitializeVM() {
   std::unique_ptr<PrintDisassembler> pdis;                                    \
   RegisterDump core;                                                          \
   HandleScope handle_scope(isolate);                                          \
-  Handle<Code> code;                                                          \
+  Handle<InstructionStream> code;                                             \
   if (i::v8_flags.trace_sim) {                                                \
     pdis.reset(new PrintDisassembler(stdout));                                \
     decoder->PrependVisitor(pdis.get());                                      \
@@ -181,7 +181,7 @@ static void InitializeVM() {
   MacroAssembler masm(isolate, v8::internal::CodeObjectRequired::kYes, \
                       owned_buf->CreateView());                        \
   HandleScope handle_scope(isolate);                                   \
-  Handle<Code> code;                                                   \
+  Handle<InstructionStream> code;                                      \
   RegisterDump core;
 
 #define RESET()                                                \
