diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index f85e64d03bc..f7447723e1b 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -452,6 +452,18 @@ inline void MaglevAssembler::CompareObjectTypeRange(Register heap_object,
                        higher_limit);
 }
 
+inline void MaglevAssembler::CompareMapWithRoot(Register object,
+                                                RootIndex index,
+                                                Register scratch) {
+  if (CanBeImmediate(index)) {
+    cmp_tagged(FieldOperand(object, HeapObject::kMapOffset),
+               Immediate(static_cast<uint32_t>(ReadOnlyRootPtr(index))));
+    return;
+  }
+  LoadMap(scratch, object);
+  CompareRoot(scratch, index);
+}
+
 inline void MaglevAssembler::CompareInstanceTypeRange(
     Register map, InstanceType lower_limit, InstanceType higher_limit) {
   CompareInstanceTypeRange(map, kScratchRegister, lower_limit, higher_limit);
