diff --git a/src/builtins/x64/builtins-x64.cc b/src/builtins/x64/builtins-x64.cc
index 3dfea65f0f..cab1032733 100644
--- a/src/builtins/x64/builtins-x64.cc
+++ b/src/builtins/x64/builtins-x64.cc
@@ -1676,8 +1676,8 @@ void Builtins::Generate_BaselineOutOfLinePrologue(MacroAssembler* masm) {
           Immediate(0));
   __ Push(bytecode_array);
 
-  // Horrible hack: This should be the bytecode offset, but we calculate that
-  // from the PC, so we cache the feedback vector in there instead.
+  // Baseline code frames store the feedback vector where interpreter would
+  // store the bytecode offset.
   __ Push(feedback_vector);
 
   __ RecordComment("]");
