diff --git a/src/codegen/loong64/assembler-loong64-inl.h b/src/codegen/loong64/assembler-loong64-inl.h
index 910f7e86f40..0475506baa0 100644
--- a/src/codegen/loong64/assembler-loong64-inl.h
+++ b/src/codegen/loong64/assembler-loong64-inl.h
@@ -137,7 +137,7 @@ Address RelocInfo::target_internal_reference_address() {
   return pc_;
 }
 
-Handle<CodeT> Assembler::relative_code_target_object_handle_at(
+Handle<CodeDataContainer> Assembler::relative_code_target_object_handle_at(
     Address pc) const {
   Instr instr = Assembler::instr_at(pc);
   int32_t code_target_index = instr & kImm26Mask;
