diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 000082af9d4..2ac2b603935 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -578,10 +578,11 @@ inline void MaglevAssembler::CompareInstanceTypeRange(
 
 inline void MaglevAssembler::CompareTagged(Register reg,
                                            Handle<HeapObject> obj) {
-  ScratchRegisterScope temps(this);
-  Register scratch = temps.Acquire();
-  Move(scratch, obj);
-  Cmp(reg, scratch);
+  CmpTagged(reg, Operand(obj, RelocInfo::COMPRESSED_EMBEDDED_OBJECT));
+}
+
+inline void MaglevAssembler::CompareTagged(Register src1, Register src2) {
+  CmpTagged(src1, src2);
 }
 
 inline void MaglevAssembler::CompareInt32(Register reg, int32_t imm) {
