diff --git a/src/diagnostics/ppc/unwinder-ppc.cc b/src/diagnostics/ppc/unwinder-ppc.cc
index 43c6acb609e..52c22221ffc 100644
--- a/src/diagnostics/ppc/unwinder-ppc.cc
+++ b/src/diagnostics/ppc/unwinder-ppc.cc
@@ -3,6 +3,9 @@
 // found in the LICENSE file.
 #include "src/diagnostics/unwinder.h"
 namespace v8 {
+
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 }  // namespace v8
