diff --git a/src/diagnostics/arm/unwinder-arm.cc b/src/diagnostics/arm/unwinder-arm.cc
index 846bbfe6bc6..e0e2f0e91f4 100644
--- a/src/diagnostics/arm/unwinder-arm.cc
+++ b/src/diagnostics/arm/unwinder-arm.cc
@@ -2,7 +2,10 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
+#include <memory>
+
 #include "include/v8-unwinder-state.h"
+#include "include/v8.h"
 #include "src/diagnostics/unwinder.h"
 #include "src/execution/frame-constants.h"
 
