diff --git a/src/diagnostics/x64/unwinder-x64.cc b/src/diagnostics/x64/unwinder-x64.cc
index 5a92512a17b..03144580055 100644
--- a/src/diagnostics/x64/unwinder-x64.cc
+++ b/src/diagnostics/x64/unwinder-x64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
-- 
2.35.1

