diff --git a/src/diagnostics/riscv64/unwinder-riscv64.cc b/src/diagnostics/riscv64/unwinder-riscv64.cc
index ccfb9268ea..84d2e41cfc 100644
--- a/src/diagnostics/riscv64/unwinder-riscv64.cc
+++ b/src/diagnostics/riscv64/unwinder-riscv64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
