diff --git a/src/diagnostics/s390/unwinder-s390.cc b/src/diagnostics/s390/unwinder-s390.cc
index 43c6acb609..52c22221ff 100644
--- a/src/diagnostics/s390/unwinder-s390.cc
+++ b/src/diagnostics/s390/unwinder-s390.cc
@@ -3,6 +3,9 @@
 // found in the LICENSE file.
 #include "src/diagnostics/unwinder.h"
 namespace v8 {
+
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 }  // namespace v8
