diff --git a/src/diagnostics/arm64/unwinder-arm64.cc b/src/diagnostics/arm64/unwinder-arm64.cc
index 5a92512a17..0314458005 100644
--- a/src/diagnostics/arm64/unwinder-arm64.cc
+++ b/src/diagnostics/arm64/unwinder-arm64.cc
@@ -6,6 +6,8 @@
 
 namespace v8 {
 
+struct RegisterState;
+
 void GetCalleeSavedRegistersFromEntryFrame(void* fp,
                                            RegisterState* register_state) {}
 
