diff --git a/src/baseline/riscv64/baseline-assembler-riscv64-inl.h b/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
index b38fbe83313..e0667d3472b 100644
--- a/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
+++ b/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
@@ -464,7 +464,7 @@ void BaselineAssembler::StoreTaggedFieldWithWriteBarrier(Register target,
   ScratchRegisterScope temps(this);
   Register tmp = temps.AcquireScratch();
   __ RecordWriteField(target, offset, value, tmp, kRAHasNotBeenSaved,
-                      kDontSaveFPRegs);
+                      SaveFPRegsMode::kIgnore);
 }
 void BaselineAssembler::StoreTaggedFieldNoWriteBarrier(Register target,
                                                        int offset,
