diff --git a/src/builtins/s390/builtins-s390.cc b/src/builtins/s390/builtins-s390.cc
index ec0074bdb19..e57a53962a1 100644
--- a/src/builtins/s390/builtins-s390.cc
+++ b/src/builtins/s390/builtins-s390.cc
@@ -3278,7 +3278,7 @@ static void CallApiFunctionAndReturn(MacroAssembler* masm,
     __ LoadU64(prev_next_address_reg, next_mem_op);
     __ LoadU64(prev_limit_reg, limit_mem_op);
     __ LoadU32(prev_level_reg, level_mem_op);
-    __ AddS64(scratch, Operand(1));
+    __ AddS64(scratch, prev_level_reg, Operand(1));
     __ StoreU32(scratch, level_mem_op);
   }
 
-- 
2.35.1

