diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 67c17fdadb8..800507a206f 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -484,9 +484,8 @@ void MaglevAssembler::TruncateDoubleToInt32(Register dst, DoubleRegister src) {
   movl(dst, dst);
 }
 
-void MaglevAssembler::CheckedTruncateDoubleToInt32(Register dst,
-                                                   DoubleRegister src,
-                                                   Label* fail) {
+void MaglevAssembler::TryTruncateDoubleToInt32(Register dst, DoubleRegister src,
+                                               Label* fail) {
   DoubleRegister converted_back = kScratchDoubleReg;
 
   // Convert the input float64 value to int32.
