diff --git a/src/maglev/arm64/maglev-assembler-arm64.cc b/src/maglev/arm64/maglev-assembler-arm64.cc
index 94e8c4a0b7d..2a2969877c6 100644
--- a/src/maglev/arm64/maglev-assembler-arm64.cc
+++ b/src/maglev/arm64/maglev-assembler-arm64.cc
@@ -701,9 +701,8 @@ void MaglevAssembler::TruncateDoubleToInt32(Register dst, DoubleRegister src) {
   Mov(dst, Operand(dst.W(), UXTW));
 }
 
-void MaglevAssembler::CheckedTruncateDoubleToInt32(Register dst,
-                                                   DoubleRegister src,
-                                                   Label* fail) {
+void MaglevAssembler::TryTruncateDoubleToInt32(Register dst, DoubleRegister src,
+                                               Label* fail) {
   ScratchRegisterScope temps(this);
   DoubleRegister converted_back = temps.AcquireDouble();
 
