diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 12be9923d64..25e8a58a4d6 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -480,6 +480,12 @@ inline void MaglevAssembler::MoveRepr(MachineRepresentation repr, Dest dst,
       UNREACHABLE();
   }
 }
+template <>
+inline void MaglevAssembler::MoveRepr(MachineRepresentation repr,
+                                      MemOperand dst, MemOperand src) {
+  MoveRepr(repr, kScratchRegister, src);
+  MoveRepr(repr, dst, kScratchRegister);
+}
 
 inline Condition ToCondition(AssertCondition cond) {
   switch (cond) {
-- 
2.35.1

