diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 27422fe80f4..e1b7dadeb77 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -456,10 +456,8 @@ inline void MaglevAssembler::LoadHeapNumberValue(DoubleRegister result,
   Movsd(result, FieldOperand(heap_number, HeapNumber::kValueOffset));
 }
 
-inline void MaglevAssembler::SmiToDouble(DoubleRegister result, Register smi) {
-  AssertSmi(smi);
-  SmiUntag(smi);
-  Cvtlsi2sd(result, smi);
+inline void MaglevAssembler::Int32ToDouble(DoubleRegister result, Register n) {
+  Cvtlsi2sd(result, n);
 }
 
 inline void MaglevAssembler::Pop(Register dst) { MacroAssembler::Pop(dst); }
