diff --git a/src/codegen/x64/register-x64.h b/src/codegen/x64/register-x64.h
index 13ed62bdd1e..c79b33ff2bb 100644
--- a/src/codegen/x64/register-x64.h
+++ b/src/codegen/x64/register-x64.h
@@ -289,7 +289,6 @@ constexpr Register kRuntimeCallFunctionRegister = rbx;
 constexpr Register kRuntimeCallArgCountRegister = rax;
 constexpr Register kRuntimeCallArgvRegister = r15;
 constexpr Register kWasmInstanceRegister = rsi;
-constexpr Register kWasmTrapHandlerFaultAddressRegister = r10;
 
 // Default scratch register used by MacroAssembler (and other code that needs
 // a spare register). The register isn't callee save, and not used by the
