diff --git a/src/codegen/arm/assembler-arm-inl.h b/src/codegen/arm/assembler-arm-inl.h
index 945acfb187d..c26e05f71cb 100644
--- a/src/codegen/arm/assembler-arm-inl.h
+++ b/src/codegen/arm/assembler-arm-inl.h
@@ -156,7 +156,7 @@ void RelocInfo::WipeOut() {
   }
 }
 
-Handle<CodeDataContainer> Assembler::relative_code_target_object_handle_at(
+Handle<Code> Assembler::relative_code_target_object_handle_at(
     Address pc) const {
   Instruction* branch = Instruction::At(pc);
   int code_target_index = branch->GetBranchOffset() / kInstrSize;
