diff --git a/src/codegen/ia32/assembler-ia32-inl.h b/src/codegen/ia32/assembler-ia32-inl.h
index 63c83118e2d..5f937f01bf6 100644
--- a/src/codegen/ia32/assembler-ia32-inl.h
+++ b/src/codegen/ia32/assembler-ia32-inl.h
@@ -166,7 +166,7 @@ void Assembler::emit(uint32_t x, RelocInfo::Mode rmode) {
   emit(x);
 }
 
-void Assembler::emit(Handle<CodeDataContainer> code, RelocInfo::Mode rmode) {
+void Assembler::emit(Handle<Code> code, RelocInfo::Mode rmode) {
   emit(code.address(), rmode);
 }
 
