diff --git a/src/compiler/backend/ppc/unwinding-info-writer-ppc.h b/src/compiler/backend/ppc/unwinding-info-writer-ppc.h
index e96a48308fb..b11e00be2d1 100644
--- a/src/compiler/backend/ppc/unwinding-info-writer-ppc.h
+++ b/src/compiler/backend/ppc/unwinding-info-writer-ppc.h
@@ -49,7 +49,7 @@ class UnwindingInfoWriter {
   }
 
  private:
-  bool enabled() const { return FLAG_perf_prof_unwinding_info; }
+  bool enabled() const { return v8_flags.perf_prof_unwinding_info; }
 
   class BlockInitialState : public ZoneObject {
    public:
