diff --git a/src/compiler/backend/x64/unwinding-info-writer-x64.h b/src/compiler/backend/x64/unwinding-info-writer-x64.h
index c85ad46a633..188a66f56f7 100644
--- a/src/compiler/backend/x64/unwinding-info-writer-x64.h
+++ b/src/compiler/backend/x64/unwinding-info-writer-x64.h
@@ -53,7 +53,7 @@ class UnwindingInfoWriter {
   }
 
  private:
-  bool enabled() const { return FLAG_perf_prof_unwinding_info; }
+  bool enabled() const { return v8_flags.perf_prof_unwinding_info; }
 
   class BlockInitialState : public ZoneObject {
    public:
