diff --git a/src/compiler/backend/s390/unwinding-info-writer-s390.h b/src/compiler/backend/s390/unwinding-info-writer-s390.h
index 2202c285957..7c472723c48 100644
--- a/src/compiler/backend/s390/unwinding-info-writer-s390.h
+++ b/src/compiler/backend/s390/unwinding-info-writer-s390.h
@@ -49,7 +49,7 @@ class UnwindingInfoWriter {
   }
 
  private:
-  bool enabled() const { return FLAG_perf_prof_unwinding_info; }
+  bool enabled() const { return v8_flags.perf_prof_unwinding_info; }
 
   class BlockInitialState : public ZoneObject {
    public:
