diff --git a/src/compiler/backend/arm64/unwinding-info-writer-arm64.h b/src/compiler/backend/arm64/unwinding-info-writer-arm64.h
index 36788735de1..e142e13d69d 100644
--- a/src/compiler/backend/arm64/unwinding-info-writer-arm64.h
+++ b/src/compiler/backend/arm64/unwinding-info-writer-arm64.h
@@ -49,7 +49,7 @@ class UnwindingInfoWriter {
   }
 
  private:
-  bool enabled() const { return FLAG_perf_prof_unwinding_info; }
+  bool enabled() const { return v8_flags.perf_prof_unwinding_info; }
 
   class BlockInitialState : public ZoneObject {
    public:
