diff --git a/src/compiler/backend/arm/unwinding-info-writer-arm.h b/src/compiler/backend/arm/unwinding-info-writer-arm.h
index 6b9ade0c483..de81134bd14 100644
--- a/src/compiler/backend/arm/unwinding-info-writer-arm.h
+++ b/src/compiler/backend/arm/unwinding-info-writer-arm.h
@@ -49,7 +49,7 @@ class UnwindingInfoWriter {
   }
 
  private:
-  bool enabled() const { return FLAG_perf_prof_unwinding_info; }
+  bool enabled() const { return v8_flags.perf_prof_unwinding_info; }
 
   class BlockInitialState : public ZoneObject {
    public:
