diff --git a/src/codegen/x64/macro-assembler-x64.h b/src/codegen/x64/macro-assembler-x64.h
index d2b42432f5..d380acdf13 100644
--- a/src/codegen/x64/macro-assembler-x64.h
+++ b/src/codegen/x64/macro-assembler-x64.h
@@ -838,7 +838,7 @@ class V8_EXPORT_PRIVATE MacroAssembler : public TurboAssembler {
                                       JumpMode jump_mode = JumpMode::kJump);
   void LoadTieringStateAndJumpIfNeedsProcessing(
       Register optimization_state, Register feedback_vector,
-      Label* has_optimized_code_or_state);
+      CodeKind current_code_kind, Label* has_optimized_code_or_state);
   void MaybeOptimizeCodeOrTailCallOptimizedCodeSlot(
       Register optimization_state, Register feedback_vector, Register closure,
       JumpMode jump_mode = JumpMode::kJump);
