diff --git a/src/builtins/ppc/builtins-ppc.cc b/src/builtins/ppc/builtins-ppc.cc
index f8d00eb74d5..f5a439175b8 100644
--- a/src/builtins/ppc/builtins-ppc.cc
+++ b/src/builtins/ppc/builtins-ppc.cc
@@ -39,7 +39,7 @@ static void AssertCodeDataContainerIsBaseline(MacroAssembler* masm,
                                               Register code, Register scratch) {
   DCHECK(!AreAliased(code, scratch));
   // Verify that the code kind is baseline code via the CodeKind.
-  __ LoadU32(scratch, FieldMemOperand(code, CodeDataContainer::kFlagsOffset));
+  __ LoadU16(scratch, FieldMemOperand(code, CodeDataContainer::kFlagsOffset));
   __ DecodeField<CodeDataContainer::KindField>(scratch);
   __ CmpS64(scratch, Operand(static_cast<int>(CodeKind::BASELINE)), r0);
   __ Assert(eq, AbortReason::kExpectedBaselineData);
