diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 516c617b732..2238d500f79 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -76,20 +76,6 @@ void MaglevAssembler::Allocate(RegisterSnapshot register_snapshot,
   bind(*done);
 }
 
-void MaglevAssembler::AllocateTwoByteString(RegisterSnapshot register_snapshot,
-                                            Register result, int length) {
-  int size = SeqTwoByteString::SizeFor(length);
-  Allocate(register_snapshot, result, size);
-  LoadRoot(kScratchRegister, RootIndex::kStringMap);
-  StoreTaggedField(FieldOperand(result, size - kObjectAlignment), Immediate(0));
-  StoreTaggedField(FieldOperand(result, HeapObject::kMapOffset),
-                   kScratchRegister);
-  StoreTaggedField(FieldOperand(result, Name::kRawHashFieldOffset),
-                   Immediate(Name::kEmptyHashField));
-  StoreTaggedField(FieldOperand(result, String::kLengthOffset),
-                   Immediate(length));
-}
-
 void MaglevAssembler::LoadSingleCharacterString(Register result,
                                                 Register char_code,
                                                 Register scratch) {
@@ -112,7 +98,7 @@ void MaglevAssembler::StoreTaggedFieldWithWriteBarrier(
   DCHECK_NE(object, kScratchRegister);
   DCHECK_NE(value, kScratchRegister);
   AssertNotSmi(object);
-  StoreTaggedField(FieldOperand(object, offset), value);
+  MacroAssembler::StoreTaggedField(FieldOperand(object, offset), value);
 
   ZoneLabelRef done(this);
   Label* deferred_write_barrier = MakeDeferredCode(
-- 
2.35.1

