diff --git a/src/codegen/x64/macro-assembler-x64.h b/src/codegen/x64/macro-assembler-x64.h
index 2a314d6fb2b..2973a77419b 100644
--- a/src/codegen/x64/macro-assembler-x64.h
+++ b/src/codegen/x64/macro-assembler-x64.h
@@ -542,10 +542,9 @@ class V8_EXPORT_PRIVATE MacroAssembler
   // Print a message to stdout and abort execution.
   void Abort(AbortReason msg);
 
+  // Check that the stack is aligned.
   void CheckStackAlignment();
 
-  void AlignStackPointer();
-
   // Activation support.
   void EnterFrame(StackFrame::Type type);
   void EnterFrame(StackFrame::Type type, bool load_constant_pool_pointer_reg) {
