diff --git a/src/baseline/loong64/baseline-assembler-loong64-inl.h b/src/baseline/loong64/baseline-assembler-loong64-inl.h
index cc5694554a..5d84367b7a 100644
--- a/src/baseline/loong64/baseline-assembler-loong64-inl.h
+++ b/src/baseline/loong64/baseline-assembler-loong64-inl.h
@@ -383,8 +383,8 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
   {
     ScratchRegisterScope temps(this);
     Register scratch = temps.AcquireScratch();
-    __ TestCodeTIsMarkedForDeoptimizationAndJump(scratch_and_result, scratch,
-                                                 eq, on_result);
+    __ TestCodeDataContainerIsMarkedForDeoptimizationAndJump(
+        scratch_and_result, scratch, eq, on_result);
     __ li(scratch, __ ClearedValue());
     StoreTaggedFieldNoWriteBarrier(
         feedback_vector, FeedbackVector::OffsetOfElementAt(slot.ToInt()),
