diff --git a/src/baseline/arm/baseline-assembler-arm-inl.h b/src/baseline/arm/baseline-assembler-arm-inl.h
index 5cb855e416..22aee5d00d 100644
--- a/src/baseline/arm/baseline-assembler-arm-inl.h
+++ b/src/baseline/arm/baseline-assembler-arm-inl.h
@@ -409,7 +409,8 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
   {
     ScratchRegisterScope temps(this);
     Register scratch = temps.AcquireScratch();
-    __ TestCodeTIsMarkedForDeoptimization(scratch_and_result, scratch);
+    __ TestCodeDataContainerIsMarkedForDeoptimization(scratch_and_result,
+                                                      scratch);
     __ b(eq, on_result);
     __ mov(scratch, __ ClearedValue());
     StoreTaggedFieldNoWriteBarrier(
