diff --git a/src/baseline/riscv/baseline-assembler-riscv-inl.h b/src/baseline/riscv/baseline-assembler-riscv-inl.h
index 59cffa9e292..d3cf5fffee9 100644
--- a/src/baseline/riscv/baseline-assembler-riscv-inl.h
+++ b/src/baseline/riscv/baseline-assembler-riscv-inl.h
@@ -383,7 +383,7 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
   // Is it marked_for_deoptimization? If yes, clear the slot.
   {
     ScratchRegisterScope temps(this);
-    __ JumpIfCodeTIsMarkedForDeoptimization(
+    __ JumpIfCodeDataContainerIsMarkedForDeoptimization(
         scratch_and_result, temps.AcquireScratch(), &clear_slot);
     Jump(on_result);
   }
