diff --git a/src/baseline/s390/baseline-assembler-s390-inl.h b/src/baseline/s390/baseline-assembler-s390-inl.h
index c04e7734aef..415aae0bb3d 100644
--- a/src/baseline/s390/baseline-assembler-s390-inl.h
+++ b/src/baseline/s390/baseline-assembler-s390-inl.h
@@ -550,7 +550,8 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
   {
     ScratchRegisterScope temps(this);
     Register scratch = temps.AcquireScratch();
-    __ TestCodeTIsMarkedForDeoptimization(scratch_and_result, scratch);
+    __ TestCodeDataContainerIsMarkedForDeoptimization(scratch_and_result,
+                                                      scratch);
     __ beq(on_result);
     __ mov(scratch, __ ClearedValue());
     StoreTaggedFieldNoWriteBarrier(
