diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 0e1e24303aa..d4422245a70 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -513,6 +513,27 @@ inline MemOperand MaglevAssembler::StackSlotOperand(StackSlot stack_slot) {
 
 inline void MaglevAssembler::IncrementInt32(Register reg) { incl(reg); }
 
+inline void MaglevAssembler::IncrementAddress(Register reg, int32_t delta) {
+  leaq(reg, MemOperand(reg, delta));
+}
+
+inline void MaglevAssembler::LoadAddress(Register dst, MemOperand location) {
+  leaq(dst, location);
+}
+
+inline int MaglevAssembler::PushOrSetReturnAddressTo(Label* target) {
+  leaq(kScratchRegister, Operand(target));
+  pushq(kScratchRegister);
+  return 1;
+}
+
+inline void MaglevAssembler::EmitEnterExitFrame(int extra_slots,
+                                                StackFrame::Type frame_type,
+                                                Register c_function,
+                                                Register scratch) {
+  EnterExitFrame(extra_slots, frame_type, c_function);
+}
+
 inline void MaglevAssembler::Move(StackSlot dst, Register src) {
   movq(StackSlotOperand(dst), src);
 }
-- 
2.35.1

