diff --git a/src/codegen/riscv64/assembler-riscv64.cc b/src/codegen/riscv64/assembler-riscv64.cc
index f574d0aaca..ef06240ffa 100644
--- a/src/codegen/riscv64/assembler-riscv64.cc
+++ b/src/codegen/riscv64/assembler-riscv64.cc
@@ -2760,6 +2760,8 @@ void Assembler::RelocateRelativeReference(RelocInfo::Mode rmode, Address pc,
 
 void Assembler::FixOnHeapReferences() { UNIMPLEMENTED(); }
 
+void Assembler::FixOnHeapReferencesToHandles() { UNIMPLEMENTED(); }
+
 void Assembler::GrowBuffer() {
   DEBUG_PRINTF("GrowBuffer: %p -> ", buffer_start_);
   // Compute new buffer size.
