diff --git a/src/codegen/s390/assembler-s390.cc b/src/codegen/s390/assembler-s390.cc
index 7dafd3818a3..fbd136b0a34 100644
--- a/src/codegen/s390/assembler-s390.cc
+++ b/src/codegen/s390/assembler-s390.cc
@@ -756,6 +756,12 @@ void Assembler::FixOnHeapReferences() {
   UNREACHABLE();
 }
 
+void Assembler::FixOnHeapReferencesToHandles() {
+  // TODO(v8:11872) This function should never be called if Sparkplug on heap
+  // compilation is not supported.
+  UNREACHABLE();
+}
+
 void Assembler::GrowBuffer(int needed) {
   DCHECK_EQ(buffer_start_, buffer_->start());
 
