diff --git a/src/execution/ppc/simulator-ppc.cc b/src/execution/ppc/simulator-ppc.cc
index cc2dd1482e7..5d7b8b77d35 100644
--- a/src/execution/ppc/simulator-ppc.cc
+++ b/src/execution/ppc/simulator-ppc.cc
@@ -39,7 +39,7 @@ DEFINE_LAZY_LEAKY_OBJECT_GETTER(Simulator::GlobalMonitor,
 // SScanF not being implemented in a platform independent way through
 // ::v8::internal::OS in the same way as SNPrintF is that the
 // Windows C Run-Time Library does not provide vsscanf.
-#define SScanF sscanf  // NOLINT
+#define SScanF sscanf
 
 // The PPCDebugger class is used by the simulator while debugging simulated
 // PowerPC code.
@@ -4928,7 +4928,7 @@ void Simulator::ExecuteGeneric(Instruction* instr) {
       break;
     }
   }
-}  // NOLINT
+}
 
 void Simulator::Trace(Instruction* instr) {
   disasm::NameConverter converter;
