diff --git a/src/deoptimizer/loong64/deoptimizer-loong64.cc b/src/deoptimizer/loong64/deoptimizer-loong64.cc
index 73d71036ed5..5692e16969c 100644
--- a/src/deoptimizer/loong64/deoptimizer-loong64.cc
+++ b/src/deoptimizer/loong64/deoptimizer-loong64.cc
@@ -7,7 +7,6 @@
 namespace v8 {
 namespace internal {
 
-const bool Deoptimizer::kSupportsFixedDeoptExitSizes = true;
 const int Deoptimizer::kNonLazyDeoptExitSize = 2 * kInstrSize;
 const int Deoptimizer::kLazyDeoptExitSize = 2 * kInstrSize;
 
