diff --git a/src/wasm/baseline/ppc/liftoff-assembler-ppc-inl.h b/src/wasm/baseline/ppc/liftoff-assembler-ppc-inl.h
index 7f9ef0c2204..3112da664da 100644
--- a/src/wasm/baseline/ppc/liftoff-assembler-ppc-inl.h
+++ b/src/wasm/baseline/ppc/liftoff-assembler-ppc-inl.h
@@ -2780,7 +2780,8 @@ void LiftoffStackSlots::Construct(int param_slots) {
           case kF32: {
             asm_->AllocateStackSpace(stack_decrement - kSystemPointerSize);
             asm_->LoadF32(kScratchDoubleReg,
-                          liftoff::GetStackSlot(slot.src_offset_), r0);
+                          liftoff::GetStackSlot(slot.src_offset_ + stack_bias),
+                          r0);
             asm_->AddS64(sp, sp, Operand(-kSystemPointerSize));
             asm_->StoreF32(kScratchDoubleReg, MemOperand(sp), r0);
             break;
