diff --git a/src/codegen/mips/constants-mips.h b/src/codegen/mips/constants-mips.h
index 3dda85ceda..4cd67bb176 100644
--- a/src/codegen/mips/constants-mips.h
+++ b/src/codegen/mips/constants-mips.h
@@ -320,7 +320,7 @@ enum SoftwareInterruptCodes {
 //   debugger.
 const uint32_t kMaxWatchpointCode = 31;
 const uint32_t kMaxStopCode = 127;
-STATIC_ASSERT(kMaxWatchpointCode < kMaxStopCode);
+static_assert(kMaxWatchpointCode < kMaxStopCode);
 
 // ----- Fields offset and length.
 const int kOpcodeShift = 26;
