diff --git a/src/codegen/s390/macro-assembler-s390.h b/src/codegen/s390/macro-assembler-s390.h
index a93eb42ac17..fb745789d5e 100644
--- a/src/codegen/s390/macro-assembler-s390.h
+++ b/src/codegen/s390/macro-assembler-s390.h
@@ -1229,6 +1229,11 @@ class V8_EXPORT_PRIVATE MacroAssembler : public MacroAssemblerBase {
                     Simd128Register scratch3);
   void I32x4DotI16x8S(Simd128Register dst, Simd128Register src1,
                       Simd128Register src2, Simd128Register scratch);
+  void I16x8DotI8x16S(Simd128Register dst, Simd128Register src1,
+                      Simd128Register src2, Simd128Register scratch);
+  void I32x4DotI8x16AddS(Simd128Register dst, Simd128Register src1,
+                         Simd128Register src2, Simd128Register src3,
+                         Simd128Register scratch1, Simd128Register scratch2);
   void I16x8Q15MulRSatS(Simd128Register dst, Simd128Register src1,
                         Simd128Register src2, Simd128Register scratch1,
                         Simd128Register scratch2, Simd128Register scratch3);
