diff --git a/src/codegen/arm64/assembler-arm64.cc b/src/codegen/arm64/assembler-arm64.cc
index 028fe48fe5c..cc975913b96 100644
--- a/src/codegen/arm64/assembler-arm64.cc
+++ b/src/codegen/arm64/assembler-arm64.cc
@@ -4493,8 +4493,8 @@ void Assembler::RecordVeneerPool(int location_offset, int size) {
 
 void Assembler::EmitVeneers(bool force_emit, bool need_protection,
                             size_t margin) {
-  ASM_CODE_COMMENT_STRING(this, "Veneers");
   BlockPoolsScope scope(this, PoolEmissionCheck::kSkip);
+  RecordComment("[ Veneers");
 
   // The exact size of the veneer pool must be recorded (see the comment at the
   // declaration site of RecordConstPool()), but computing the number of
@@ -4587,11 +4587,12 @@ void Assembler::EmitVeneers(bool force_emit, bool need_protection,
   RecordVeneerPool(veneer_pool_relocinfo_loc, pool_size);
 
   bind(&end);
+
+  RecordComment("]");
 }
 
 void Assembler::CheckVeneerPool(bool force_emit, bool require_jump,
                                 size_t margin) {
-  ASM_CODE_COMMENT(this);
   // There is nothing to do if there are no pending veneer pool entries.
   if (unresolved_branches_.empty()) {
     DCHECK_EQ(next_veneer_pool_check_, kMaxInt);
