diff --git a/src/baseline/arm64/baseline-assembler-arm64-inl.h b/src/baseline/arm64/baseline-assembler-arm64-inl.h
index 0d2f15c0271..6aa1419e920 100644
--- a/src/baseline/arm64/baseline-assembler-arm64-inl.h
+++ b/src/baseline/arm64/baseline-assembler-arm64-inl.h
@@ -448,9 +448,8 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
                                                 FeedbackSlot slot,
                                                 Label* on_result,
                                                 Label::Distance) {
-  __ TryLoadOptimizedOsrCode(scratch_and_result, CodeKind::MAGLEV,
-                             feedback_vector, slot, on_result,
-                             Label::Distance::kFar);
+  __ TryLoadOptimizedOsrCode(scratch_and_result, feedback_vector, slot,
+                             on_result, Label::Distance::kFar);
 }
 
 void BaselineAssembler::AddToInterruptBudgetAndJumpIfNotExceeded(
