diff --git a/src/execution/ia32/frame-constants-ia32.cc b/src/execution/ia32/frame-constants-ia32.cc
index 16e9e75a44..bdf48e2b8f 100644
--- a/src/execution/ia32/frame-constants-ia32.cc
+++ b/src/execution/ia32/frame-constants-ia32.cc
@@ -27,6 +27,12 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
 
