diff --git a/src/execution/s390/frame-constants-s390.cc b/src/execution/s390/frame-constants-s390.cc
index 50f3445556c..86071459617 100644
--- a/src/execution/s390/frame-constants-s390.cc
+++ b/src/execution/s390/frame-constants-s390.cc
@@ -26,6 +26,12 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
 
