diff --git a/src/execution/x64/frame-constants-x64.cc b/src/execution/x64/frame-constants-x64.cc
index fb242505dd0..df612a08cfd 100644
--- a/src/execution/x64/frame-constants-x64.cc
+++ b/src/execution/x64/frame-constants-x64.cc
@@ -26,6 +26,14 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  // Include one extra slot for the single argument into StackGuardWithGap +
+  // register input count.
+  return StandardFrameConstants::kFixedFrameSizeFromFp +
+         (1 + register_input_count) * kSystemPointerSize;
+}
+
 }  // namespace internal
 }  // namespace v8
 
