diff --git a/src/execution/arm64/frame-constants-arm64.cc b/src/execution/arm64/frame-constants-arm64.cc
index 96f6f25e75..c7f61b3fc4 100644
--- a/src/execution/arm64/frame-constants-arm64.cc
+++ b/src/execution/arm64/frame-constants-arm64.cc
@@ -35,6 +35,16 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return rounded_slot_count - slot_count;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  // Include any paddings from kFixedFrameSizeFromFp, an extra slot + padding
+  // for the single argument into StackGuardWithGap and finally padded register
+  // input count.
+  int slot_count = RoundUp(StandardFrameConstants::kFixedSlotCountFromFp, 2) +
+                   2 /* argument */ + RoundUp(register_input_count, 2);
+  return slot_count * kSystemPointerSize;
+}
+
 }  // namespace internal
 }  // namespace v8
 
