diff --git a/src/execution/mips64/frame-constants-mips64.cc b/src/execution/mips64/frame-constants-mips64.cc
index cfe899730ca..66f14f85167 100644
--- a/src/execution/mips64/frame-constants-mips64.cc
+++ b/src/execution/mips64/frame-constants-mips64.cc
@@ -26,6 +26,12 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
 
