diff --git a/src/execution/arm/frame-constants-arm.cc b/src/execution/arm/frame-constants-arm.cc
index 7a72dab870a..bcf26f489ae 100644
--- a/src/execution/arm/frame-constants-arm.cc
+++ b/src/execution/arm/frame-constants-arm.cc
@@ -26,6 +26,12 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
 
