diff --git a/src/execution/loong64/frame-constants-loong64.cc b/src/execution/loong64/frame-constants-loong64.cc
index 4bd809266c6..86d271d207c 100644
--- a/src/execution/loong64/frame-constants-loong64.cc
+++ b/src/execution/loong64/frame-constants-loong64.cc
@@ -26,6 +26,12 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
 
