diff --git a/src/execution/ppc/frame-constants-ppc.cc b/src/execution/ppc/frame-constants-ppc.cc
index 45a53b07b6..f4256a0e63 100644
--- a/src/execution/ppc/frame-constants-ppc.cc
+++ b/src/execution/ppc/frame-constants-ppc.cc
@@ -29,6 +29,12 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
 
