diff --git a/src/execution/riscv/frame-constants-riscv.cc b/src/execution/riscv/frame-constants-riscv.cc
index 833af91e7e6..3fbbc7ab64a 100644
--- a/src/execution/riscv/frame-constants-riscv.cc
+++ b/src/execution/riscv/frame-constants-riscv.cc
@@ -23,5 +23,11 @@ int BuiltinContinuationFrameConstants::PaddingSlotCount(int register_count) {
   return 0;
 }
 
+// static
+intptr_t MaglevFrame::StackGuardFrameSize(int register_input_count) {
+  USE(register_input_count);
+  UNREACHABLE();
+}
+
 }  // namespace internal
 }  // namespace v8
