diff --git a/src/codegen/x64/register-x64.h b/src/codegen/x64/register-x64.h
index a54880a00cd..6adf816dd3c 100644
--- a/src/codegen/x64/register-x64.h
+++ b/src/codegen/x64/register-x64.h
@@ -295,8 +295,6 @@ constexpr Register kPtrComprCageBaseRegister = r14;  // callee save
 constexpr Register kPtrComprCageBaseRegister = no_reg;
 #endif
 
-constexpr Register kOffHeapTrampolineRegister = kScratchRegister;
-
 constexpr DoubleRegister kFPReturnRegister0 = xmm0;
 
 }  // namespace internal
