diff --git a/src/compiler/backend/x64/instruction-codes-x64.h b/src/compiler/backend/x64/instruction-codes-x64.h
index bf2c9b00c7..c2c6ca946e 100644
--- a/src/compiler/backend/x64/instruction-codes-x64.h
+++ b/src/compiler/backend/x64/instruction-codes-x64.h
@@ -163,8 +163,8 @@ namespace compiler {
   V(X64MovqDecompressTaggedPointer)                  \
   V(X64MovqDecompressAnyTagged)                      \
   V(X64MovqCompressTagged)                           \
-  V(X64MovqEncodeCagedPointer)                       \
-  V(X64MovqDecodeCagedPointer)                       \
+  V(X64MovqEncodeSandboxedPointer)                   \
+  V(X64MovqDecodeSandboxedPointer)                   \
   V(X64BitcastFI)                                    \
   V(X64BitcastDL)                                    \
   V(X64BitcastIF)                                    \
