diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index e1b7dadeb77..b4104040d90 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -444,6 +444,12 @@ inline void MaglevAssembler::JumpIfSmi(Register src, Label* on_smi,
   MacroAssembler::JumpIfSmi(src, on_smi, distance);
 }
 
+void MaglevAssembler::JumpIfByte(Condition cc, Register value, int32_t byte,
+                                 Label* target, Label::Distance distance) {
+  cmpb(value, Immediate(byte));
+  j(cc, target, distance);
+}
+
 void MaglevAssembler::CompareInt32AndJumpIf(Register r1, Register r2,
                                             Condition cond, Label* target,
                                             Label::Distance distance) {
-- 
2.35.1

