diff --git a/src/execution/ppc/simulator-ppc.h b/src/execution/ppc/simulator-ppc.h
index e3dbd1efa9..bfabd9af5c 100644
--- a/src/execution/ppc/simulator-ppc.h
+++ b/src/execution/ppc/simulator-ppc.h
@@ -21,7 +21,6 @@
 #include "src/base/hashmap.h"
 #include "src/base/lazy-instance.h"
 #include "src/base/platform/mutex.h"
-#include "src/base/platform/wrappers.h"
 #include "src/codegen/assembler.h"
 #include "src/codegen/ppc/constants-ppc.h"
 #include "src/execution/simulator-base.h"
