diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 46b800f9d4a..a7b3e69c00c 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -537,31 +537,6 @@ inline void MaglevAssembler::MoveRepr(MachineRepresentation repr,
   MoveRepr(repr, dst, kScratchRegister);
 }
 
-inline Condition ToCondition(AssertCondition cond) {
-  switch (cond) {
-    case AssertCondition::kLess:
-      return less;
-    case AssertCondition::kLessOrEqual:
-      return less_equal;
-    case AssertCondition::kGreater:
-      return greater;
-    case AssertCondition::kGeaterOrEqual:
-      return greater_equal;
-    case AssertCondition::kBelow:
-      return below;
-    case AssertCondition::kBelowOrEqual:
-      return below_equal;
-    case AssertCondition::kAbove:
-      return above;
-    case AssertCondition::kAboveOrEqual:
-      return above_equal;
-    case AssertCondition::kEqual:
-      return equal;
-    case AssertCondition::kNotEqual:
-      return not_equal;
-  }
-}
-
 }  // namespace maglev
 }  // namespace internal
 }  // namespace v8
-- 
2.35.1

