diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 23643b8e9e1..143aad96c42 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -673,31 +673,6 @@ inline void MaglevAssembler::MoveRepr(MachineRepresentation repr,
   MoveRepr(repr, dst, scratch);
 }
 
-inline Condition ToCondition(AssertCondition cond) {
-  switch (cond) {
-    case AssertCondition::kLess:
-      return lt;
-    case AssertCondition::kLessOrEqual:
-      return le;
-    case AssertCondition::kGreater:
-      return gt;
-    case AssertCondition::kGeaterOrEqual:
-      return ge;
-    case AssertCondition::kBelow:
-      return lo;
-    case AssertCondition::kBelowOrEqual:
-      return ls;
-    case AssertCondition::kAbove:
-      return hi;
-    case AssertCondition::kAboveOrEqual:
-      return hs;
-    case AssertCondition::kEqual:
-      return eq;
-    case AssertCondition::kNotEqual:
-      return ne;
-  }
-}
-
 }  // namespace maglev
 }  // namespace internal
 }  // namespace v8
