diff --git a/test/cctest/test-helper-riscv64.cc b/test/cctest/test-helper-riscv64.cc
index e4cf6bd29b8..a7841da9c82 100644
--- a/test/cctest/test-helper-riscv64.cc
+++ b/test/cctest/test-helper-riscv64.cc
@@ -38,7 +38,7 @@ Handle<Code> AssembleCodeImpl(Isolate* isolate, Func assemble) {
   Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
   if (v8_flags.print_code) {
-    code->Print();
+    Print(*code);
   }
   return code;
 }
