diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 3c4ced54265..9c92fa72ead 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -192,6 +192,11 @@ inline void MaglevAssembler::DoubleToInt64Repr(Register dst,
   Movq(dst, src);
 }
 
+inline void MaglevAssembler::SmiTagInt32(Register obj, Label* fail) {
+  addl(obj, obj);
+  JumpIf(overflow, fail);
+}
+
 inline Condition MaglevAssembler::IsInt64Constant(Register reg,
                                                   int64_t constant) {
   movq(kScratchRegister, kHoleNanInt64);
diff --git a/test/mjsunit/maglev/regress/regress-crbug-1421237.js b/test/mjsunit/maglev/regress/regress-crbug-1421237.js
new file mode 100644
index 00000000000..6e06c80e5df
