diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index cef2b23c247..516c617b732 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -76,20 +76,6 @@ void MaglevAssembler::Allocate(RegisterSnapshot register_snapshot,
   bind(*done);
 }
 
-void MaglevAssembler::AllocateHeapNumber(RegisterSnapshot register_snapshot,
-                                         Register result,
-                                         DoubleRegister value) {
-  // In the case we need to call the runtime, we should spill the value
-  // register. Even if it is not live in the next node, otherwise the
-  // allocation call might trash it.
-  register_snapshot.live_double_registers.set(value);
-  Allocate(register_snapshot, result, HeapNumber::kSize);
-  LoadRoot(kScratchRegister, RootIndex::kHeapNumberMap);
-  StoreTaggedField(FieldOperand(result, HeapObject::kMapOffset),
-                   kScratchRegister);
-  Movsd(FieldOperand(result, HeapNumber::kValueOffset), value);
-}
-
 void MaglevAssembler::AllocateTwoByteString(RegisterSnapshot register_snapshot,
                                             Register result, int length) {
   int size = SeqTwoByteString::SizeFor(length);
-- 
2.35.1

