diff --git a/test/cctest/test-macro-assembler-riscv64.cc b/test/cctest/test-macro-assembler-riscv64.cc
index 0bf9b5b363..7148ac344c 100644
--- a/test/cctest/test-macro-assembler-riscv64.cc
+++ b/test/cctest/test-macro-assembler-riscv64.cc
@@ -1059,20 +1059,25 @@ TEST(macro_float_minmax_f32) {
   auto f = AssembleCode<F4>(
       GenerateMacroFloat32MinMax<FPURegister, Inputs, Results>);
 
-#define CHECK_MINMAX(src1, src2, min, max)                                    \
-  do {                                                                        \
-    Inputs inputs = {src1, src2};                                             \
-    Results results;                                                          \
-    f.Call(&inputs, &results, 0, 0, 0);                                       \
-    CHECK_EQ(bit_cast<uint32_t>(min), bit_cast<uint32_t>(results.min_abc_));  \
-    CHECK_EQ(bit_cast<uint32_t>(min), bit_cast<uint32_t>(results.min_aab_));  \
-    CHECK_EQ(bit_cast<uint32_t>(min), bit_cast<uint32_t>(results.min_aba_));  \
-    CHECK_EQ(bit_cast<uint32_t>(max), bit_cast<uint32_t>(results.max_abc_));  \
-    CHECK_EQ(bit_cast<uint32_t>(max), bit_cast<uint32_t>(results.max_aab_));  \
-    CHECK_EQ(                                                                 \
-        bit_cast<uint32_t>(max),                                              \
-        bit_cast<uint32_t>(results.max_aba_)); /* Use a bit_cast to correctly \
-                                                  identify -0.0 and NaNs. */  \
+#define CHECK_MINMAX(src1, src2, min, max)                                \
+  do {                                                                    \
+    Inputs inputs = {src1, src2};                                         \
+    Results results;                                                      \
+    f.Call(&inputs, &results, 0, 0, 0);                                   \
+    CHECK_EQ(base::bit_cast<uint32_t>(min),                               \
+             base::bit_cast<uint32_t>(results.min_abc_));                 \
+    CHECK_EQ(base::bit_cast<uint32_t>(min),                               \
+             base::bit_cast<uint32_t>(results.min_aab_));                 \
+    CHECK_EQ(base::bit_cast<uint32_t>(min),                               \
+             base::bit_cast<uint32_t>(results.min_aba_));                 \
+    CHECK_EQ(base::bit_cast<uint32_t>(max),                               \
+             base::bit_cast<uint32_t>(results.max_abc_));                 \
+    CHECK_EQ(base::bit_cast<uint32_t>(max),                               \
+             base::bit_cast<uint32_t>(results.max_aab_));                 \
+    CHECK_EQ(base::bit_cast<uint32_t>(max),                               \
+             base::bit_cast<uint32_t>(                                    \
+                 results.max_aba_)); /* Use a base::bit_cast to correctly \
+                              identify -0.0 and NaNs. */                  \
   } while (0)
 
   float nan_a = std::numeric_limits<float>::quiet_NaN();
@@ -1154,18 +1159,24 @@ TEST(macro_float_minmax_f64) {
   auto f = AssembleCode<F4>(
       GenerateMacroFloat64MinMax<DoubleRegister, Inputs, Results>);
 
-#define CHECK_MINMAX(src1, src2, min, max)                                   \
-  do {                                                                       \
-    Inputs inputs = {src1, src2};                                            \
-    Results results;                                                         \
-    f.Call(&inputs, &results, 0, 0, 0);                                      \
-    CHECK_EQ(bit_cast<uint64_t>(min), bit_cast<uint64_t>(results.min_abc_)); \
-    CHECK_EQ(bit_cast<uint64_t>(min), bit_cast<uint64_t>(results.min_aab_)); \
-    CHECK_EQ(bit_cast<uint64_t>(min), bit_cast<uint64_t>(results.min_aba_)); \
-    CHECK_EQ(bit_cast<uint64_t>(max), bit_cast<uint64_t>(results.max_abc_)); \
-    CHECK_EQ(bit_cast<uint64_t>(max), bit_cast<uint64_t>(results.max_aab_)); \
-    CHECK_EQ(bit_cast<uint64_t>(max), bit_cast<uint64_t>(results.max_aba_)); \
-    /* Use a bit_cast to correctly identify -0.0 and NaNs. */                \
+#define CHECK_MINMAX(src1, src2, min, max)                          \
+  do {                                                              \
+    Inputs inputs = {src1, src2};                                   \
+    Results results;                                                \
+    f.Call(&inputs, &results, 0, 0, 0);                             \
+    CHECK_EQ(base::bit_cast<uint64_t>(min),                         \
+             base::bit_cast<uint64_t>(results.min_abc_));           \
+    CHECK_EQ(base::bit_cast<uint64_t>(min),                         \
+             base::bit_cast<uint64_t>(results.min_aab_));           \
+    CHECK_EQ(base::bit_cast<uint64_t>(min),                         \
+             base::bit_cast<uint64_t>(results.min_aba_));           \
+    CHECK_EQ(base::bit_cast<uint64_t>(max),                         \
+             base::bit_cast<uint64_t>(results.max_abc_));           \
+    CHECK_EQ(base::bit_cast<uint64_t>(max),                         \
+             base::bit_cast<uint64_t>(results.max_aab_));           \
+    CHECK_EQ(base::bit_cast<uint64_t>(max),                         \
+             base::bit_cast<uint64_t>(results.max_aba_));           \
+    /* Use a base::bit_cast to correctly identify -0.0 and NaNs. */ \
   } while (0)
 
   double nan_a = qnan_d;
