diff --git a/test/unittests/assembler/disasm-x64-unittest.cc b/test/unittests/assembler/disasm-x64-unittest.cc
index 89d115f9f6b..9027e575086 100644
--- a/test/unittests/assembler/disasm-x64-unittest.cc
+++ b/test/unittests/assembler/disasm-x64-unittest.cc
@@ -305,8 +305,8 @@ TEST_F(DisasmX64Test, DisasmX64) {
 #ifdef OBJECT_PRINT
   StdoutStream os;
   code->Print(os);
-  Address begin = code->raw_instruction_start();
-  Address end = code->raw_instruction_end();
+  Address begin = code->InstructionStart();
+  Address end = code->InstructionStart();
   disasm::Disassembler::Disassemble(stdout, reinterpret_cast<byte*>(begin),
                                     reinterpret_cast<byte*>(end));
 #endif
