diff --git a/src/wasm/baseline/ia32/liftoff-assembler-ia32.h b/src/wasm/baseline/ia32/liftoff-assembler-ia32.h
index 243fa3fd79b..5a815213dfa 100644
--- a/src/wasm/baseline/ia32/liftoff-assembler-ia32.h
+++ b/src/wasm/baseline/ia32/liftoff-assembler-ia32.h
@@ -4668,35 +4668,33 @@ void LiftoffAssembler::DropStackSlotsAndRet(uint32_t num_stack_slots) {
   ret(static_cast<int>(num_stack_slots * kSystemPointerSize));
 }
 
-void LiftoffAssembler::CallC(const ValueKindSig* sig, const VarState* args,
-                             const LiftoffRegister* rets,
+void LiftoffAssembler::CallC(const std::initializer_list<VarState> args,
+                             const LiftoffRegister* rets, ValueKind return_kind,
                              ValueKind out_argument_kind, int stack_bytes,
                              ExternalReference ext_ref) {
   AllocateStackSpace(stack_bytes);
 
   int arg_offset = 0;
-  const VarState* current_arg = args;
-  for (ValueKind param_kind : sig->parameters()) {
-    if (current_arg->is_reg()) {
-      liftoff::Store(this, esp, arg_offset, current_arg->reg(), param_kind);
-    } else if (current_arg->is_const()) {
-      DCHECK_EQ(kI32, param_kind);
-      mov(Operand(esp, arg_offset), Immediate(current_arg->i32_const()));
-    } else if (value_kind_size(current_arg->kind()) == 4) {
+  for (const VarState& arg : args) {
+    if (arg.is_reg()) {
+      liftoff::Store(this, esp, arg_offset, arg.reg(), arg.kind());
+    } else if (arg.is_const()) {
+      DCHECK_EQ(kI32, arg.kind());
+      mov(Operand(esp, arg_offset), Immediate(arg.i32_const()));
+    } else if (value_kind_size(arg.kind()) == 4) {
       // We do not have a scratch register, so move via the stack. Note that
       // {push} decrements {esp} by 4 and {pop} increments it again, but the
       // destionation operand uses the {esp} value after increasing.
-      push(liftoff::GetStackSlot(current_arg->offset()));
+      push(liftoff::GetStackSlot(arg.offset()));
       pop(Operand(esp, arg_offset));
     } else {
-      DCHECK_EQ(8, value_kind_size(current_arg->kind()));
-      push(liftoff::GetStackSlot(current_arg->offset()));
+      DCHECK_EQ(8, value_kind_size(arg.kind()));
+      push(liftoff::GetStackSlot(arg.offset()));
       pop(Operand(esp, arg_offset + 4));
-      push(liftoff::GetStackSlot(current_arg->offset() + 4));
+      push(liftoff::GetStackSlot(arg.offset() + 4));
       pop(Operand(esp, arg_offset));
     }
-    ++current_arg;
-    arg_offset += value_kind_size(param_kind);
+    arg_offset += value_kind_size(arg.kind());
   }
   DCHECK_LE(arg_offset, stack_bytes);
 
@@ -4715,11 +4713,10 @@ void LiftoffAssembler::CallC(const ValueKindSig* sig, const VarState* args,
 
   // Move return value to the right register.
   const LiftoffRegister* next_result_reg = rets;
-  if (sig->return_count() > 0) {
-    DCHECK_EQ(1, sig->return_count());
+  if (return_kind != kVoid) {
     constexpr Register kReturnReg = eax;
     if (kReturnReg != next_result_reg->gp()) {
-      Move(*next_result_reg, LiftoffRegister(kReturnReg), sig->GetReturn(0));
+      Move(*next_result_reg, LiftoffRegister(kReturnReg), return_kind);
     }
     ++next_result_reg;
   }
