diff --git a/src/codegen/arm/macro-assembler-arm.cc b/src/codegen/arm/macro-assembler-arm.cc
index 55958db53bf..9be1d37e030 100644
--- a/src/codegen/arm/macro-assembler-arm.cc
+++ b/src/codegen/arm/macro-assembler-arm.cc
@@ -344,15 +344,6 @@ void MacroAssembler::LoadCodeEntry(Register destination, Register code_object) {
   ldr(destination, FieldMemOperand(code_object, Code::kCodeEntryPointOffset));
 }
 
-void MacroAssembler::LoadCodeInstructionStreamNonBuiltin(Register destination,
-                                                         Register code_object) {
-  ASM_CODE_COMMENT(this);
-  // Compute the InstructionStream object pointer from the code entry point.
-  ldr(destination, FieldMemOperand(code_object, Code::kCodeEntryPointOffset));
-  sub(destination, destination,
-      Operand(InstructionStream::kHeaderSize - kHeapObjectTag));
-}
-
 void MacroAssembler::CallCodeObject(Register code_object) {
   ASM_CODE_COMMENT(this);
   LoadCodeEntry(code_object, code_object);
@@ -399,7 +390,7 @@ void MacroAssembler::Drop(Register count, Condition cond) {
 void MacroAssembler::TestCodeIsMarkedForDeoptimization(Register code,
                                                        Register scratch) {
   ldr(scratch, FieldMemOperand(code, Code::kKindSpecificFlagsOffset));
-  tst(scratch, Operand(1 << InstructionStream::kMarkedForDeoptimizationBit));
+  tst(scratch, Operand(1 << Code::kMarkedForDeoptimizationBit));
 }
 
 Operand MacroAssembler::ClearedValue() const {
