diff --git a/src/codegen/x64/macro-assembler-x64.h b/src/codegen/x64/macro-assembler-x64.h
index 98627f27f0b..37e6bfdc4b0 100644
--- a/src/codegen/x64/macro-assembler-x64.h
+++ b/src/codegen/x64/macro-assembler-x64.h
@@ -925,8 +925,16 @@ class V8_EXPORT_PRIVATE MacroAssembler : public TurboAssembler {
 
   // ---------------------------------------------------------------------------
   // StatsCounter support
-  void IncrementCounter(StatsCounter* counter, int value);
-  void DecrementCounter(StatsCounter* counter, int value);
+  void IncrementCounter(StatsCounter* counter, int value) {
+    if (!FLAG_native_code_counters) return;
+    EmitIncrementCounter(counter, value);
+  }
+  void EmitIncrementCounter(StatsCounter* counter, int value);
+  void DecrementCounter(StatsCounter* counter, int value) {
+    if (!FLAG_native_code_counters) return;
+    EmitDecrementCounter(counter, value);
+  }
+  void EmitDecrementCounter(StatsCounter* counter, int value);
 
   // ---------------------------------------------------------------------------
   // Stack limit utilities
-- 
2.35.1

