diff --git a/src/maglev/arm/maglev-ir-arm.cc b/src/maglev/arm/maglev-ir-arm.cc
index 3ce9d7f16ed..419ce5bef97 100644
--- a/src/maglev/arm/maglev-ir-arm.cc
+++ b/src/maglev/arm/maglev-ir-arm.cc
@@ -279,7 +279,7 @@ void Int32DivideWithOverflow::GenerateCode(MaglevAssembler* masm,
         __ JumpIf(ne, *done);
         __ cmp(right, Operand(-1));
         __ JumpIf(ne, *done);
-        __ JumpToDeopt(deopt);
+        __ Jump(deopt);
       },
       done, left, right, this);
   __ bind(*done);
