diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 267fc24f03c..458f1774db5 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -337,6 +337,12 @@ inline void MaglevAssembler::LoadUnsignedField(Register result, Operand operand,
   }
 }
 
+inline void MaglevAssembler::StoreTaggedFieldNoWriteBarrier(Register object,
+                                                            int offset,
+                                                            Register value) {
+  MacroAssembler::StoreTaggedField(FieldOperand(object, offset), value);
+}
+
 inline void MaglevAssembler::StoreTaggedSignedField(Register object, int offset,
                                                     Register value) {
   AssertSmi(value);
@@ -348,6 +354,11 @@ inline void MaglevAssembler::StoreTaggedSignedField(Register object, int offset,
   MacroAssembler::StoreTaggedSignedField(FieldOperand(object, offset), value);
 }
 
+inline void MaglevAssembler::StoreInt32Field(Register object, int offset,
+                                             int32_t value) {
+  movl(FieldOperand(object, offset), Immediate(value));
+}
+
 inline void MaglevAssembler::StoreField(Operand operand, Register value,
                                         int size) {
   DCHECK(size == 1 || size == 2 || size == 4);
