diff --git a/src/codegen/arm/assembler-arm.cc b/src/codegen/arm/assembler-arm.cc
index ea2f8537b50..a81ef98f2af 100644
--- a/src/codegen/arm/assembler-arm.cc
+++ b/src/codegen/arm/assembler-arm.cc
@@ -5354,9 +5354,9 @@ void Assembler::CheckConstPool(bool force_emit, bool require_jump) {
   while (buffer_space() <= needed_space) GrowBuffer();
 
   {
+    ASM_CODE_COMMENT_STRING(this, "Constant Pool");
     // Block recursive calls to CheckConstPool.
     BlockConstPoolScope block_const_pool(this);
-    RecordComment("[ Constant Pool");
     RecordConstPool(size);
 
     Label size_check;
@@ -5421,8 +5421,6 @@ void Assembler::CheckConstPool(bool force_emit, bool require_jump) {
 
     first_const_pool_32_use_ = -1;
 
-    RecordComment("]");
-
     DCHECK_EQ(size, SizeOfCodeGeneratedSince(&size_check));
 
     if (after_pool.is_linked()) {
