diff --git a/src/baseline/riscv64/baseline-assembler-riscv64-inl.h b/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
index d342b30fb30..301503f5238 100644
--- a/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
+++ b/src/baseline/riscv64/baseline-assembler-riscv64-inl.h
@@ -120,7 +120,7 @@ void BaselineAssembler::CallBuiltin(Builtins::Name builtin) {
     Register temp = t6;
     __ LoadEntryFromBuiltinIndex(builtin, temp);
     __ Call(temp);
-    if (FLAG_code_comments) __ RecordComment("]");
+    __ RecordComment("]");
   }
 }
 
@@ -135,7 +135,7 @@ void BaselineAssembler::TailCallBuiltin(Builtins::Name builtin) {
     Register temp = t6;
     __ LoadEntryFromBuiltinIndex(builtin, temp);
     __ Jump(temp);
-    if (FLAG_code_comments) __ RecordComment("]");
+    __ RecordComment("]");
   }
 }
 
