diff --git a/src/codegen/x64/register-x64.h b/src/codegen/x64/register-x64.h
index 7dc9b8bc4d0..c79b33ff2bb 100644
--- a/src/codegen/x64/register-x64.h
+++ b/src/codegen/x64/register-x64.h
@@ -267,6 +267,7 @@ DEFINE_REGISTER_NAMES(XMMRegister, DOUBLE_REGISTERS)
 DEFINE_REGISTER_NAMES(YMMRegister, YMM_REGISTERS)
 
 // Give alias names to registers for calling conventions.
+constexpr Register kStackPointerRegister = rsp;
 constexpr Register kReturnRegister0 = rax;
 constexpr Register kReturnRegister1 = rdx;
 constexpr Register kReturnRegister2 = r8;
