diff --git a/src/codegen/arm/register-arm.h b/src/codegen/arm/register-arm.h
index 15bc1e88f05..8e72d19b6e4 100644
--- a/src/codegen/arm/register-arm.h
+++ b/src/codegen/arm/register-arm.h
@@ -299,6 +299,7 @@ DEFINE_REGISTER_NAMES(QwNeonRegister, SIMD128_REGISTERS)
 DEFINE_REGISTER_NAMES(CRegister, C_REGISTERS)
 
 // Give alias names to registers for calling conventions.
+constexpr Register kStackPointerRegister = sp;
 constexpr Register kReturnRegister0 = r0;
 constexpr Register kReturnRegister1 = r1;
 constexpr Register kReturnRegister2 = r2;
