diff --git a/src/maglev/arm/maglev-ir-arm.cc b/src/maglev/arm/maglev-ir-arm.cc
index 5c0d9250f41..9ae42732b75 100644
--- a/src/maglev/arm/maglev-ir-arm.cc
+++ b/src/maglev/arm/maglev-ir-arm.cc
@@ -848,6 +848,7 @@ void FunctionEntryStackCheck::SetValueLocationConstraints() {
 }
 void FunctionEntryStackCheck::GenerateCode(MaglevAssembler* masm,
                                            const ProcessingState& state) {
+  if (!masm->code_gen_state()->needs_stack_check()) return;
   // Stack check. This folds the checks for both the interrupt stack limit
   // check and the real stack limit into one by just checking for the
   // interrupt limit. The interrupt limit is either equal to the real
