diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 081469dc094..49e86eb5623 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -360,6 +360,14 @@ inline void MaglevAssembler::DeoptIfBufferDetached(Register array,
   }
 }
 
+inline void MaglevAssembler::LoadByte(Register dst, MemOperand src) {
+  movb(dst, src);
+}
+
+inline void MaglevAssembler::CompareInt32(Register reg, int32_t imm) {
+  cmpl(reg, Immediate(imm));
+}
+
 inline void MaglevAssembler::CompareInt32(Register src1, Register src2) {
   cmpl(src1, src2);
 }
