diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 646fdc396cc..8a57462ff95 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -472,6 +472,14 @@ inline void MaglevAssembler::DeoptIfBufferDetached(Register array,
   }
 }
 
+inline void MaglevAssembler::LoadByte(Register dst, MemOperand src) {
+  ldrb(dst.W(), src);
+}
+
+inline void MaglevAssembler::CompareInt32(Register reg, int32_t imm) {
+  Cmp(reg.W(), Immediate(imm));
+}
+
 inline void MaglevAssembler::CompareInt32(Register src1, Register src2) {
   Cmp(src1.W(), src2.W());
 }
