diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index fed26c752f7..250f0ba5adb 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -246,11 +246,12 @@ void MacroAssembler::LoadTaggedField(Register destination,
 
 void MacroAssembler::LoadTaggedField(TaggedRegister destination,
                                      Operand field_operand) {
-  if (COMPRESS_POINTERS_BOOL) {
-    movl(destination.reg(), field_operand);
-  } else {
-    mov_tagged(destination.reg(), field_operand);
-  }
+  LoadTaggedFieldWithoutDecompressing(destination.reg(), field_operand);
+}
+
+void MacroAssembler::LoadTaggedFieldWithoutDecompressing(
+    Register destination, Operand field_operand) {
+  mov_tagged(destination, field_operand);
 }
 
 #ifdef V8_MAP_PACKING
