diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index d9b25aa6f9a..e9be97bb7cf 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -589,7 +589,6 @@ inline void MaglevAssembler::CompareFloat64(DoubleRegister src1,
 }
 
 inline void MaglevAssembler::CallSelf() {
-  DCHECK(allow_call());
   DCHECK(code_gen_state()->entry_label()->is_bound());
   Call(code_gen_state()->entry_label());
 }
