diff --git a/test/cctest/test-assembler-arm64.cc b/test/cctest/test-assembler-arm64.cc
index 4d9ac4f71a6..0c416025149 100644
--- a/test/cctest/test-assembler-arm64.cc
+++ b/test/cctest/test-assembler-arm64.cc
@@ -14741,7 +14741,7 @@ TEST(near_call_no_relocation) {
   {
     Assembler::BlockConstPoolScope scope(&masm);
     int offset = (function.pos() - __ pc_offset()) / kInstrSize;
-    __ near_call(offset, RelocInfo::NONE);
+    __ near_call(offset, RelocInfo::NO_INFO);
   }
   END();
 
