diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index d917b97c5e9..b7011482541 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -70,6 +70,11 @@ class MaglevAssembler::ScratchRegisterScope {
   }
   ~ScratchRegisterScope() { masm_->scratch_register_scope_ = prev_scope_; }
 
+  void ResetToDefault() {
+    available_ = {};
+    available_double_ = {};
+  }
+
   Register GetDefaultScratchRegister() { return kScratchRegister; }
   DoubleRegister GetDefaultScratchDoubleRegister() { return kScratchDoubleReg; }
 
-- 
2.35.1

