diff --git a/src/wasm/baseline/arm64/liftoff-assembler-arm64.h b/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
index b8cbc6abf2..928126addc 100644
--- a/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
+++ b/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
@@ -485,7 +485,7 @@ void LiftoffAssembler::StoreTaggedPointer(Register dst_addr,
       dst_addr,
       dst_op.IsRegisterOffset() ? Operand(dst_op.regoffset().X())
                                 : Operand(dst_op.offset()),
-      EMIT_REMEMBERED_SET, kSaveFPRegs, wasm::WasmCode::kRecordWrite);
+      RememberedSetAction::kEmit, kSaveFPRegs, wasm::WasmCode::kRecordWrite);
   bind(&exit);
 }
 
