diff --git a/src/codegen/s390/interface-descriptors-s390-inl.h b/src/codegen/s390/interface-descriptors-s390-inl.h
index b09db00dcd..d23c4fde34 100644
--- a/src/codegen/s390/interface-descriptors-s390-inl.h
+++ b/src/codegen/s390/interface-descriptors-s390-inl.h
@@ -24,14 +24,14 @@ template <typename DerivedDescriptor>
 void StaticCallInterfaceDescriptor<DerivedDescriptor>::
     VerifyArgumentRegisterCount(CallInterfaceDescriptorData* data, int argc) {
   RegList allocatable_regs = data->allocatable_registers();
-  if (argc >= 1) DCHECK(allocatable_regs | r2.bit());
-  if (argc >= 2) DCHECK(allocatable_regs | r3.bit());
-  if (argc >= 3) DCHECK(allocatable_regs | r4.bit());
-  if (argc >= 4) DCHECK(allocatable_regs | r5.bit());
-  if (argc >= 5) DCHECK(allocatable_regs | r6.bit());
-  if (argc >= 6) DCHECK(allocatable_regs | r7.bit());
-  if (argc >= 7) DCHECK(allocatable_regs | r8.bit());
-  if (argc >= 8) DCHECK(allocatable_regs | r9.bit());
+  if (argc >= 1) DCHECK(allocatable_regs.has(r2));
+  if (argc >= 2) DCHECK(allocatable_regs.has(r3));
+  if (argc >= 3) DCHECK(allocatable_regs.has(r4));
+  if (argc >= 4) DCHECK(allocatable_regs.has(r5));
+  if (argc >= 5) DCHECK(allocatable_regs.has(r6));
+  if (argc >= 6) DCHECK(allocatable_regs.has(r7));
+  if (argc >= 7) DCHECK(allocatable_regs.has(r8));
+  if (argc >= 8) DCHECK(allocatable_regs.has(r9));
   // Additional arguments are passed on the stack.
 }
 #endif  // DEBUG
