diff --git a/src/codegen/arm/interface-descriptors-arm-inl.h b/src/codegen/arm/interface-descriptors-arm-inl.h
index b1157a4eeb..83d82fe3ce 100644
--- a/src/codegen/arm/interface-descriptors-arm-inl.h
+++ b/src/codegen/arm/interface-descriptors-arm-inl.h
@@ -47,6 +47,12 @@ constexpr auto DynamicCheckMapsDescriptor::registers() {
   return RegisterArray(r0, r1, r2, r3, cp);
 }
 
+// static
+constexpr auto DynamicCheckMapsWithFeedbackVectorDescriptor::registers() {
+  STATIC_ASSERT(kReturnRegister0 == r0);
+  return RegisterArray(r0, r1, r2, r3, cp);
+}
+
 // static
 constexpr Register LoadDescriptor::ReceiverRegister() { return r1; }
 // static
