diff --git a/src/codegen/x64/interface-descriptors-x64-inl.h b/src/codegen/x64/interface-descriptors-x64-inl.h
index 63e6489b2f..50ba12b836 100644
--- a/src/codegen/x64/interface-descriptors-x64-inl.h
+++ b/src/codegen/x64/interface-descriptors-x64-inl.h
@@ -65,6 +65,18 @@ constexpr auto DynamicCheckMapsDescriptor::registers() {
 #endif  // V8_TARGET_OS_WIN
 }
 
+// static
+constexpr auto DynamicCheckMapsWithFeedbackVectorDescriptor::registers() {
+#if V8_TARGET_OS_WIN
+  return RegisterArray(kReturnRegister0, arg_reg_1, arg_reg_2, arg_reg_3,
+                       kRuntimeCallFunctionRegister, kContextRegister);
+#else
+  STATIC_ASSERT(kContextRegister == arg_reg_2);
+  return RegisterArray(kReturnRegister0, arg_reg_1, arg_reg_2, arg_reg_3,
+                       kRuntimeCallFunctionRegister);
+#endif  // V8_TARGET_OS_WIN
+}
+
 // static
 constexpr Register LoadDescriptor::ReceiverRegister() { return rdx; }
 // static
