diff --git a/src/codegen/ia32/interface-descriptors-ia32-inl.h b/src/codegen/ia32/interface-descriptors-ia32-inl.h
index 5ad23d66261..b383b5df76b 100644
--- a/src/codegen/ia32/interface-descriptors-ia32-inl.h
+++ b/src/codegen/ia32/interface-descriptors-ia32-inl.h
@@ -42,6 +42,13 @@ constexpr auto DynamicCheckMapsDescriptor::registers() {
   return RegisterArray(eax, ecx, edx, edi, esi);
 }
 
+// static
+constexpr auto DynamicCheckMapsWithFeedbackVectorDescriptor::registers() {
+  STATIC_ASSERT(esi == kContextRegister);
+  STATIC_ASSERT(eax == kReturnRegister0);
+  return RegisterArray(eax, ecx, edx, edi, esi);
+}
+
 // static
 constexpr Register LoadDescriptor::ReceiverRegister() { return edx; }
 // static
