diff --git a/src/codegen/arm64/interface-descriptors-arm64-inl.h b/src/codegen/arm64/interface-descriptors-arm64-inl.h
index ec9786f5a62..e8fe4ef1d32 100644
--- a/src/codegen/arm64/interface-descriptors-arm64-inl.h
+++ b/src/codegen/arm64/interface-descriptors-arm64-inl.h
@@ -47,6 +47,12 @@ constexpr auto DynamicCheckMapsDescriptor::registers() {
   return RegisterArray(x0, x1, x2, x3, cp);
 }
 
+// static
+constexpr auto DynamicCheckMapsWithFeedbackVectorDescriptor::registers() {
+  STATIC_ASSERT(kReturnRegister0 == x0);
+  return RegisterArray(x0, x1, x2, x3, cp);
+}
+
 // static
 constexpr Register LoadDescriptor::ReceiverRegister() { return x1; }
 // static
