diff --git a/src/codegen/ppc/interface-descriptors-ppc-inl.h b/src/codegen/ppc/interface-descriptors-ppc-inl.h
index 9e00508aeb..adc36e2407 100644
--- a/src/codegen/ppc/interface-descriptors-ppc-inl.h
+++ b/src/codegen/ppc/interface-descriptors-ppc-inl.h
@@ -47,6 +47,12 @@ constexpr auto DynamicCheckMapsDescriptor::registers() {
   return RegisterArray(r3, r4, r5, r6, cp);
 }
 
+// static
+constexpr auto DynamicCheckMapsWithFeedbackVectorDescriptor::registers() {
+  STATIC_ASSERT(kReturnRegister0 == r3);
+  return RegisterArray(r3, r4, r5, r6, cp);
+}
+
 // static
 constexpr Register LoadDescriptor::ReceiverRegister() { return r4; }
 // static
