diff --git a/src/compiler/backend/ppc/instruction-codes-ppc.h b/src/compiler/backend/ppc/instruction-codes-ppc.h
index df0fadc492..8635852438 100644
--- a/src/compiler/backend/ppc/instruction-codes-ppc.h
+++ b/src/compiler/backend/ppc/instruction-codes-ppc.h
@@ -368,10 +368,10 @@ namespace compiler {
   V(PPC_I8x16Shuffle)                \
   V(PPC_I8x16Swizzle)                \
   V(PPC_I8x16BitMask)                \
-  V(PPC_V64x2AllTrue)                \
-  V(PPC_V32x4AllTrue)                \
-  V(PPC_V16x8AllTrue)                \
-  V(PPC_V8x16AllTrue)                \
+  V(PPC_I64x2AllTrue)                \
+  V(PPC_I32x4AllTrue)                \
+  V(PPC_I16x8AllTrue)                \
+  V(PPC_I8x16AllTrue)                \
   V(PPC_V128AnyTrue)                 \
   V(PPC_S128And)                     \
   V(PPC_S128Or)                      \
