diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index d67ebd80e99..77a9a4296ee 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -3061,7 +3061,6 @@ void MacroAssembler::AssertCode(Register object) {
 }
 
 void MacroAssembler::AssertSmiOrHeapObjectInCompressionCage(Register object) {
-  DCHECK(PointerCompressionIsEnabled());
   if (!v8_flags.debug_code) return;
   ASM_CODE_COMMENT(this);
   Label is_smi;
