diff --git a/src/maglev/arm64/maglev-ir-arm64.cc b/src/maglev/arm64/maglev-ir-arm64.cc
index e6f52675f41..9099027dad2 100644
--- a/src/maglev/arm64/maglev-ir-arm64.cc
+++ b/src/maglev/arm64/maglev-ir-arm64.cc
@@ -2493,8 +2493,9 @@ void FunctionLength::GenerateCode(MaglevAssembler* masm,
   Register shared = temps.AcquireX();
   __ LoadTaggedPointerField(
       shared, FieldMemOperand(object, JSFunction::kSharedFunctionInfoOffset));
-  __ Ldr(ToRegister(result()).W(),
-         FieldMemOperand(shared, SharedFunctionInfo::kLengthOffset));
+  __ LoadSignedField(ToRegister(result()).W(),
+                     FieldMemOperand(shared, SharedFunctionInfo::kLengthOffset),
+                     2);
 }
 
 void TestUndetectable::SetValueLocationConstraints() {
