diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 006b20725f9..1376e503fd3 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -1342,8 +1342,9 @@ void FunctionLength::GenerateCode(MaglevAssembler* masm,
   __ LoadTaggedPointerField(
       kScratchRegister,
       FieldOperand(object, JSFunction::kSharedFunctionInfoOffset));
-  __ movl(ToRegister(result()),
-          FieldOperand(kScratchRegister, SharedFunctionInfo::kLengthOffset));
+  __ LoadSignedField(
+      ToRegister(result()),
+      FieldMemOperand(kScratchRegister, SharedFunctionInfo::kLengthOffset), 2);
 }
 
 void Int32AddWithOverflow::SetValueLocationConstraints() {
-- 
2.35.1

