diff --git a/test/cctest/test-helper-riscv32.cc b/test/cctest/test-helper-riscv32.cc
index 0e9738a7b7..79468c7373 100644
--- a/test/cctest/test-helper-riscv32.cc
+++ b/test/cctest/test-helper-riscv32.cc
@@ -24,12 +24,11 @@ int32_t GenAndRunTest(Func test_generator) {
   assm.GetCode(isolate, &desc);
   Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
-  auto f = GeneratedCode<int32_t()>::FromCode(*code);
+  auto f = GeneratedCode<int32_t()>::FromCode(isolate, *code);
   return f.Call();
 }
 
-Handle<Code> AssembleCodeImpl(Func assemble) {
-  Isolate* isolate = CcTest::i_isolate();
+Handle<Code> AssembleCodeImpl(Isolate* isolate, Func assemble) {
   MacroAssembler assm(isolate, CodeObjectRequired::kYes);
 
   assemble(assm);
