diff --git a/src/builtins/ia32/builtins-ia32.cc b/src/builtins/ia32/builtins-ia32.cc
index e3ad2c390fc..be736200fa8 100644
--- a/src/builtins/ia32/builtins-ia32.cc
+++ b/src/builtins/ia32/builtins-ia32.cc
@@ -3150,8 +3150,7 @@ void Builtins::Generate_WasmOnStackReplace(MacroAssembler* masm) {
 #endif  // V8_ENABLE_WEBASSEMBLY
 
 void Builtins::Generate_CEntry(MacroAssembler* masm, int result_size,
-                               ArgvMode argv_mode, bool builtin_exit_frame,
-                               bool switch_to_central_stack) {
+                               ArgvMode argv_mode, bool builtin_exit_frame) {
   CHECK(result_size == 1 || result_size == 2);
 
   using ER = ExternalReference;
