diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index b01c972f6c5..f7b1b6cf572 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -3002,12 +3002,6 @@ void MacroAssembler::InvokePrologue(Register expected_parameter_count,
       return;
     }
     Label regular_invoke;
-    // If the expected parameter count is equal to the adaptor sentinel, no need
-    // to push undefined value as arguments.
-    if (kDontAdaptArgumentsSentinel != 0) {
-      cmpl(expected_parameter_count, Immediate(kDontAdaptArgumentsSentinel));
-      j(equal, &regular_invoke, Label::kFar);
-    }
 
     // If overapplication or if the actual argument count is equal to the
     // formal parameter count, no need to push extra undefined values.
