diff --git a/src/baseline/ppc/baseline-assembler-ppc-inl.h b/src/baseline/ppc/baseline-assembler-ppc-inl.h
index fbf89af4de8..4196551aa13 100644
--- a/src/baseline/ppc/baseline-assembler-ppc-inl.h
+++ b/src/baseline/ppc/baseline-assembler-ppc-inl.h
@@ -94,7 +94,7 @@ static void JumpIfHelper(MacroAssembler* assm, Condition cc, Register lhs,
       __ CmpU32(lhs, rhs);
     }
   }
-  __ b(cc, target);
+  __ b(check_condition(cc), target);
 }
 #undef __
 
@@ -160,7 +160,7 @@ void BaselineAssembler::TestAndBranch(Register value, int mask, Condition cc,
                                       Label* target, Label::Distance) {
   ASM_CODE_COMMENT(masm_);
   __ AndU64(r0, value, Operand(mask), ip, SetRC);
-  __ b(cc, target, cr0);
+  __ b(check_condition(cc), target, cr0);
 }
 
 void BaselineAssembler::JumpIf(Condition cc, Register lhs, const Operand& rhs,
@@ -171,7 +171,7 @@ void BaselineAssembler::JumpIf(Condition cc, Register lhs, const Operand& rhs,
   } else {
     __ CmpU64(lhs, rhs, r0);
   }
-  __ b(cc, target);
+  __ b(check_condition(cc), target);
 }
 
 void BaselineAssembler::JumpIfObjectType(Condition cc, Register object,
