diff --git a/src/execution/riscv/simulator-riscv.cc b/src/execution/riscv/simulator-riscv.cc
index 0eb9cbaf98e..4e9123a6ae7 100644
--- a/src/execution/riscv/simulator-riscv.cc
+++ b/src/execution/riscv/simulator-riscv.cc
@@ -1946,7 +1946,7 @@ void RiscvDebugger::Debug() {
             if (IsSmi(obj)) {
               PrintF("smi %d", Smi::ToInt(obj));
             } else {
-              obj.ShortPrint();
+              ShortPrint(obj);
             }
             PrintF(")");
           }
@@ -7699,7 +7699,7 @@ void Simulator::InstructionDecode(Instruction* instr) {
     //   if (obj.IsSmi()) {
     //     PrintF("smi %d", Smi::ToInt(obj));
     //   } else {
-    //     obj.ShortPrint();
+    //     ShortPrint(obj);
     //   }
     //   PrintF(")");
     // }
