diff --git a/src/codegen/arm/interface-descriptors-arm-inl.h b/src/codegen/arm/interface-descriptors-arm-inl.h
index d2ce1204f15..296f72d1578 100644
--- a/src/codegen/arm/interface-descriptors-arm-inl.h
+++ b/src/codegen/arm/interface-descriptors-arm-inl.h
@@ -243,11 +243,6 @@ constexpr auto ResumeGeneratorDescriptor::registers() {
                        r1);  // the JSGeneratorObject to resume
 }
 
-// static
-constexpr auto FrameDropperTrampolineDescriptor::registers() {
-  return RegisterArray(r1);  // loaded new FP
-}
-
 // static
 constexpr auto RunMicrotasksEntryDescriptor::registers() {
   return RegisterArray(r0, r1);
