diff --git a/src/codegen/mips/interface-descriptors-mips-inl.h b/src/codegen/mips/interface-descriptors-mips-inl.h
index 859f9b39268..edea1b3844a 100644
--- a/src/codegen/mips/interface-descriptors-mips-inl.h
+++ b/src/codegen/mips/interface-descriptors-mips-inl.h
@@ -245,12 +245,6 @@ constexpr auto ResumeGeneratorDescriptor::registers() {
   return RegisterArray(v0, a1);
 }
 
-// static
-constexpr auto FrameDropperTrampolineDescriptor::registers() {
-  // a1 : loaded new FP
-  return RegisterArray(a1);
-}
-
 // static
 constexpr auto RunMicrotasksEntryDescriptor::registers() {
   return RegisterArray(a0, a1);
