diff --git a/src/codegen/x64/interface-descriptors-x64-inl.h b/src/codegen/x64/interface-descriptors-x64-inl.h
index e5cc5df4f20..a24330a4c77 100644
--- a/src/codegen/x64/interface-descriptors-x64-inl.h
+++ b/src/codegen/x64/interface-descriptors-x64-inl.h
@@ -245,11 +245,6 @@ constexpr auto ResumeGeneratorDescriptor::registers() {
       rdx);  // the JSGeneratorObject / JSAsyncGeneratorObject to resume
 }
 
-// static
-constexpr auto FrameDropperTrampolineDescriptor::registers() {
-  return RegisterArray(rbx);  // loaded new FP
-}
-
 // static
 constexpr auto RunMicrotasksEntryDescriptor::registers() {
   return RegisterArray(arg_reg_1, arg_reg_2);
