diff --git a/src/codegen/s390/interface-descriptors-s390-inl.h b/src/codegen/s390/interface-descriptors-s390-inl.h
index 97bdbc51b4c..d672c4354dd 100644
--- a/src/codegen/s390/interface-descriptors-s390-inl.h
+++ b/src/codegen/s390/interface-descriptors-s390-inl.h
@@ -243,11 +243,6 @@ constexpr auto ResumeGeneratorDescriptor::registers() {
                        r3);  // the JSGeneratorObject to resume
 }
 
-// static
-constexpr auto FrameDropperTrampolineDescriptor::registers() {
-  return RegisterArray(r3);  // loaded new FP
-}
-
 // static
 constexpr auto RunMicrotasksEntryDescriptor::registers() {
   return RegisterArray(r2, r3);
