diff --git a/src/codegen/arm64/interface-descriptors-arm64-inl.h b/src/codegen/arm64/interface-descriptors-arm64-inl.h
index 0ac1d2adca..90123dbdcb 100644
--- a/src/codegen/arm64/interface-descriptors-arm64-inl.h
+++ b/src/codegen/arm64/interface-descriptors-arm64-inl.h
@@ -252,11 +252,6 @@ constexpr auto ResumeGeneratorDescriptor::registers() {
                        x1);  // the JSGeneratorObject to resume
 }
 
-// static
-constexpr auto FrameDropperTrampolineDescriptor::registers() {
-  return RegisterArray(x1);  // loaded new FP
-}
-
 // static
 constexpr auto RunMicrotasksEntryDescriptor::registers() {
   return RegisterArray(x0, x1);
