diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 8a57462ff95..6902390c5c4 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -473,7 +473,7 @@ inline void MaglevAssembler::DeoptIfBufferDetached(Register array,
 }
 
 inline void MaglevAssembler::LoadByte(Register dst, MemOperand src) {
-  ldrb(dst.W(), src);
+  Ldrb(dst, src);
 }
 
 inline void MaglevAssembler::CompareInt32(Register reg, int32_t imm) {
