diff --git a/src/codegen/s390/assembler-s390.cc b/src/codegen/s390/assembler-s390.cc
index 635fa41675..c2608b0492 100644
--- a/src/codegen/s390/assembler-s390.cc
+++ b/src/codegen/s390/assembler-s390.cc
@@ -689,7 +689,7 @@ void Assembler::EnsureSpaceFor(int space_needed) {
   }
 }
 
-void Assembler::call(Handle<CodeT> target, RelocInfo::Mode rmode) {
+void Assembler::call(Handle<CodeDataContainer> target, RelocInfo::Mode rmode) {
   DCHECK(RelocInfo::IsCodeTarget(rmode));
   EnsureSpace ensure_space(this);
 
@@ -698,7 +698,7 @@ void Assembler::call(Handle<CodeT> target, RelocInfo::Mode rmode) {
   brasl(r14, Operand(target_index));
 }
 
-void Assembler::jump(Handle<CodeT> target, RelocInfo::Mode rmode,
+void Assembler::jump(Handle<CodeDataContainer> target, RelocInfo::Mode rmode,
                      Condition cond) {
   DCHECK(RelocInfo::IsRelativeCodeTarget(rmode));
   EnsureSpace ensure_space(this);
