diff --git a/src/codegen/ia32/assembler-ia32.cc b/src/codegen/ia32/assembler-ia32.cc
index a2960bb7d0..ddbe5b82c0 100644
--- a/src/codegen/ia32/assembler-ia32.cc
+++ b/src/codegen/ia32/assembler-ia32.cc
@@ -1639,7 +1639,7 @@ void Assembler::call(Operand adr) {
   emit_operand(edx, adr);
 }
 
-void Assembler::call(Handle<CodeT> code, RelocInfo::Mode rmode) {
+void Assembler::call(Handle<CodeDataContainer> code, RelocInfo::Mode rmode) {
   EnsureSpace ensure_space(this);
   DCHECK(RelocInfo::IsCodeTarget(rmode));
   EMIT(0xE8);
@@ -1709,7 +1709,7 @@ void Assembler::jmp(Operand adr) {
   emit_operand(esp, adr);
 }
 
-void Assembler::jmp(Handle<CodeT> code, RelocInfo::Mode rmode) {
+void Assembler::jmp(Handle<CodeDataContainer> code, RelocInfo::Mode rmode) {
   EnsureSpace ensure_space(this);
   DCHECK(RelocInfo::IsCodeTarget(rmode));
   EMIT(0xE9);
@@ -1769,7 +1769,8 @@ void Assembler::j(Condition cc, byte* entry, RelocInfo::Mode rmode) {
   emit(entry - (pc_ + sizeof(int32_t)), rmode);
 }
 
-void Assembler::j(Condition cc, Handle<CodeT> code, RelocInfo::Mode rmode) {
+void Assembler::j(Condition cc, Handle<CodeDataContainer> code,
+                  RelocInfo::Mode rmode) {
   EnsureSpace ensure_space(this);
   // 0000 1111 1000 tttn #32-bit disp
   EMIT(0x0F);
