diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index a778d115323..cef2b23c247 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -639,6 +639,7 @@ void MaglevAssembler::Prologue(Graph* graph) {
     BailoutIfDeoptimized(rbx);
   }
 
+  CHECK_IMPLIES(graph->is_osr(), !graph->has_recursive_calls());
   if (graph->has_recursive_calls()) {
     bind(code_gen_state()->entry_label());
   }
-- 
2.35.1

