diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 69bd5d44d65..ee0bf030694 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -129,6 +129,8 @@ void MaglevAssembler::PushReverse(T... vals) {
   detail::PushAllHelper<T...>::PushReverse(this, vals...);
 }
 
+inline void MaglevAssembler::BindJumpTarget(Label* label) { bind(label); }
+
 inline void MaglevAssembler::BindBlock(BasicBlock* block) {
   bind(block->label());
 }
-- 
2.35.1

