diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 369f8336edf..1a69e352b7e 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -241,6 +241,10 @@ void MaglevAssembler::PushReverse(T... vals) {
   detail::PushAllReverse(this, vals...);
 }
 
+inline void MaglevAssembler::BindJumpTarget(Label* label) {
+  MacroAssembler::BindJumpTarget(label);
+}
+
 inline void MaglevAssembler::BindBlock(BasicBlock* block) {
   if (block->is_start_block_of_switch_case()) {
     BindJumpTarget(block->label());
