diff --git a/src/maglev/arm/maglev-assembler-arm.cc b/src/maglev/arm/maglev-assembler-arm.cc
index 432f27b914a..c9f06bcd1f5 100644
--- a/src/maglev/arm/maglev-assembler-arm.cc
+++ b/src/maglev/arm/maglev-assembler-arm.cc
@@ -559,8 +559,8 @@ void MaglevAssembler::TryChangeFloat64ToIndex(Register result,
   // Check that the result of the float64->int32->float64 is equal to
   // the input (i.e. that the conversion didn't truncate).
   VFPCompareAndSetFlags(value, converted_back);
-  JumpIf(kNotEqual, fail);
-  Jump(success);
+  JumpIf(kEqual, success);
+  Jump(fail);
 }
 
 }  // namespace maglev
