diff --git a/src/maglev/arm64/maglev-assembler-arm64.cc b/src/maglev/arm64/maglev-assembler-arm64.cc
index 416b87a4486..1de7a887d93 100644
--- a/src/maglev/arm64/maglev-assembler-arm64.cc
+++ b/src/maglev/arm64/maglev-assembler-arm64.cc
@@ -588,8 +588,8 @@ void MaglevAssembler::TryChangeFloat64ToIndex(Register result,
   // Check that the result of the float64->int32->float64 is equal to
   // the input (i.e. that the conversion didn't truncate).
   Fcmp(value, converted_back);
-  JumpIf(kNotEqual, fail);
-  Jump(success);
+  JumpIf(kEqual, success);
+  Jump(fail);
 }
 
 }  // namespace maglev
