diff --git a/src/codegen/arm/assembler-arm-inl.h b/src/codegen/arm/assembler-arm-inl.h
index 9080b3e0b3a..dbaff9e6fc3 100644
--- a/src/codegen/arm/assembler-arm-inl.h
+++ b/src/codegen/arm/assembler-arm-inl.h
@@ -38,8 +38,8 @@
 #define V8_CODEGEN_ARM_ASSEMBLER_ARM_INL_H_
 
 #include "src/codegen/arm/assembler-arm.h"
-
 #include "src/codegen/assembler.h"
+#include "src/codegen/flush-instruction-cache.h"
 #include "src/debug/debug.h"
 #include "src/objects/objects-inl.h"
 #include "src/objects/smi.h"
