diff --git a/src/deoptimizer/loong64/deoptimizer-loong64.cc b/src/deoptimizer/loong64/deoptimizer-loong64.cc
index 5692e16969..f513e3fbab 100644
--- a/src/deoptimizer/loong64/deoptimizer-loong64.cc
+++ b/src/deoptimizer/loong64/deoptimizer-loong64.cc
@@ -7,7 +7,7 @@
 namespace v8 {
 namespace internal {
 
-const int Deoptimizer::kNonLazyDeoptExitSize = 2 * kInstrSize;
+const int Deoptimizer::kEagerDeoptExitSize = 2 * kInstrSize;
 const int Deoptimizer::kLazyDeoptExitSize = 2 * kInstrSize;
 
 Float32 RegisterValues::GetFloatRegister(unsigned n) const {
