diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 2c1a79b0011..b8073b960cb 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -100,7 +100,8 @@ void MaglevAssembler::AllocateTwoByteString(RegisterSnapshot register_snapshot,
 void MaglevAssembler::LoadSingleCharacterString(Register result,
                                                 Register char_code,
                                                 Register scratch) {
-  AssertZeroExtended(char_code);
+  // Make sure char_code is zero extended.
+  movl(char_code, char_code);
   if (v8_flags.debug_code) {
     cmpq(char_code, Immediate(String::kMaxOneByteCharCode));
     Assert(below_equal, AbortReason::kUnexpectedValue);
@@ -364,8 +365,6 @@ void MaglevAssembler::TruncateDoubleToInt32(Register dst, DoubleRegister src) {
       },
       src, dst, done);
   bind(*done);
-  // Zero extend the converted value to complete the truncation.
-  movl(dst, dst);
 }
 
 void MaglevAssembler::Prologue(Graph* graph) {
