diff --git a/src/regexp/x64/regexp-macro-assembler-x64.cc b/src/regexp/x64/regexp-macro-assembler-x64.cc
index 823c3f46cb7..638b52c2b0c 100644
--- a/src/regexp/x64/regexp-macro-assembler-x64.cc
+++ b/src/regexp/x64/regexp-macro-assembler-x64.cc
@@ -1103,7 +1103,7 @@ Handle<HeapObject> RegExpMacroAssemblerX64::GetCode(Handle<String> source) {
                           .Build();
   Handle<InstructionStream> istream(code->instruction_stream(), isolate);
   PROFILE(isolate,
-          RegExpCodeCreateEvent(Handle<AbstractCode>::cast(istream), source));
+          RegExpCodeCreateEvent(Handle<AbstractCode>::cast(code), source));
   return Handle<HeapObject>::cast(istream);
 }
 
