diff --git a/src/codegen/s390/assembler-s390.cc b/src/codegen/s390/assembler-s390.cc
index 7e922c3aba..f5cf03b79f 100644
--- a/src/codegen/s390/assembler-s390.cc
+++ b/src/codegen/s390/assembler-s390.cc
@@ -759,18 +759,6 @@ void Assembler::dumy(int r1, int x2, int b2, int d2) {
 #endif
 }
 
-void Assembler::FixOnHeapReferences(bool update_embedded_objects) {
-  // TODO(v8:11872) This function should never be called if Sparkplug on heap
-  // compilation is not supported.
-  UNREACHABLE();
-}
-
-void Assembler::FixOnHeapReferencesToHandles() {
-  // TODO(v8:11872) This function should never be called if Sparkplug on heap
-  // compilation is not supported.
-  UNREACHABLE();
-}
-
 void Assembler::GrowBuffer(int needed) {
   DCHECK_EQ(buffer_start_, buffer_->start());
 
