diff --git a/src/codegen/ppc/assembler-ppc.cc b/src/codegen/ppc/assembler-ppc.cc
index 3e154e4c29..ccb144dc61 100644
--- a/src/codegen/ppc/assembler-ppc.cc
+++ b/src/codegen/ppc/assembler-ppc.cc
@@ -1930,18 +1930,6 @@ bool Assembler::IsNop(Instr instr, int type) {
   return instr == (ORI | reg * B21 | reg * B16);
 }
 
-void Assembler::FixOnHeapReferences(bool update_embedded_objects) {
-  // TODO(v8:11872) This function should never be called if Sparkplug on heap
-  // compilation is not supported.
-  UNREACHABLE();
-}
-
-void Assembler::FixOnHeapReferencesToHandles() {
-  // TODO(v8:11872) This function should never be called if Sparkplug on heap
-  // compilation is not supported.
-  UNREACHABLE();
-}
-
 void Assembler::GrowBuffer(int needed) {
   DCHECK_EQ(buffer_start_, buffer_->start());
 
