diff --git a/test/cctest/test-assembler-mips64.cc b/test/cctest/test-assembler-mips64.cc
index fecd5779b27..5cea9f8872a 100644
--- a/test/cctest/test-assembler-mips64.cc
+++ b/test/cctest/test-assembler-mips64.cc
@@ -4731,7 +4731,7 @@ uint64_t run_aluipc(int16_t offset) {
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 
   auto f = GeneratedCode<F2>::FromCode(isolate, *code);
-  PC = (uint64_t)code->code_entry_point();  // Set the program counter.
+  PC = (uint64_t)code->instruction_start();  // Set the program counter.
 
   uint64_t res = reinterpret_cast<uint64_t>(f.Call(0, 0, 0, 0, 0));
 
@@ -4784,7 +4784,7 @@ uint64_t run_auipc(int16_t offset) {
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 
   auto f = GeneratedCode<F2>::FromCode(isolate, *code);
-  PC = (uint64_t)code->code_entry_point();  // Set the program counter.
+  PC = (uint64_t)code->instruction_start();  // Set the program counter.
 
   uint64_t res = reinterpret_cast<uint64_t>(f.Call(0, 0, 0, 0, 0));
 
@@ -5780,7 +5780,7 @@ uint64_t run_addiupc(int32_t imm19) {
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 
   auto f = GeneratedCode<F2>::FromCode(isolate, *code);
-  PC = (uint64_t)code->code_entry_point();  // Set the program counter.
+  PC = (uint64_t)code->instruction_start();  // Set the program counter.
 
   uint64_t res = reinterpret_cast<uint64_t>(f.Call(0, 0, 0, 0, 0));
 
