diff --git a/src/compiler/backend/s390/instruction-codes-s390.h b/src/compiler/backend/s390/instruction-codes-s390.h
index 793ebf18862..70216904faf 100644
--- a/src/compiler/backend/s390/instruction-codes-s390.h
+++ b/src/compiler/backend/s390/instruction-codes-s390.h
@@ -419,7 +419,8 @@ namespace compiler {
   V(MR)   /* [%r0          ] */        \
   V(MRI)  /* [%r0       + K] */        \
   V(MRR)  /* [%r0 + %r1    ] */        \
-  V(MRRI) /* [%r0 + %r1 + K] */
+  V(MRRI) /* [%r0 + %r1 + K] */        \
+  V(Root) /* [%r0 + K] */
 
 }  // namespace compiler
 }  // namespace internal
