diff --git a/src/codegen/arm64/constants-arm64.h b/src/codegen/arm64/constants-arm64.h
index 1866842f1eb..da9621467e2 100644
--- a/src/codegen/arm64/constants-arm64.h
+++ b/src/codegen/arm64/constants-arm64.h
@@ -300,7 +300,7 @@ SYSTEM_REGISTER_FIELDS_LIST(DECLARE_FIELDS_OFFSETS, NOTHING)
 constexpr int ImmPCRel_mask = ImmPCRelLo_mask | ImmPCRelHi_mask;
 
 // Condition codes.
-enum Condition : uint8_t {
+enum Condition : int {
   eq = 0,   // Equal
   ne = 1,   // Not equal
   hs = 2,   // Unsigned higher or same (or carry set)
