diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index 250f0ba5adb..68434a1992d 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -2140,11 +2140,6 @@ void MacroAssembler::Jump(Handle<Code> code_object, RelocInfo::Mode rmode,
   j(cc, code_object, rmode);
 }
 
-void MacroAssembler::JumpToOffHeapInstructionStream(Address entry) {
-  Move(kOffHeapTrampolineRegister, entry, RelocInfo::OFF_HEAP_TARGET);
-  jmp(kOffHeapTrampolineRegister);
-}
-
 void MacroAssembler::Call(ExternalReference ext) {
   LoadAddress(kScratchRegister, ext);
   call(kScratchRegister);
