diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 5b863d949c4..c06cb643bfe 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -362,6 +362,12 @@ void MaglevAssembler::LoadFixedDoubleArrayElement(DoubleRegister result,
         FieldOperand(array, index, times_8, FixedDoubleArray::kHeaderSize));
 }
 
+inline void MaglevAssembler::StoreFixedDoubleArrayElement(
+    Register array, Register index, DoubleRegister value) {
+  Movsd(FieldOperand(array, index, times_8, FixedDoubleArray::kHeaderSize),
+        value);
+}
+
 inline void MaglevAssembler::LoadSignedField(Register result, Operand operand,
                                              int size) {
   if (size == 1) {
