diff --git a/src/codegen/ppc/register-ppc.h b/src/codegen/ppc/register-ppc.h
index 68adfdb1557..17625fea636 100644
--- a/src/codegen/ppc/register-ppc.h
+++ b/src/codegen/ppc/register-ppc.h
@@ -5,7 +5,7 @@
 #ifndef V8_CODEGEN_PPC_REGISTER_PPC_H_
 #define V8_CODEGEN_PPC_REGISTER_PPC_H_
 
-#include "src/codegen/register.h"
+#include "src/codegen/register-base.h"
 #include "src/codegen/reglist.h"
 
 namespace v8 {
