diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 6d51afb6277..d8a23362c6f 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -498,8 +498,8 @@ void CheckJSTypedArrayBounds::GenerateCode(MaglevAssembler* masm,
                                JSTypedArray::kRawByteLengthOffset);
   int element_size = ElementsKindSize(elements_kind_);
   if (element_size > 1) {
-    DCHECK(element_size == 2 || element_size == 4);
-    __ shlq(index, Immediate(element_size / 2));
+    DCHECK(element_size == 2 || element_size == 4 || element_size == 8);
+    __ shlq(index, Immediate(base::bits::CountTrailingZeros(element_size)));
   }
   __ cmpq(index, byte_length);
   // We use {above_equal} which does an unsigned comparison to handle negative
-- 
2.35.1

