diff --git a/test/unittests/assembler/macro-assembler-riscv-unittest.cc b/test/unittests/assembler/macro-assembler-riscv-unittest.cc
index 8e74ae692c4..028a66dede6 100644
--- a/test/unittests/assembler/macro-assembler-riscv-unittest.cc
+++ b/test/unittests/assembler/macro-assembler-riscv-unittest.cc
@@ -29,7 +29,7 @@ TEST_F(MacroAssemblerTest, TestHardAbort) {
   __ Abort(AbortReason::kNoReason);
 
   CodeDesc desc;
-  masm.GetCode(nullptr, &desc);
+  masm.GetCode(static_cast<LocalIsolate*>(nullptr), &desc);
   buffer->MakeExecutable();
   // We need an isolate here to execute in the simulator.
   auto f = GeneratedCode<void>::FromBuffer(isolate(), buffer->start());
@@ -48,7 +48,7 @@ TEST_F(MacroAssemblerTest, TestCheck) {
   __ Ret();
 
   CodeDesc desc;
-  masm.GetCode(nullptr, &desc);
+  masm.GetCode(static_cast<LocalIsolate*>(nullptr), &desc);
   buffer->MakeExecutable();
   // We need an isolate here to execute in the simulator.
   auto f = GeneratedCode<void, int>::FromBuffer(isolate(), buffer->start());
