diff --git a/src/codegen/s390/constants-s390.h b/src/codegen/s390/constants-s390.h
index 009b46e9ccd..8bade7b2abe 100644
--- a/src/codegen/s390/constants-s390.h
+++ b/src/codegen/s390/constants-s390.h
@@ -104,22 +104,45 @@ enum Condition {
   mask0xF = 15,
 
   // Unified cross-platform condition names/aliases.
+  // Do not set unsigned constants equal to their signed variants.
+  // We need to be able to differentiate between signed and unsigned enum
+  // constants in order to emit the right instructions (i.e CmpS64 vs CmpU64).
   kEqual = eq,
   kNotEqual = ne,
   kLessThan = lt,
   kGreaterThan = gt,
   kLessThanEqual = le,
   kGreaterThanEqual = ge,
-  kUnsignedLessThan = lt,
-  kUnsignedGreaterThan = gt,
-  kUnsignedLessThanEqual = le,
-  kUnsignedGreaterThanEqual = ge,
+  kUnsignedLessThan = 16,
+  kUnsignedGreaterThan = 17,
+  kUnsignedLessThanEqual = 18,
+  kUnsignedGreaterThanEqual = 19,
   kOverflow = overflow,
   kNoOverflow = nooverflow,
-  kZero = eq,
-  kNotZero = ne,
+  kZero = 20,
+  kNotZero = 21,
 };
 
+inline Condition check_condition(Condition cond) {
+  switch (cond) {
+    case kUnsignedLessThan:
+      return lt;
+    case kUnsignedGreaterThan:
+      return gt;
+    case kUnsignedLessThanEqual:
+      return le;
+    case kUnsignedGreaterThanEqual:
+      return ge;
+    case kZero:
+      return eq;
+    case kNotZero:
+      return ne;
+    default:
+      break;
+  }
+  return cond;
+}
+
 inline Condition NegateCondition(Condition cond) {
   DCHECK(cond != al);
   switch (cond) {
-- 
2.35.1

