diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index f4949e598b1..2995ca68b6c 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -531,6 +531,14 @@ inline void MaglevAssembler::MoveRepr(MachineRepresentation repr,
       UNREACHABLE();
   }
 }
+template <>
+inline void MaglevAssembler::MoveRepr(MachineRepresentation repr,
+                                      MemOperand dst, MemOperand src) {
+  UseScratchRegisterScope temps(this);
+  Register scratch = temps.AcquireX();
+  MoveRepr(repr, scratch, src);
+  MoveRepr(repr, dst, scratch);
+}
 
 inline Condition ToCondition(AssertCondition cond) {
   switch (cond) {
