diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 8f6e896f2ac..ec72fc5a8f7 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -67,8 +67,8 @@ void MaglevAssembler::Allocate(RegisterSnapshot register_snapshot,
         __ jmp(*done);
       },
       register_snapshot, object,
-      in_new_space ? Builtin::kAllocateRegularInYoungGeneration
-                   : Builtin::kAllocateRegularInOldGeneration,
+      in_new_space ? Builtin::kAllocateInYoungGeneration
+                   : Builtin::kAllocateInOldGeneration,
       size_in_bytes, done);
   // Store new top and tag object.
   movq(ExternalReferenceAsOperand(top), new_top);
