diff --git a/src/codegen/arm64/macro-assembler-arm64.cc b/src/codegen/arm64/macro-assembler-arm64.cc
index 30fabe7561b..21784c4661c 100644
--- a/src/codegen/arm64/macro-assembler-arm64.cc
+++ b/src/codegen/arm64/macro-assembler-arm64.cc
@@ -3017,8 +3017,7 @@ void MacroAssembler::IsObjectType(Register object, Register scratch1,
       UseScratchRegisterScope temps(this);
       Tagged_t ptr = ReadOnlyRootPtr(*expected);
       if (IsImmAddSub(ptr) || scratch1 != scratch2 || temps.CanAcquire()) {
-        // Load without decompression.
-        Ldr(scratch1.W(), FieldMemOperand(object, HeapObject::kMapOffset));
+        LoadCompressedMap(scratch1, object);
         if (!IsImmAddSub(ptr) && scratch1 != scratch2) {
           Operand imm_operand =
               MoveImmediateForShiftedOp(scratch2, ptr, kAnyShift);
@@ -3042,6 +3041,11 @@ void MacroAssembler::CompareObjectType(Register object, Register map,
   CompareInstanceType(map, type_reg, type);
 }
 
+void MacroAssembler::LoadCompressedMap(Register dst, Register object) {
+  ASM_CODE_COMMENT(this);
+  Ldr(dst.W(), FieldMemOperand(object, HeapObject::kMapOffset));
+}
+
 void MacroAssembler::LoadMap(Register dst, Register object) {
   ASM_CODE_COMMENT(this);
   LoadTaggedField(dst, FieldMemOperand(object, HeapObject::kMapOffset));
