diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index c731062644e..a4f60010ed0 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -2778,11 +2778,6 @@ void MacroAssembler::TestCodeIsMarkedForDeoptimization(Register code) {
         Immediate(1 << Code::kMarkedForDeoptimizationBit));
 }
 
-void MacroAssembler::TestCodeIsTurbofanned(Register code) {
-  testl(FieldOperand(code, Code::kFlagsOffset),
-        Immediate(1 << Code::kIsTurbofannedBit));
-}
-
 Immediate MacroAssembler::ClearedValue() const {
   return Immediate(
       static_cast<int32_t>(HeapObjectReference::ClearedValue(isolate()).ptr()));
@@ -3403,13 +3398,11 @@ void MacroAssembler::LoadNativeContextSlot(Register dst, int index) {
 }
 
 void MacroAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
-                                             CodeKind min_opt_level,
                                              Register feedback_vector,
                                              FeedbackSlot slot,
                                              Label* on_result,
                                              Label::Distance distance) {
-  ASM_CODE_COMMENT(this);
-  Label fallthrough, on_mark_deopt;
+  Label fallthrough;
   LoadTaggedField(
       scratch_and_result,
       FieldOperand(feedback_vector,
@@ -3419,19 +3412,7 @@ void MacroAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
   // Is it marked_for_deoptimization? If yes, clear the slot.
   {
     TestCodeIsMarkedForDeoptimization(scratch_and_result);
-
-    if (min_opt_level == CodeKind::TURBOFAN) {
-      j(not_zero, &on_mark_deopt, Label::Distance::kNear);
-
-      TestCodeIsTurbofanned(scratch_and_result);
-      j(not_zero, on_result, distance);
-      jmp(&fallthrough);
-    } else {
-      DCHECK_EQ(min_opt_level, CodeKind::MAGLEV);
-      j(equal, on_result, distance);
-    }
-
-    bind(&on_mark_deopt);
+    j(equal, on_result, distance);
     StoreTaggedField(
         FieldOperand(feedback_vector,
                      FeedbackVector::OffsetOfElementAt(slot.ToInt())),
