diff --git a/src/maglev/arm64/maglev-assembler-arm64.cc b/src/maglev/arm64/maglev-assembler-arm64.cc
index a3814cfcfb3..0470bf890aa 100644
--- a/src/maglev/arm64/maglev-assembler-arm64.cc
+++ b/src/maglev/arm64/maglev-assembler-arm64.cc
@@ -816,6 +816,18 @@ void MaglevAssembler::StringLength(Register result, Register string) {
   Ldr(result.W(), FieldMemOperand(string, String::kLengthOffset));
 }
 
+void MaglevAssembler::FunctionLength(Register result, Register function) {
+  AssertFunction(function);
+  MaglevAssembler::ScratchRegisterScope temps(this);
+  Register scratch = temps.Acquire();
+  LoadTaggedField(
+      scratch,
+      FieldMemOperand(function, JSFunction::kSharedFunctionInfoOffset));
+  LoadUnsignedField(result.W(),
+                    FieldMemOperand(scratch, SharedFunctionInfo::kLengthOffset),
+                    2);
+}
+
 void MaglevAssembler::StoreFixedArrayElementWithWriteBarrier(
     Register array, Register index, Register value,
     RegisterSnapshot register_snapshot) {
