diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 4d7eee06db7..4f7d82d727f 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -733,6 +733,16 @@ void MaglevAssembler::StringLength(Register result, Register string) {
   movl(result, FieldOperand(string, String::kLengthOffset));
 }
 
+void MaglevAssembler::FunctionLength(Register result, Register function) {
+  AssertFunction(function);
+  LoadTaggedField(
+      kScratchRegister,
+      FieldOperand(function, JSFunction::kSharedFunctionInfoOffset));
+  LoadUnsignedField(
+      result,
+      FieldMemOperand(kScratchRegister, SharedFunctionInfo::kLengthOffset), 2);
+}
+
 void MaglevAssembler::StoreFixedArrayElementWithWriteBarrier(
     Register array, Register index, Register value,
     RegisterSnapshot register_snapshot) {
