diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 183af0f717c..d917b97c5e9 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -200,7 +200,24 @@ inline void MaglevAssembler::DoubleToInt64Repr(Register dst,
 
 inline void MaglevAssembler::SmiTagInt32(Register obj, Label* fail) {
   addl(obj, obj);
-  JumpIf(overflow, fail);
+  if (fail != nullptr) {
+    JumpIf(kOverflow, fail);
+  } else if (v8_flags.debug_code) {
+    Check(kNoOverflow, AbortReason::kInputDoesNotFitSmi);
+  }
+}
+
+inline void MaglevAssembler::SmiTagUint32(Register obj, Label* fail) {
+  // Perform an unsigned comparison against Smi::kMaxValue.
+  if (fail != nullptr) {
+    cmpl(obj, Immediate(Smi::kMaxValue));
+    JumpIf(kUnsignedGreaterThan, fail);
+  } else if (v8_flags.debug_code) {
+    cmpl(obj, Immediate(Smi::kMaxValue));
+    Check(kUnsignedLessThanEqual, AbortReason::kInputDoesNotFitSmi);
+  }
+  addl(obj, obj);
+  Assert(kNoOverflow, AbortReason::kInputDoesNotFitSmi);
 }
 
 inline Condition MaglevAssembler::IsInt64Constant(Register reg,
@@ -616,6 +633,11 @@ inline void MaglevAssembler::JumpIfSmi(Register src, Label* on_smi,
   MacroAssembler::JumpIfSmi(src, on_smi, distance);
 }
 
+inline void MaglevAssembler::JumpIfNotSmi(Register src, Label* on_not_smi,
+                                          Label::Distance distance) {
+  MacroAssembler::JumpIfNotSmi(src, on_not_smi, distance);
+}
+
 void MaglevAssembler::JumpIfByte(Condition cc, Register value, int32_t byte,
                                  Label* target, Label::Distance distance) {
   cmpb(value, Immediate(byte));
