diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index c2a00efc7c0..6f6cd5f61c2 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -385,6 +385,11 @@ inline void MaglevAssembler::LoadByte(Register dst, MemOperand src) {
   movzxbl(dst, src);
 }
 
+inline void MaglevAssembler::CompareTagged(Register reg,
+                                           Handle<HeapObject> obj) {
+  Cmp(reg, obj);
+}
+
 inline void MaglevAssembler::CompareInt32(Register reg, int32_t imm) {
   cmpl(reg, Immediate(imm));
 }
@@ -402,6 +407,16 @@ inline void MaglevAssembler::JumpIf(Condition cond, Label* target,
   j(cond, target, distance);
 }
 
+inline void MaglevAssembler::JumpIfEqual(Label* target,
+                                         Label::Distance distance) {
+  j(equal, target, distance);
+}
+
+inline void MaglevAssembler::JumpIfNotEqual(Label* target,
+                                            Label::Distance distance) {
+  j(not_equal, target, distance);
+}
+
 inline void MaglevAssembler::JumpIfTaggedEqual(Register r1, Register r2,
                                                Label* target,
                                                Label::Distance distance) {
-- 
2.35.1

