diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 01ae18190a7..adf82e88a80 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -496,6 +496,14 @@ inline void MaglevAssembler::LoadByte(Register dst, MemOperand src) {
   Ldrb(dst, src);
 }
 
+inline void MaglevAssembler::CompareTagged(Register reg,
+                                           Handle<HeapObject> obj) {
+  ScratchRegisterScope temps(this);
+  Register scratch = temps.Acquire();
+  Move(scratch, obj);
+  Cmp(reg, scratch);
+}
+
 inline void MaglevAssembler::CompareInt32(Register reg, int32_t imm) {
   Cmp(reg.W(), Immediate(imm));
 }
@@ -511,6 +519,16 @@ inline void MaglevAssembler::JumpIf(Condition cond, Label* target,
   b(target, cond);
 }
 
+inline void MaglevAssembler::JumpIfEqual(Label* target,
+                                         Label::Distance distance) {
+  b(target, eq);
+}
+
+inline void MaglevAssembler::JumpIfNotEqual(Label* target,
+                                            Label::Distance distance) {
+  b(target, ne);
+}
+
 inline void MaglevAssembler::JumpIfTaggedEqual(Register r1, Register r2,
                                                Label* target, Label::Distance) {
   CmpTagged(r1, r2);
