diff --git a/test/cctest/test-assembler-s390.cc b/test/cctest/test-assembler-s390.cc
index 69b22e6907c..4402ae83e31 100644
--- a/test/cctest/test-assembler-s390.cc
+++ b/test/cctest/test-assembler-s390.cc
@@ -201,7 +201,7 @@ TEST(3) {
   __ mvc(MemOperand(r0, 123), MemOperand(r4, 567), Operand(88));
   __ sll(r13, Operand(10));
 
-  v8::internal::byte* bufPos = assm.buffer_pos();
+  v8::internal::uint8_t* bufPos = assm.buffer_pos();
   ::printf("buffer position = %p", static_cast<void*>(bufPos));
   ::fflush(stdout);
   // OS::DebugBreak();
