diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index ec72fc5a8f7..5e0ba089c2e 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -387,8 +387,8 @@ void MaglevAssembler::TryChangeFloat64ToIndex(Register result,
   // the input (i.e. that the conversion didn't truncate).
   Ucomisd(value, converted_back);
   JumpIf(parity_even, fail);
-  JumpIf(kEqual, success);
-  Jump(fail);
+  JumpIf(kNotEqual, fail);
+  Jump(success);
 }
 
 void MaglevAssembler::OSRPrologue(Graph* graph) {
-- 
2.35.1

