diff --git a/test/cctest/test-helper-riscv64.h b/test/cctest/test-helper-riscv64.h
index a6914e20c28..9c4fccf287d 100644
--- a/test/cctest/test-helper-riscv64.h
+++ b/test/cctest/test-helper-riscv64.h
@@ -254,7 +254,7 @@ void GenAndRunTestForLRSC(T value, Func test_generator) {
   Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #if defined(DEBUG)
-  code->Print();
+  Print(*code);
 #endif
   using INT_T =
       typename std::conditional<sizeof(T) == 4, int32_t, int64_t>::type;
@@ -318,7 +318,7 @@ OUTPUT_T GenAndRunTestForAMO(INPUT_T input0, INPUT_T input1,
   Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #if defined(DEBUG)
-  code->Print();
+  Print(*code);
 #endif
   OUTPUT_T tmp = 0;
   auto f = GeneratedCode<OUTPUT_T(void* base, INPUT_T, INPUT_T)>::FromCode(
