diff --git a/src/codegen/s390/constants-s390.h b/src/codegen/s390/constants-s390.h
index 8bade7b2abe..c8d1121a915 100644
--- a/src/codegen/s390/constants-s390.h
+++ b/src/codegen/s390/constants-s390.h
@@ -123,7 +123,7 @@ enum Condition {
   kNotZero = 21,
 };
 
-inline Condition check_condition(Condition cond) {
+inline Condition to_condition(Condition cond) {
   switch (cond) {
     case kUnsignedLessThan:
       return lt;
@@ -143,6 +143,31 @@ inline Condition check_condition(Condition cond) {
   return cond;
 }
 
+inline bool is_signed(Condition cond) {
+  switch (cond) {
+    case kEqual:
+    case kNotEqual:
+    case kLessThan:
+    case kGreaterThan:
+    case kLessThanEqual:
+    case kGreaterThanEqual:
+    case kOverflow:
+    case kNoOverflow:
+    case kZero:
+    case kNotZero:
+      return true;
+
+    case kUnsignedLessThan:
+    case kUnsignedGreaterThan:
+    case kUnsignedLessThanEqual:
+    case kUnsignedGreaterThanEqual:
+      return false;
+
+    default:
+      UNREACHABLE();
+  }
+}
+
 inline Condition NegateCondition(Condition cond) {
   DCHECK(cond != al);
   switch (cond) {
