diff --git a/src/baseline/ppc/baseline-compiler-ppc-inl.h b/src/baseline/ppc/baseline-compiler-ppc-inl.h
index 66514b373a9..7804a25e86f 100644
--- a/src/baseline/ppc/baseline-compiler-ppc-inl.h
+++ b/src/baseline/ppc/baseline-compiler-ppc-inl.h
@@ -37,7 +37,7 @@ void BaselineCompiler::PrologueFillFrame() {
   // Inlined register frame fill
   interpreter::Register new_target_or_generator_register =
       bytecode_->incoming_new_target_or_generator_register();
-  if (FLAG_debug_code) {
+  if (v8_flags.debug_code) {
     __ masm()->CompareRoot(kInterpreterAccumulatorRegister,
                            RootIndex::kUndefinedValue);
     __ masm()->Assert(eq, AbortReason::kUnexpectedValue);
