diff --git a/test/cctest/test-disasm-ppc.cc b/test/cctest/test-disasm-ppc.cc
index f28ebf4dac6..f7460161772 100644
--- a/test/cctest/test-disasm-ppc.cc
+++ b/test/cctest/test-disasm-ppc.cc
@@ -43,7 +43,7 @@ namespace internal {
 bool DisassembleAndCompare(byte* pc, const char* compare_string) {
   disasm::NameConverter converter;
   disasm::Disassembler disasm(converter);
-  EmbeddedVector<char, 128> disasm_buffer;
+  base::EmbeddedVector<char, 128> disasm_buffer;
 
   disasm.InstructionDecode(disasm_buffer, pc);
 
