diff --git a/test/cctest/test-disasm-s390.cc b/test/cctest/test-disasm-s390.cc
index a01760b6ba9..f44b3e1599c 100644
--- a/test/cctest/test-disasm-s390.cc
+++ b/test/cctest/test-disasm-s390.cc
@@ -43,7 +43,7 @@ namespace internal {
 bool DisassembleAndCompare(byte* pc, const char* compare_string) {
   disasm::NameConverter converter;
   disasm::Disassembler disasm(converter);
-  EmbeddedVector<char, 128> disasm_buffer;
+  base::EmbeddedVector<char, 128> disasm_buffer;
 
   disasm.InstructionDecode(disasm_buffer, pc);
 
