diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 8efb6682400..61f4250113f 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -521,7 +521,11 @@ inline void MaglevAssembler::LoadAddress(Register dst, MemOperand location) {
   leaq(dst, location);
 }
 
-inline void MaglevAssembler::Call(Label* target) { call(target); }
+inline int MaglevAssembler::PushOrSetReturnAddressTo(Label* target) {
+  leaq(kScratchRegister, Operand(target));
+  pushq(kScratchRegister);
+  return 1;
+}
 
 inline void MaglevAssembler::EmitEnterExitFrame(int extra_slots,
                                                 StackFrame::Type frame_type,
