diff --git a/src/codegen/ppc/macro-assembler-ppc.cc b/src/codegen/ppc/macro-assembler-ppc.cc
index ae4a7824cff..04ca54fcd13 100644
--- a/src/codegen/ppc/macro-assembler-ppc.cc
+++ b/src/codegen/ppc/macro-assembler-ppc.cc
@@ -612,7 +612,7 @@ void MacroAssembler::MultiPopF64AndV128(DoubleRegList dregs,
 
 void MacroAssembler::LoadTaggedRoot(Register destination, RootIndex index) {
   ASM_CODE_COMMENT(this);
-  if (V8_STATIC_ROOTS_BOOL && RootsTable::IsReadOnly(index)) {
+  if (CanBeImmediate(index)) {
     mov(destination, Operand(ReadOnlyRootPtr(index), RelocInfo::Mode::NO_INFO));
     return;
   }
@@ -622,7 +622,7 @@ void MacroAssembler::LoadTaggedRoot(Register destination, RootIndex index) {
 void MacroAssembler::LoadRoot(Register destination, RootIndex index,
                               Condition cond) {
   DCHECK(cond == al);
-  if (V8_STATIC_ROOTS_BOOL && RootsTable::IsReadOnly(index)) {
+  if (CanBeImmediate(index)) {
     DecompressTagged(destination, ReadOnlyRootPtr(index));
     return;
   }
