diff --git a/src/execution/s390/simulator-s390.cc b/src/execution/s390/simulator-s390.cc
index 0102020099d..c2a6885634e 100644
--- a/src/execution/s390/simulator-s390.cc
+++ b/src/execution/s390/simulator-s390.cc
@@ -427,7 +427,7 @@ void S390Debugger::Debug() {
           Object obj(*cur);
           Heap* current_heap = sim_->isolate_->heap();
           if (!skip_obj_print) {
-            if (obj.IsSmi()) {
+            if (IsSmi(obj)) {
               PrintF(" (smi %d)", Smi::ToInt(obj));
             } else if (IsValidHeapObject(current_heap, HeapObject::cast(obj))) {
               PrintF(" (");
-- 
2.35.1

