diff --git a/src/compiler/backend/arm64/instruction-scheduler-arm64.cc b/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
index 3cffa51e90..a21f454c8b 100644
--- a/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
+++ b/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
@@ -317,7 +317,7 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kArm64LdarDecompressTaggedSigned:
     case kArm64LdarDecompressTaggedPointer:
     case kArm64LdarDecompressAnyTagged:
-    case kArm64LdrDecodeCagedPointer:
+    case kArm64LdrDecodeSandboxedPointer:
     case kArm64Peek:
     case kArm64LoadSplat:
     case kArm64LoadLane:
@@ -341,7 +341,7 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kArm64Str:
     case kArm64StrCompressTagged:
     case kArm64StlrCompressTagged:
-    case kArm64StrEncodeCagedPointer:
+    case kArm64StrEncodeSandboxedPointer:
     case kArm64DmbIsh:
     case kArm64DsbIsb:
     case kArm64StoreLane:
