diff --git a/src/execution/arm64/simulator-arm64.cc b/src/execution/arm64/simulator-arm64.cc
index fa49a80cb97..146183e5d39 100644
--- a/src/execution/arm64/simulator-arm64.cc
+++ b/src/execution/arm64/simulator-arm64.cc
@@ -4005,10 +4005,10 @@ bool Simulator::ExecDebugCommand(ArrayUniquePtr<char> line_ptr) {
       if (!skip_obj_print) {
         Object obj(*cur);
         Heap* current_heap = isolate_->heap();
-        if (obj.IsSmi() ||
+        if (IsSmi(obj) ||
             IsValidHeapObject(current_heap, HeapObject::cast(obj))) {
           PrintF(" (");
-          if (obj.IsSmi()) {
+          if (IsSmi(obj)) {
             PrintF("smi %" PRId32, Smi::ToInt(obj));
           } else {
             obj.ShortPrint();
