diff --git a/src/baseline/x64/baseline-assembler-x64-inl.h b/src/baseline/x64/baseline-assembler-x64-inl.h
index 9f475ad4a01..1f2217f9178 100644
--- a/src/baseline/x64/baseline-assembler-x64-inl.h
+++ b/src/baseline/x64/baseline-assembler-x64-inl.h
@@ -386,8 +386,9 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
                                                 FeedbackSlot slot,
                                                 Label* on_result,
                                                 Label::Distance distance) {
-  __ MacroAssembler::TryLoadOptimizedOsrCode(
-      scratch_and_result, feedback_vector, slot, on_result, distance);
+  __ MacroAssembler::TryLoadOptimizedOsrCode(scratch_and_result,
+                                             CodeKind::MAGLEV, feedback_vector,
+                                             slot, on_result, distance);
 }
 
 void BaselineAssembler::AddToInterruptBudgetAndJumpIfNotExceeded(
