diff --git a/src/codegen/s390/assembler-s390.h b/src/codegen/s390/assembler-s390.h
index 324d7c1ae9..cdde38d8ec 100644
--- a/src/codegen/s390/assembler-s390.h
+++ b/src/codegen/s390/assembler-s390.h
@@ -1073,8 +1073,9 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
     basr(r14, r1);
   }
 
-  void call(Handle<CodeT> target, RelocInfo::Mode rmode);
-  void jump(Handle<CodeT> target, RelocInfo::Mode rmode, Condition cond);
+  void call(Handle<CodeDataContainer> target, RelocInfo::Mode rmode);
+  void jump(Handle<CodeDataContainer> target, RelocInfo::Mode rmode,
+            Condition cond);
 
 // S390 instruction generation
 #define DECLARE_VRR_A_INSTRUCTIONS(name, opcode_name, opcode_value)           \
