diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index f8b4e52fed6..782df0fdbc3 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -629,6 +629,16 @@ inline Condition MaglevAssembler::IsNotCallableNorUndetactable(
   return kEqual;
 }
 
+inline void MaglevAssembler::CheckHeapObjectIsNumeric(Register heap_object,
+                                                      Label* fail) {
+  Label done;
+  CompareMapWithRoot(heap_object, RootIndex::kHeapNumberMap, kScratchRegister);
+  JumpIf(kEqual, &done, Label::Distance::kNear);
+  CompareMapWithRoot(heap_object, RootIndex::kBigIntMap, kScratchRegister);
+  JumpIf(kNotEqual, fail);
+  bind(&done);
+}
+
 inline void MaglevAssembler::LoadInstanceType(Register instance_type,
                                               Register heap_object) {
   LoadMap(instance_type, heap_object);
