diff --git a/src/maglev/arm/maglev-assembler-arm-inl.h b/src/maglev/arm/maglev-assembler-arm-inl.h
index 2b61d4f67a9..ea7340e681e 100644
--- a/src/maglev/arm/maglev-assembler-arm-inl.h
+++ b/src/maglev/arm/maglev-assembler-arm-inl.h
@@ -667,6 +667,19 @@ inline Condition MaglevAssembler::IsNotCallableNorUndetactable(
   return kEqual;
 }
 
+inline void MaglevAssembler::CheckHeapObjectIsNumeric(Register heap_object,
+                                                      Label* fail) {
+  ScratchRegisterScope temps(this);
+  Register scratch = temps.Acquire();
+  Label done;
+  LoadMap(scratch, heap_object);
+  CompareRoot(scratch, RootIndex::kHeapNumberMap);
+  JumpIf(kEqual, &done, Label::Distance::kNear);
+  CompareRoot(scratch, RootIndex::kBigIntMap);
+  JumpIf(kNotEqual, fail);
+  bind(&done);
+}
+
 inline void MaglevAssembler::LoadInstanceType(Register instance_type,
                                               Register heap_object) {
   LoadMap(instance_type, heap_object);
