diff --git a/src/baseline/mips/baseline-assembler-mips-inl.h b/src/baseline/mips/baseline-assembler-mips-inl.h
index 1ebd48c2a2..a25c1a0a71 100644
--- a/src/baseline/mips/baseline-assembler-mips-inl.h
+++ b/src/baseline/mips/baseline-assembler-mips-inl.h
@@ -60,7 +60,7 @@ enum class Condition : uint32_t {
 inline internal::Condition AsMasmCondition(Condition cond) {
   // This is important for arm, where the internal::Condition where each value
   // represents an encoded bit field value.
-  STATIC_ASSERT(sizeof(internal::Condition) == sizeof(Condition));
+  static_assert(sizeof(internal::Condition) == sizeof(Condition));
   return static_cast<internal::Condition>(cond);
 }
 
