diff --git a/src/wasm/baseline/s390/liftoff-assembler-s390-inl.h b/src/wasm/baseline/s390/liftoff-assembler-s390-inl.h
index 7125b34f715..74bed1536b5 100644
--- a/src/wasm/baseline/s390/liftoff-assembler-s390-inl.h
+++ b/src/wasm/baseline/s390/liftoff-assembler-s390-inl.h
@@ -2978,10 +2978,10 @@ void LiftoffAssembler::DropStackSlotsAndRet(uint32_t num_stack_slots) {
   Ret();
 }
 
-void LiftoffAssembler::CallCWithStackBuffer(
-    const std::initializer_list<VarState> args, const LiftoffRegister* rets,
-    ValueKind return_kind, ValueKind out_argument_kind, int stack_bytes,
-    ExternalReference ext_ref) {
+void LiftoffAssembler::CallC(const std::initializer_list<VarState> args,
+                             const LiftoffRegister* rets, ValueKind return_kind,
+                             ValueKind out_argument_kind, int stack_bytes,
+                             ExternalReference ext_ref) {
   int total_size = RoundUp(stack_bytes, 8);
 
   int size = total_size;
