diff --git a/src/codegen/ppc/macro-assembler-ppc.cc b/src/codegen/ppc/macro-assembler-ppc.cc
index 878e400cb94..7fd5ec4fafd 100644
--- a/src/codegen/ppc/macro-assembler-ppc.cc
+++ b/src/codegen/ppc/macro-assembler-ppc.cc
@@ -1618,14 +1618,12 @@ void MacroAssembler::InvokeFunctionCode(Register function, Register new_target,
   // We call indirectly through the code field in the function to
   // allow recompilation to take effect without changing any of the
   // call sites.
-  Register code = kJavaScriptCallCodeStartRegister;
-  LoadTaggedField(code, FieldMemOperand(function, JSFunction::kCodeOffset), r0);
   switch (type) {
     case InvokeType::kCall:
-      CallCodeObject(code);
+      CallJSFunction(function, r0);
       break;
     case InvokeType::kJump:
-      JumpCodeObject(code);
+      JumpJSFunction(function, r0);
       break;
   }
 
@@ -5146,6 +5144,21 @@ void MacroAssembler::JumpCodeObject(Register code_object, JumpMode jump_mode) {
   Jump(code_object);
 }
 
+void MacroAssembler::CallJSFunction(Register function_object,
+                                    Register scratch) {
+  Register code = kJavaScriptCallCodeStartRegister;
+  LoadTaggedField(
+      code, FieldMemOperand(function_object, JSFunction::kCodeOffset), scratch);
+  CallCodeObject(code);
+}
+void MacroAssembler::JumpJSFunction(Register function_object, Register scratch,
+                                    JumpMode jump_mode) {
+  Register code = kJavaScriptCallCodeStartRegister;
+  LoadTaggedField(
+      code, FieldMemOperand(function_object, JSFunction::kCodeOffset), scratch);
+  JumpCodeObject(code, jump_mode);
+}
+
 void MacroAssembler::StoreReturnAddressAndCall(Register target) {
   // This generates the final instruction sequence for calls to C functions
   // once an exit frame has been constructed.
