diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 2cb16ab8adf..a2979d59344 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -427,12 +427,6 @@ inline void MaglevAssembler::CompareObjectType(Register heap_object,
   CmpInstanceType(kScratchRegister, type);
 }
 
-inline void MaglevAssembler::JumpIfJSAnyIsNotPrimitive(
-    Register heap_object, Label* target, Label::Distance distance) {
-  MacroAssembler::JumpIfJSAnyIsNotPrimitive(heap_object, kScratchRegister,
-                                            target, distance);
-}
-
 inline void MaglevAssembler::CompareObjectType(Register heap_object,
                                                InstanceType type,
                                                Register scratch) {
