diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index f22da0a785d..5b863d949c4 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -621,9 +621,16 @@ inline void MaglevAssembler::CompareObjectType(Register heap_object,
 inline void MaglevAssembler::CompareObjectTypeRange(Register heap_object,
                                                     InstanceType lower_limit,
                                                     InstanceType higher_limit) {
-  LoadMap(kScratchRegister, heap_object);
-  CmpInstanceTypeRange(kScratchRegister, kScratchRegister, lower_limit,
-                       higher_limit);
+  CompareObjectTypeRange(heap_object, kScratchRegister, lower_limit,
+                         higher_limit);
+}
+
+inline void MaglevAssembler::CompareObjectTypeRange(Register heap_object,
+                                                    Register scratch,
+                                                    InstanceType lower_limit,
+                                                    InstanceType higher_limit) {
+  LoadMap(scratch, heap_object);
+  CmpInstanceTypeRange(scratch, scratch, lower_limit, higher_limit);
 }
 
 inline void MaglevAssembler::CompareMapWithRoot(Register object,
diff --git a/test/mjsunit/maglev/regress-1456853.js b/test/mjsunit/maglev/regress-1456853.js
new file mode 100644
index 00000000000..886458238c7
