diff --git a/src/regexp/loong64/regexp-macro-assembler-loong64.h b/src/regexp/loong64/regexp-macro-assembler-loong64.h
index a141936613..fe40a4e74f 100644
--- a/src/regexp/loong64/regexp-macro-assembler-loong64.h
+++ b/src/regexp/loong64/regexp-macro-assembler-loong64.h
@@ -56,8 +56,8 @@ class V8_EXPORT_PRIVATE RegExpMacroAssemblerLOONG64
   // Checks whether the given offset from the current position is before
   // the end of the string.
   void CheckPosition(int cp_offset, Label* on_outside_input) override;
-  bool CheckSpecialCharacterClass(StandardCharacterSet type,
-                                  Label* on_no_match) override;
+  bool CheckSpecialClassRanges(StandardCharacterSet type,
+                               Label* on_no_match) override;
   void Fail() override;
   Handle<HeapObject> GetCode(Handle<String> source) override;
   void GoTo(Label* label) override;
