diff --git a/src/codegen/mips64/assembler-mips64-inl.h b/src/codegen/mips64/assembler-mips64-inl.h
index 3cf2461e5cf..2ef960d7803 100644
--- a/src/codegen/mips64/assembler-mips64-inl.h
+++ b/src/codegen/mips64/assembler-mips64-inl.h
@@ -203,6 +203,8 @@ Address RelocInfo::target_internal_reference_address() {
   return pc_;
 }
 
+Builtin RelocInfo::target_builtin_at(Assembler* origin) { UNREACHABLE(); }
+
 Address RelocInfo::target_runtime_entry(Assembler* origin) {
   DCHECK(IsRuntimeEntry(rmode_));
   return target_address();
