diff --git a/src/diagnostics/mips/disasm-mips.cc b/src/diagnostics/mips/disasm-mips.cc
index 9aee24a1d4..1ec705fd78 100644
--- a/src/diagnostics/mips/disasm-mips.cc
+++ b/src/diagnostics/mips/disasm-mips.cc
@@ -30,6 +30,8 @@
 #if V8_TARGET_ARCH_MIPS
 
 #include "src/base/platform/platform.h"
+#include "src/base/strings.h"
+#include "src/base/vector.h"
 #include "src/codegen/macro-assembler.h"
 #include "src/codegen/mips/constants-mips.h"
 #include "src/diagnostics/disasm.h"
