diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 71312204524..e84970f3baf 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -495,18 +495,6 @@ void MaglevAssembler::MaybeEmitDeoptBuiltinsCall(size_t eager_deopt_count,
                                                  size_t lazy_deopt_count,
                                                  Label* lazy_deopt_entry) {}
 
-void MaglevAssembler::StringLength(Register result, Register string) {
-  if (v8_flags.debug_code) {
-    // Check if {string} is a string.
-    AssertNotSmi(string);
-    LoadMap(kScratchRegister, string);
-    CmpInstanceTypeRange(kScratchRegister, kScratchRegister, FIRST_STRING_TYPE,
-                         LAST_STRING_TYPE);
-    Check(below_equal, AbortReason::kUnexpectedValue);
-  }
-  movl(result, FieldOperand(string, String::kLengthOffset));
-}
-
 }  // namespace maglev
 }  // namespace internal
 }  // namespace v8
-- 
2.35.1

