diff --git a/test/unittests/assembler/disasm-ia32-unittest.cc b/test/unittests/assembler/disasm-ia32-unittest.cc
index 17d89b1e0fa..147d6c3e48e 100644
--- a/test/unittests/assembler/disasm-ia32-unittest.cc
+++ b/test/unittests/assembler/disasm-ia32-unittest.cc
@@ -47,7 +47,7 @@ using DisasmIa320Test = TestWithIsolate;
 
 TEST_F(DisasmIa320Test, DisasmIa320) {
   HandleScope scope(isolate());
-  v8::internal::byte buffer[8192];
+  uint8_t buffer[8192];
   Assembler assm(AssemblerOptions{},
                  ExternalAssemblerBuffer(buffer, sizeof buffer));
   // Short immediate instructions
@@ -986,8 +986,8 @@ TEST_F(DisasmIa320Test, DisasmIa320) {
   code->Print(os);
   Address begin = code->instruction_start();
   Address end = code->instruction_end();
-  disasm::Disassembler::Disassemble(stdout, reinterpret_cast<byte*>(begin),
-                                    reinterpret_cast<byte*>(end));
+  disasm::Disassembler::Disassemble(stdout, reinterpret_cast<uint8_t*>(begin),
+                                    reinterpret_cast<uint8_t*>(end));
 #endif
 }
 
