diff --git a/src/maglev/x64/maglev-assembler-x64.cc b/src/maglev/x64/maglev-assembler-x64.cc
index 2238d500f79..c2bbedc4604 100644
--- a/src/maglev/x64/maglev-assembler-x64.cc
+++ b/src/maglev/x64/maglev-assembler-x64.cc
@@ -367,76 +367,6 @@ void MaglevAssembler::StringCharCodeOrCodePointAt(
   }
 }
 
-void MaglevAssembler::ToBoolean(Register value, CheckType check_type,
-                                ZoneLabelRef is_true, ZoneLabelRef is_false,
-                                bool fallthrough_when_true) {
-  Register map = kScratchRegister;
-
-  if (check_type == CheckType::kCheckHeapObject) {
-    // Check if {{value}} is Smi.
-    CheckSmi(value);
-    JumpToDeferredIf(
-        zero,
-        [](MaglevAssembler* masm, Register value, ZoneLabelRef is_true,
-           ZoneLabelRef is_false) {
-          // Check if {value} is not zero.
-          __ SmiCompare(value, Smi::FromInt(0));
-          __ j(equal, *is_false);
-          __ jmp(*is_true);
-        },
-        value, is_true, is_false);
-  } else if (v8_flags.debug_code) {
-    AssertNotSmi(value);
-  }
-  // Check if {{value}} is false.
-  CompareRoot(value, RootIndex::kFalseValue);
-  j(equal, *is_false);
-
-  // Check if {{value}} is empty string.
-  CompareRoot(value, RootIndex::kempty_string);
-  j(equal, *is_false);
-
-  // Check if {{value}} is undetectable.
-  LoadMap(map, value);
-  testl(FieldOperand(map, Map::kBitFieldOffset),
-        Immediate(Map::Bits1::IsUndetectableBit::kMask));
-  j(not_zero, *is_false);
-
-  // Check if {{value}} is a HeapNumber.
-  CompareRoot(map, RootIndex::kHeapNumberMap);
-  JumpToDeferredIf(
-      equal,
-      [](MaglevAssembler* masm, Register value, ZoneLabelRef is_true,
-         ZoneLabelRef is_false) {
-        // Sets scratch register to 0.0.
-        __ Xorpd(kScratchDoubleReg, kScratchDoubleReg);
-        // Sets ZF if equal to 0.0, -0.0 or NaN.
-        __ Ucomisd(kScratchDoubleReg,
-                   FieldOperand(value, HeapNumber::kValueOffset));
-        __ j(zero, *is_false);
-        __ jmp(*is_true);
-      },
-      value, is_true, is_false);
-
-  // Check if {{value}} is a BigInt.
-  CompareRoot(map, RootIndex::kBigIntMap);
-  JumpToDeferredIf(
-      equal,
-      [](MaglevAssembler* masm, Register value, ZoneLabelRef is_true,
-         ZoneLabelRef is_false) {
-        __ testl(FieldOperand(value, BigInt::kBitfieldOffset),
-                 Immediate(BigInt::LengthBits::kMask));
-        __ j(zero, *is_false);
-        __ jmp(*is_true);
-      },
-      value, is_true, is_false);
-
-  // Otherwise true.
-  if (!fallthrough_when_true) {
-    jmp(*is_true);
-  }
-}
-
 void MaglevAssembler::TestTypeOf(
     Register object, interpreter::TestTypeOfFlags::LiteralFlag literal,
     Label* is_true, Label::Distance true_distance, bool fallthrough_when_true,
-- 
2.35.1

