diff --git a/src/compiler/backend/x64/instruction-scheduler-x64.cc b/src/compiler/backend/x64/instruction-scheduler-x64.cc
index 11e265605a9..02acdb5f041 100644
--- a/src/compiler/backend/x64/instruction-scheduler-x64.cc
+++ b/src/compiler/backend/x64/instruction-scheduler-x64.cc
@@ -403,8 +403,7 @@ int InstructionScheduler::GetTargetInstructionFlags(
       }
 
     case kX64MovqDecompressTaggedSigned:
-    case kX64MovqDecompressTaggedPointer:
-    case kX64MovqDecompressAnyTagged:
+    case kX64MovqDecompressTagged:
     case kX64MovqCompressTagged:
     case kX64MovqDecodeSandboxedPointer:
     case kX64MovqEncodeSandboxedPointer:
