diff --git a/src/compiler/backend/arm64/instruction-scheduler-arm64.cc b/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
index eba6cdf75ee..dcfb0151f3f 100644
--- a/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
+++ b/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
@@ -315,11 +315,9 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kArm64LdrW:
     case kArm64Ldr:
     case kArm64LdrDecompressTaggedSigned:
-    case kArm64LdrDecompressTaggedPointer:
-    case kArm64LdrDecompressAnyTagged:
+    case kArm64LdrDecompressTagged:
     case kArm64LdarDecompressTaggedSigned:
-    case kArm64LdarDecompressTaggedPointer:
-    case kArm64LdarDecompressAnyTagged:
+    case kArm64LdarDecompressTagged:
     case kArm64LdrDecodeSandboxedPointer:
     case kArm64Peek:
     case kArm64LoadSplat:
@@ -431,8 +429,7 @@ int InstructionScheduler::GetInstructionLatency(const Instruction* instr) {
       return 1;
 
     case kArm64LdrDecompressTaggedSigned:
-    case kArm64LdrDecompressTaggedPointer:
-    case kArm64LdrDecompressAnyTagged:
+    case kArm64LdrDecompressTagged:
     case kArm64Ldr:
     case kArm64LdrD:
     case kArm64LdrS:
