diff --git a/src/codegen/ia32/macro-assembler-ia32.h b/src/codegen/ia32/macro-assembler-ia32.h
index fa7eec6bfde..8935d82c952 100644
--- a/src/codegen/ia32/macro-assembler-ia32.h
+++ b/src/codegen/ia32/macro-assembler-ia32.h
@@ -129,7 +129,7 @@ class V8_EXPORT_PRIVATE MacroAssembler
     }
   }
   void Move(Register dst, const Immediate& src);
-  void Move(Register dst, Smi src) { Move(dst, Immediate(src)); }
+  void Move(Register dst, Tagged<Smi> src) { Move(dst, Immediate(src)); }
   void Move(Register dst, Handle<HeapObject> src);
   void Move(Register dst, Register src);
   void Move(Register dst, Operand src);
@@ -361,7 +361,7 @@ class V8_EXPORT_PRIVATE MacroAssembler
   void Push(Operand src) { push(src); }
   void Push(Immediate value);
   void Push(Handle<HeapObject> handle) { push(Immediate(handle)); }
-  void Push(Smi smi) { Push(Immediate(smi)); }
+  void Push(Tagged<Smi> smi) { Push(Immediate(smi)); }
   void Push(XMMRegister src, Register scratch) {
     movd(scratch, src);
     push(scratch);
