diff --git a/src/wasm/baseline/arm64/liftoff-assembler-arm64.h b/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
index 928126addc1..07f0cc9eb21 100644
--- a/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
+++ b/src/wasm/baseline/arm64/liftoff-assembler-arm64.h
@@ -481,11 +481,12 @@ void LiftoffAssembler::StoreTaggedPointer(Register dst_addr,
   }
   CheckPageFlag(src.gp(), MemoryChunk::kPointersToHereAreInterestingMask, ne,
                 &exit);
-  CallRecordWriteStub(
-      dst_addr,
-      dst_op.IsRegisterOffset() ? Operand(dst_op.regoffset().X())
-                                : Operand(dst_op.offset()),
-      RememberedSetAction::kEmit, kSaveFPRegs, wasm::WasmCode::kRecordWrite);
+  CallRecordWriteStub(dst_addr,
+                      dst_op.IsRegisterOffset()
+                          ? Operand(dst_op.regoffset().X())
+                          : Operand(dst_op.offset()),
+                      RememberedSetAction::kEmit, SaveFPRegsMode::kSave,
+                      wasm::WasmCode::kRecordWrite);
   bind(&exit);
 }
 
