diff --git a/src/maglev/arm/maglev-assembler-arm.cc b/src/maglev/arm/maglev-assembler-arm.cc
index 02873cd91aa..74522b6aff3 100644
--- a/src/maglev/arm/maglev-assembler-arm.cc
+++ b/src/maglev/arm/maglev-assembler-arm.cc
@@ -79,13 +79,6 @@ void MaglevAssembler::Allocate(RegisterSnapshot register_snapshot,
   bind(*done);
 }
 
-void MaglevAssembler::StoreTaggedFieldWithWriteBarrier(
-    Register object, int offset, Register value,
-    RegisterSnapshot register_snapshot, ValueIsCompressed value_is_compressed,
-    ValueCanBeSmi value_can_be_smi) {
-  MAGLEV_NOT_IMPLEMENTED();
-}
-
 void MaglevAssembler::Prologue(Graph* graph) {
   ScratchRegisterScope temps(this);
   temps.Include({r4, r8});
@@ -320,18 +313,6 @@ void MaglevAssembler::StringLength(Register result, Register string) {
   MAGLEV_NOT_IMPLEMENTED();
 }
 
-void MaglevAssembler::StoreFixedArrayElementWithWriteBarrier(
-    Register array, Register index, Register value,
-    RegisterSnapshot register_snapshot) {
-  MAGLEV_NOT_IMPLEMENTED();
-}
-
-void MaglevAssembler::StoreFixedArrayElementNoWriteBarrier(Register array,
-                                                           Register index,
-                                                           Register value) {
-  MAGLEV_NOT_IMPLEMENTED();
-}
-
 }  // namespace maglev
 }  // namespace internal
 }  // namespace v8
