diff --git a/src/wasm/baseline/x64/liftoff-assembler-x64.h b/src/wasm/baseline/x64/liftoff-assembler-x64.h
index 1a12f7012ec..6a2e247addb 100644
--- a/src/wasm/baseline/x64/liftoff-assembler-x64.h
+++ b/src/wasm/baseline/x64/liftoff-assembler-x64.h
@@ -371,10 +371,20 @@ void LiftoffAssembler::LoadTaggedPointerFromInstance(Register dst,
   LoadTaggedField(dst, Operand(instance, offset));
 }
 
-void LiftoffAssembler::LoadExternalPointer(Register dst, Register instance,
+void LiftoffAssembler::LoadExternalPointer(Register dst, Register src_addr,
                                            int offset, ExternalPointerTag tag,
                                            Register scratch) {
-  LoadExternalPointerField(dst, FieldOperand(instance, offset), tag, scratch,
+  LoadExternalPointerField(dst, Operand(src_addr, offset), tag, scratch,
+                           IsolateRootLocation::kInRootRegister);
+}
+
+void LiftoffAssembler::LoadExternalPointer(Register dst, Register src_addr,
+                                           int offset, Register index,
+                                           ExternalPointerTag tag,
+                                           Register scratch) {
+  MemOperand src_op = liftoff::GetMemOp(this, src_addr, index, offset,
+                                        times_external_pointer_size);
+  LoadExternalPointerField(dst, src_op, tag, scratch,
                            IsolateRootLocation::kInRootRegister);
 }
 
