diff --git a/src/codegen/loong64/assembler-loong64.h b/src/codegen/loong64/assembler-loong64.h
index 944c9ff7c65..2e26a6ded92 100644
--- a/src/codegen/loong64/assembler-loong64.h
+++ b/src/codegen/loong64/assembler-loong64.h
@@ -1106,13 +1106,28 @@ class EnsureSpace {
 
 class V8_EXPORT_PRIVATE V8_NODISCARD UseScratchRegisterScope {
  public:
-  explicit UseScratchRegisterScope(Assembler* assembler);
-  ~UseScratchRegisterScope();
+  explicit UseScratchRegisterScope(Assembler* assembler)
+      : available_(assembler->GetScratchRegisterList()),
+        availablefp_(assembler->GetScratchFPRegisterList()),
+        old_available_(*available_),
+        old_availablefp_(*availablefp_) {}
+
+  ~UseScratchRegisterScope() {
+    *available_ = old_available_;
+    *availablefp_ = old_availablefp_;
+  }
+
+  Register Acquire() {
+    return available_->PopFirst();
+  }
+
+  DoubleRegister AcquireFp() {
+    return availablefp_->PopFirst();
+  }
+
+  bool hasAvailable() const { return !available_->is_empty(); }
 
-  Register Acquire();
-  DoubleRegister AcquireFp();
-  bool hasAvailable() const;
-  bool hasAvailableFp() const;
+  bool hasAvailableFp() const { return !availablefp_->is_empty(); }
 
   void Include(const RegList& list) { *available_ |= list; }
   void IncludeFp(const DoubleRegList& list) { *availablefp_ |= list; }
