diff --git a/test/unittests/assembler/turbo-assembler-arm64-unittest.cc b/test/unittests/assembler/turbo-assembler-arm64-unittest.cc
index dd94c30c562..77123ef5651 100644
--- a/test/unittests/assembler/turbo-assembler-arm64-unittest.cc
+++ b/test/unittests/assembler/turbo-assembler-arm64-unittest.cc
@@ -217,7 +217,7 @@ TEST_P(TurboAssemblerTestMoveObjectAndSlot, MoveObjectAndSlot) {
       CodeDesc desc;
       tasm.GetCode(nullptr, &desc);
       if (v8_flags.print_code) {
-        Handle<InstructionStream> code =
+        Handle<Code> code =
             Factory::CodeBuilder(isolate(), desc, CodeKind::FOR_TESTING)
                 .Build();
         StdoutStream os;
