diff --git a/test/unittests/assembler/turbo-assembler-arm-unittest.cc b/test/unittests/assembler/turbo-assembler-arm-unittest.cc
index 63598467c4f..6fa1bd59273 100644
--- a/test/unittests/assembler/turbo-assembler-arm-unittest.cc
+++ b/test/unittests/assembler/turbo-assembler-arm-unittest.cc
@@ -154,7 +154,7 @@ TEST_P(TurboAssemblerTestMoveObjectAndSlot, MoveObjectAndSlot) {
     CodeDesc desc;
     tasm.GetCode(nullptr, &desc);
     if (v8_flags.print_code) {
-      Handle<InstructionStream> code =
+      Handle<Code> code =
           Factory::CodeBuilder(isolate(), desc, CodeKind::FOR_TESTING).Build();
       StdoutStream os;
       code->Print(os);
