diff --git a/src/codegen/x64/macro-assembler-x64.h b/src/codegen/x64/macro-assembler-x64.h
index d53f5f9ff98..dd5555451ae 100644
--- a/src/codegen/x64/macro-assembler-x64.h
+++ b/src/codegen/x64/macro-assembler-x64.h
@@ -836,7 +836,6 @@ class V8_EXPORT_PRIVATE MacroAssembler
   }
 
   void TestCodeIsMarkedForDeoptimization(Register code);
-  void TestCodeIsTurbofanned(Register code);
   Immediate ClearedValue() const;
 
   // Tiering support.
@@ -901,9 +900,8 @@ class V8_EXPORT_PRIVATE MacroAssembler
   // Falls through and sets scratch_and_result to 0 on failure, jumps to
   // on_result on success.
   void TryLoadOptimizedOsrCode(Register scratch_and_result,
-                               CodeKind min_opt_level, Register feedback_vector,
-                               FeedbackSlot slot, Label* on_result,
-                               Label::Distance distance);
+                               Register feedback_vector, FeedbackSlot slot,
+                               Label* on_result, Label::Distance distance);
 
   // ---------------------------------------------------------------------------
   // Runtime calls
