diff --git a/src/codegen/x64/assembler-x64.h b/src/codegen/x64/assembler-x64.h
index 1555fc04e2a..f2f7aa781cb 100644
--- a/src/codegen/x64/assembler-x64.h
+++ b/src/codegen/x64/assembler-x64.h
@@ -424,7 +424,7 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   // This function is called when on-heap-compilation invariants are
   // invalidated. For instance, when the assembler buffer grows or a GC happens
   // between Code object allocation and Code object finalization.
-  void FixOnHeapReferences();
+  void FixOnHeapReferences(bool update_embedded_objects = true);
 
   // This function is called when we fallback from on-heap to off-heap
   // compilation and patch on-heap references to handles.
-- 
2.35.1

