diff --git a/src/codegen/s390/assembler-s390.h b/src/codegen/s390/assembler-s390.h
index 0f3fc270eb..7a99750151 100644
--- a/src/codegen/s390/assembler-s390.h
+++ b/src/codegen/s390/assembler-s390.h
@@ -244,7 +244,7 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   // This function is called when on-heap-compilation invariants are
   // invalidated. For instance, when the assembler buffer grows or a GC happens
   // between Code object allocation and Code object finalization.
-  void FixOnHeapReferences();
+  void FixOnHeapReferences(bool update_embedded_objects = true);
 
   // This function is called when we fallback from on-heap to off-heap
   // compilation and patch on-heap references to handles.
