diff --git a/src/codegen/mips64/assembler-mips64.h b/src/codegen/mips64/assembler-mips64.h
index 68306f42526..37a4e174a53 100644
--- a/src/codegen/mips64/assembler-mips64.h
+++ b/src/codegen/mips64/assembler-mips64.h
@@ -170,7 +170,7 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   // This function is called when on-heap-compilation invariants are
   // invalidated. For instance, when the assembler buffer grows or a GC happens
   // between Code object allocation and Code object finalization.
-  void FixOnHeapReferences();
+  void FixOnHeapReferences(bool update_embedded_objects = true);
 
   // This function is called when we fallback from on-heap to off-heap
   // compilation and patch on-heap references to handles.
