diff --git a/src/codegen/arm/interface-descriptors-arm-inl.h b/src/codegen/arm/interface-descriptors-arm-inl.h
index 23d4b31bc01..f1affc1bee6 100644
--- a/src/codegen/arm/interface-descriptors-arm-inl.h
+++ b/src/codegen/arm/interface-descriptors-arm-inl.h
@@ -99,6 +99,9 @@ constexpr Register StoreDescriptor::SlotRegister() { return r4; }
 // static
 constexpr Register StoreWithVectorDescriptor::VectorRegister() { return r3; }
 
+// static
+constexpr Register DefineKeyedOwnDescriptor::FlagsRegister() { return r5; }
+
 // static
 constexpr Register StoreTransitionDescriptor::MapRegister() { return r5; }
 
