diff --git a/src/codegen/s390/interface-descriptors-s390-inl.h b/src/codegen/s390/interface-descriptors-s390-inl.h
index cf92bbff006..f2a406394bc 100644
--- a/src/codegen/s390/interface-descriptors-s390-inl.h
+++ b/src/codegen/s390/interface-descriptors-s390-inl.h
@@ -99,6 +99,9 @@ constexpr Register StoreDescriptor::SlotRegister() { return r6; }
 // static
 constexpr Register StoreWithVectorDescriptor::VectorRegister() { return r5; }
 
+// static
+constexpr Register DefineKeyedOwnDescriptor::FlagsRegister() { return r7; }
+
 // static
 constexpr Register StoreTransitionDescriptor::MapRegister() { return r7; }
 
