diff --git a/src/codegen/x64/interface-descriptors-x64-inl.h b/src/codegen/x64/interface-descriptors-x64-inl.h
index 39f190747d3..cc662c4899d 100644
--- a/src/codegen/x64/interface-descriptors-x64-inl.h
+++ b/src/codegen/x64/interface-descriptors-x64-inl.h
@@ -111,6 +111,9 @@ constexpr Register StoreDescriptor::SlotRegister() { return rdi; }
 // static
 constexpr Register StoreWithVectorDescriptor::VectorRegister() { return rbx; }
 
+// static
+constexpr Register DefineKeyedOwnDescriptor::FlagsRegister() { return r11; }
+
 // static
 constexpr Register StoreTransitionDescriptor::MapRegister() { return r11; }
 
