diff --git a/src/codegen/arm64/interface-descriptors-arm64-inl.h b/src/codegen/arm64/interface-descriptors-arm64-inl.h
index 3b1801e0318..78027177da5 100644
--- a/src/codegen/arm64/interface-descriptors-arm64-inl.h
+++ b/src/codegen/arm64/interface-descriptors-arm64-inl.h
@@ -100,6 +100,9 @@ constexpr Register StoreDescriptor::SlotRegister() { return x4; }
 // static
 constexpr Register StoreWithVectorDescriptor::VectorRegister() { return x3; }
 
+// static
+constexpr Register DefineKeyedOwnDescriptor::FlagsRegister() { return x5; }
+
 // static
 constexpr Register StoreTransitionDescriptor::MapRegister() { return x5; }
 
