diff --git a/src/codegen/ia32/interface-descriptors-ia32-inl.h b/src/codegen/ia32/interface-descriptors-ia32-inl.h
index 9902cc1e2b3..d3f6101f4ea 100644
--- a/src/codegen/ia32/interface-descriptors-ia32-inl.h
+++ b/src/codegen/ia32/interface-descriptors-ia32-inl.h
@@ -95,6 +95,9 @@ constexpr Register StoreWithVectorDescriptor::VectorRegister() {
   return no_reg;
 }
 
+// static
+constexpr Register DefineKeyedOwnDescriptor::FlagsRegister() { return no_reg; }
+
 // static
 constexpr Register StoreTransitionDescriptor::MapRegister() { return edi; }
 
