diff --git a/src/codegen/riscv/interface-descriptors-riscv-inl.h b/src/codegen/riscv/interface-descriptors-riscv-inl.h
index f379d829147..dfda1ef7201 100644
--- a/src/codegen/riscv/interface-descriptors-riscv-inl.h
+++ b/src/codegen/riscv/interface-descriptors-riscv-inl.h
@@ -99,6 +99,9 @@ constexpr Register StoreDescriptor::SlotRegister() { return a4; }
 // static
 constexpr Register StoreWithVectorDescriptor::VectorRegister() { return a3; }
 
+// static
+constexpr Register DefineKeyedOwnDescriptor::FlagsRegister() { return a5; }
+
 // static
 constexpr Register StoreTransitionDescriptor::MapRegister() { return a5; }
 
