diff --git a/src/heap/base/asm/loong64/save_registers_asm.cc b/src/heap/base/asm/loong64/save_registers_asm.cc
index e01cd3eecc..d5b110d7ee 100644
--- a/src/heap/base/asm/loong64/save_registers_asm.cc
+++ b/src/heap/base/asm/loong64/save_registers_asm.cc
@@ -14,7 +14,7 @@
 // GN toolchain (e.g. ChromeOS) and not provide them.
 
 // 11 64-bit registers = 11 intprt_t
-static_assert(heap::base::Stack::NumberOfCalleeSavedRegisters == 11,
+static_assert(heap::base::Stack::NumberOfCalleeSavedRegisters() == 11,
               "Mismatch in the number of callee-saved registers");
 static_assert(sizeof(intptr_t) == 8, "Mismatch in word size");
 
