diff --git a/src/heap/base/asm/arm/save_registers_asm.cc b/src/heap/base/asm/arm/save_registers_asm.cc
index e29babfce80..ace95036f2e 100644
--- a/src/heap/base/asm/arm/save_registers_asm.cc
+++ b/src/heap/base/asm/arm/save_registers_asm.cc
@@ -21,7 +21,7 @@
 // http://infocenter.arm.com/help/index.jsp?topic=/com.arm.doc.faqs/ka4127.html
 
 // 8 32-bit registers = 8 intprt_t
-static_assert(heap::base::Stack::NumberOfCalleeSavedRegisters == 8,
+static_assert(heap::base::Stack::NumberOfCalleeSavedRegisters() == 8,
               "Mismatch in the number of callee-saved registers");
 static_assert(sizeof(intptr_t) == 4, "Mismatch in word size");
 
