diff --git a/src/heap/base/asm/s390/save_registers_asm.cc b/src/heap/base/asm/s390/save_registers_asm.cc
index 57831883a52..71bb2b7a78e 100644
--- a/src/heap/base/asm/s390/save_registers_asm.cc
+++ b/src/heap/base/asm/s390/save_registers_asm.cc
@@ -17,7 +17,7 @@
 // http://refspecs.linuxbase.org/ELF/zSeries/lzsabi0_zSeries.html
 
 // 10 64-bit registers = 10 intprt_t
-static_assert(heap::base::Stack::NumberOfCalleeSavedRegisters == 10,
+static_assert(heap::base::Stack::NumberOfCalleeSavedRegisters() == 10,
               "Mismatch in the number of callee-saved registers");
 static_assert(sizeof(intptr_t) == 8, "Mismatch in word size");
 
