diff --git a/src/codegen/arm64/macro-assembler-arm64.cc b/src/codegen/arm64/macro-assembler-arm64.cc
index 362614ab1a0..36b32596098 100644
--- a/src/codegen/arm64/macro-assembler-arm64.cc
+++ b/src/codegen/arm64/macro-assembler-arm64.cc
@@ -3564,10 +3564,10 @@ void MacroAssembler::LoadIndirectPointerField(Register destination,
   } else {
     CHECK(root_array_available_);
     Mov(table,
-        ExternalReference::indirect_pointer_table_base_address(isolate()));
-    Mov(destination, Operand(destination, LSR, kIndirectPointerHandleShift));
-    Ldr(destination, MemOperand(table, destination, LSL,
-                                kIndirectPointerTableEntrySizeLog2));
+        ExternalReference::trusted_pointer_table_base_address(isolate()));
+    Mov(destination, Operand(destination, LSR, kTrustedPointerHandleShift));
+    Ldr(destination,
+        MemOperand(table, destination, LSL, kTrustedPointerTableEntrySizeLog2));
   }
   Orr(destination, destination, Immediate(kHeapObjectTag));
 #else
