diff --git a/src/codegen/ia32/assembler-ia32-inl.h b/src/codegen/ia32/assembler-ia32-inl.h
index 18aa39461d6..b805d160b62 100644
--- a/src/codegen/ia32/assembler-ia32-inl.h
+++ b/src/codegen/ia32/assembler-ia32-inl.h
@@ -185,14 +185,6 @@ void Assembler::emit(Handle<HeapObject> handle) {
 void Assembler::emit(uint32_t x, RelocInfo::Mode rmode) {
   if (!RelocInfo::IsNone(rmode)) {
     RecordRelocInfo(rmode);
-    if (rmode == RelocInfo::FULL_EMBEDDED_OBJECT && IsOnHeap()) {
-      int offset = pc_offset();
-      Handle<HeapObject> object(reinterpret_cast<Address*>(x));
-      saved_handles_for_raw_object_ptr_.push_back(std::make_pair(offset, x));
-      emit(object->ptr());
-      DCHECK(EmbeddedObjectMatches(offset, object));
-      return;
-    }
   }
   emit(x);
 }
@@ -213,14 +205,6 @@ void Assembler::emit(const Immediate& x) {
     emit(0);
     return;
   }
-  if (x.is_embedded_object() && IsOnHeap()) {
-    int offset = pc_offset();
-    saved_handles_for_raw_object_ptr_.push_back(
-        std::make_pair(offset, x.immediate()));
-    emit(x.embedded_object()->ptr());
-    DCHECK(EmbeddedObjectMatches(offset, x.embedded_object()));
-    return;
-  }
   emit(x.immediate());
 }
 
