diff --git a/src/codegen/mips64/macro-assembler-mips64.h b/src/codegen/mips64/macro-assembler-mips64.h
index 68402cf3875..bff64ec7850 100644
--- a/src/codegen/mips64/macro-assembler-mips64.h
+++ b/src/codegen/mips64/macro-assembler-mips64.h
@@ -244,12 +244,12 @@ class V8_EXPORT_PRIVATE TurboAssembler : public TurboAssemblerBase {
   // it to register use ld, it can be used in wasm jump table for concurrent
   // patching.
   void PatchAndJump(Address target);
-  void Jump(Handle<CodeT> code, RelocInfo::Mode rmode, COND_ARGS);
+  void Jump(Handle<CodeDataContainer> code, RelocInfo::Mode rmode, COND_ARGS);
   void Jump(const ExternalReference& reference);
   void Call(Register target, COND_ARGS);
   void Call(Address target, RelocInfo::Mode rmode, COND_ARGS);
-  void Call(Handle<CodeT> code, RelocInfo::Mode rmode = RelocInfo::CODE_TARGET,
-            COND_ARGS);
+  void Call(Handle<CodeDataContainer> code,
+            RelocInfo::Mode rmode = RelocInfo::CODE_TARGET, COND_ARGS);
   void Call(Label* target);
   void LoadAddress(Register dst, Label* target);
 
@@ -1005,9 +1005,9 @@ class V8_EXPORT_PRIVATE MacroAssembler : public TurboAssembler {
   // less efficient form using xor instead of mov is emitted.
   void Swap(Register reg1, Register reg2, Register scratch = no_reg);
 
-  void TestCodeTIsMarkedForDeoptimizationAndJump(Register codet,
-                                                 Register scratch,
-                                                 Condition cond, Label* target);
+  void TestCodeDataContainerIsMarkedForDeoptimizationAndJump(
+      Register code_data_container, Register scratch, Condition cond,
+      Label* target);
   Operand ClearedValue() const;
 
   void PushRoot(RootIndex index) {
