diff --git a/test/cctest/test-simple-riscv64.cc b/test/cctest/test-simple-riscv64.cc
index 3cb1dddcec4..5bacfb5be8a 100644
--- a/test/cctest/test-simple-riscv64.cc
+++ b/test/cctest/test-simple-riscv64.cc
@@ -115,8 +115,8 @@ TEST(RISCV_SIMPLE2) {
   assm.GetCode(isolate, &desc);
   Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
-#ifdef DEBUG
-  code->Print();
+#ifdef OBJECT_PRINT
+  Print(*code);
 #endif
   auto f = GeneratedCode<F1>::FromCode(isolate, *code);
   int64_t res = reinterpret_cast<int64_t>(f.Call(100, 0, 0, 0, 0));
-- 
2.35.1

