diff --git a/src/codegen/mips/assembler-mips.h b/src/codegen/mips/assembler-mips.h
index 091f236061..90bfb78bc0 100644
--- a/src/codegen/mips/assembler-mips.h
+++ b/src/codegen/mips/assembler-mips.h
@@ -169,6 +169,8 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
 
   void FixOnHeapReferences();
 
+  void FixOnHeapReferencesToHandles();
+
   // Unused on this architecture.
   void MaybeEmitOutOfLineConstantPool() {}
 
