diff --git a/src/codegen/ia32/assembler-ia32.h b/src/codegen/ia32/assembler-ia32.h
index a39afbc9ab1..94366e15ee2 100644
--- a/src/codegen/ia32/assembler-ia32.h
+++ b/src/codegen/ia32/assembler-ia32.h
@@ -397,6 +397,10 @@ class V8_EXPORT_PRIVATE Assembler : public AssemblerBase {
   // between Code object allocation and Code object finalization.
   void FixOnHeapReferences();
 
+  // This function is called when we fallback from on-heap to off-heap
+  // compilation and patch on-heap references to handles.
+  void FixOnHeapReferencesToHandles();
+
   void FinalizeJumpOptimizationInfo();
 
   // Unused on this architecture.
