diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index d9b25aa6f9a..1149e8fa181 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -21,24 +21,6 @@ namespace v8 {
 namespace internal {
 namespace maglev {
 
-constexpr Condition ConditionFor(Operation operation) {
-  switch (operation) {
-    case Operation::kEqual:
-    case Operation::kStrictEqual:
-      return equal;
-    case Operation::kLessThan:
-      return less;
-    case Operation::kLessThanOrEqual:
-      return less_equal;
-    case Operation::kGreaterThan:
-      return greater;
-    case Operation::kGreaterThanOrEqual:
-      return greater_equal;
-    default:
-      UNREACHABLE();
-  }
-}
-
 constexpr Condition ConditionForFloat64(Operation operation) {
   switch (operation) {
     case Operation::kEqual:
-- 
2.35.1

