diff --git a/test/cctest/test-assembler-riscv32.cc b/test/cctest/test-assembler-riscv32.cc
index ac1f603e385..29f157055f8 100644
--- a/test/cctest/test-assembler-riscv32.cc
+++ b/test/cctest/test-assembler-riscv32.cc
@@ -2068,9 +2068,7 @@ UTEST_RVV_VF_VV_FORM_WITH_OP(vfdiv_vv, /)
       }                                                                        \
       __ instr_name(v0, v2, fa1);                                              \
       __ li(t1, Operand(int64_t(result)));                                     \
-      __ li(t2, Operand(int64_t(&result[n / 2])));                             \
       __ vs(v0, t1, 0, VSew::E64);                                             \
-      __ vs(v1, t2, 0, VSew::E64);                                             \
     };                                                                         \
     for (float rs1_fval : compiler::ValueHelper::GetVector<float>()) {         \
       for (float rs2_fval : compiler::ValueHelper::GetVector<float>()) {       \
