diff --git a/src/codegen/mips/constants-mips.h b/src/codegen/mips/constants-mips.h
index 5ed338e3e0..3dda85ceda 100644
--- a/src/codegen/mips/constants-mips.h
+++ b/src/codegen/mips/constants-mips.h
@@ -1006,8 +1006,6 @@ enum MSAMinorOpcode : uint32_t {
 // Opposite conditions must be paired as odd/even numbers
 // because 'NegateCondition' function flips LSB to negate condition.
 enum Condition {
-  // Any value < 0 is considered no_condition.
-  kNoCondition = -1,
   overflow = 0,
   no_overflow = 1,
   Uless = 2,
@@ -1053,13 +1051,9 @@ enum Condition {
   uge = Ugreater_equal,
   ule = Uless_equal,
   ugt = Ugreater,
-  cc_default = kNoCondition
 };
 
 // Returns the equivalent of !cc.
-// Negation of the default kNoCondition (-1) results in a non-default
-// no_condition value (-2). As long as tests for no_condition check
-// for condition < 0, this will work as expected.
 inline Condition NegateCondition(Condition cc) {
   DCHECK(cc != cc_always);
   return static_cast<Condition>(cc ^ 1);
