diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index fe44283ab7c..c56b7a35ca0 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -46,6 +46,9 @@ class MaglevAssembler::ScratchRegisterScope {
 
   ~ScratchRegisterScope() { masm_->scratch_register_scope_ = prev_scope_; }
 
+  Register GetDefaultScratchRegister() { return Acquire(); }
+  DoubleRegister GetDefaultScratchDoubleRegister() { return AcquireDouble(); }
+
   Register Acquire() { return wrapped_scope_.AcquireX(); }
   void Include(Register reg) { wrapped_scope_.Include(reg); }
   void Include(const RegList list) {
