diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 1149e8fa181..267fc24f03c 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -70,6 +70,9 @@ class MaglevAssembler::ScratchRegisterScope {
   }
   ~ScratchRegisterScope() { masm_->scratch_register_scope_ = prev_scope_; }
 
+  Register GetDefaultScratchRegister() { return kScratchRegister; }
+  DoubleRegister GetDefaultScratchDoubleRegister() { return kScratchDoubleReg; }
+
   Register Acquire() { return available_.PopFirst(); }
   void Include(Register reg) { available_.set(reg); }
   void Include(const RegList list) { available_ = available_ | list; }
-- 
2.35.1

