diff --git a/src/baseline/arm/baseline-compiler-arm-inl.h b/src/baseline/arm/baseline-compiler-arm-inl.h
index 0fc2389c35c..3e0b5bfcd18 100644
--- a/src/baseline/arm/baseline-compiler-arm-inl.h
+++ b/src/baseline/arm/baseline-compiler-arm-inl.h
@@ -27,7 +27,7 @@ void BaselineCompiler::Prologue() {
 }
 
 void BaselineCompiler::PrologueFillFrame() {
-  ASM_CODE_COMMENT(&masm_);
+  __ RecordComment("[ Fill frame");
   // Inlined register frame fill
   interpreter::Register new_target_or_generator_register =
       bytecode_->incoming_new_target_or_generator_register();
@@ -77,6 +77,7 @@ void BaselineCompiler::PrologueFillFrame() {
     __ masm()->sub(scratch, scratch, Operand(1), SetCC);
     __ masm()->b(gt, &loop);
   }
+  __ RecordComment("]");
 }
 
 void BaselineCompiler::VerifyFrameSize() {
