diff --git a/src/codegen/riscv/assembler-riscv-inl.h b/src/codegen/riscv/assembler-riscv-inl.h
index a81d2566895..ba5e3459069 100644
--- a/src/codegen/riscv/assembler-riscv-inl.h
+++ b/src/codegen/riscv/assembler-riscv-inl.h
@@ -206,7 +206,7 @@ void RelocInfo::set_target_object(Heap* heap, HeapObject target,
                                      icache_flush_mode);
   }
   if (write_barrier_mode == UPDATE_WRITE_BARRIER && !host().is_null() &&
-      !FLAG_disable_write_barriers) {
+      !v8_flags.disable_write_barriers) {
     WriteBarrierForCode(host(), this, target);
   }
 }
