diff --git a/src/maglev/arm/maglev-assembler-arm.cc b/src/maglev/arm/maglev-assembler-arm.cc
index 85e91eb5ce0..02873cd91aa 100644
--- a/src/maglev/arm/maglev-assembler-arm.cc
+++ b/src/maglev/arm/maglev-assembler-arm.cc
@@ -86,14 +86,6 @@ void MaglevAssembler::StoreTaggedFieldWithWriteBarrier(
   MAGLEV_NOT_IMPLEMENTED();
 }
 
-void MaglevAssembler::TestTypeOf(
-    Register object, interpreter::TestTypeOfFlags::LiteralFlag literal,
-    Label* is_true, Label::Distance true_distance, bool fallthrough_when_true,
-    Label* is_false, Label::Distance false_distance,
-    bool fallthrough_when_false) {
-  MAGLEV_NOT_IMPLEMENTED();
-}
-
 void MaglevAssembler::Prologue(Graph* graph) {
   ScratchRegisterScope temps(this);
   temps.Include({r4, r8});
