diff --git a/src/codegen/loong64/macro-assembler-loong64.cc b/src/codegen/loong64/macro-assembler-loong64.cc
index 11acc7c4d29..4143fc7fd54 100644
--- a/src/codegen/loong64/macro-assembler-loong64.cc
+++ b/src/codegen/loong64/macro-assembler-loong64.cc
@@ -3892,16 +3892,25 @@ void TurboAssembler::Float64MinOutOfLine(FPURegister dst, FPURegister src1,
 }
 
 static const int kRegisterPassedArguments = 8;
+static const int kFPRegisterPassedArguments = 8;
 
 int TurboAssembler::CalculateStackPassedWords(int num_reg_arguments,
                                               int num_double_arguments) {
   int stack_passed_words = 0;
-  num_reg_arguments += 2 * num_double_arguments;
 
   // Up to eight simple arguments are passed in registers a0..a7.
   if (num_reg_arguments > kRegisterPassedArguments) {
     stack_passed_words += num_reg_arguments - kRegisterPassedArguments;
   }
+  if (num_double_arguments > kFPRegisterPassedArguments) {
+    int num_count = num_double_arguments - kFPRegisterPassedArguments;
+    if (num_reg_arguments >= kRegisterPassedArguments) {
+      stack_passed_words += num_count;
+    } else if (num_count > kRegisterPassedArguments - num_reg_arguments) {
+      stack_passed_words +=
+          num_count - (kRegisterPassedArguments - num_reg_arguments);
+    }
+  }
   return stack_passed_words;
 }
 
