diff --git a/src/wasm/baseline/x64/liftoff-assembler-x64.h b/src/wasm/baseline/x64/liftoff-assembler-x64.h
index 41be676971b..3b7fbb6e06e 100644
--- a/src/wasm/baseline/x64/liftoff-assembler-x64.h
+++ b/src/wasm/baseline/x64/liftoff-assembler-x64.h
@@ -69,8 +69,6 @@ static_assert((kLiftoffAssemblerFpCacheRegs &
 constexpr int kInstanceOffset = 16;
 // rbp-24 is the feedback vector.
 constexpr int kFeedbackVectorOffset = 24;
-// rbp-32 is the remaining tier-up budget.
-constexpr int kTierupBudgetOffset = 32;
 
 inline constexpr Operand GetStackSlot(int offset) {
   return Operand(rbp, -offset);
