diff --git a/src/codegen/ppc/constants-ppc.h b/src/codegen/ppc/constants-ppc.h
index 99d53bfc96..6ddb3da411 100644
--- a/src/codegen/ppc/constants-ppc.h
+++ b/src/codegen/ppc/constants-ppc.h
@@ -83,15 +83,7 @@ const int kLoadDoubleMaxReachBits = 15;
 
 // The actual value of the kRootRegister is offset from the IsolateData's start
 // to take advantage of negative displacement values.
-#ifdef V8_COMPRESS_POINTERS_IN_SHARED_CAGE
 constexpr int kRootRegisterBias = 128;
-// Problems with #include order prevent this static_assert:
-// static_assert(kRootRegister != kPtrComprCageBaseRegister);
-#else
-constexpr int kRootRegisterBias = 0;
-// Problems with #include order prevent this static_assert:
-// static_assert(kRootRegister == kPtrComprCageBaseRegister);
-#endif  // V8_COMPRESS_POINTERS_IN_SHARED_CAGE
 
 // sign-extend the least significant 5-bits of value <imm>
 #define SIGN_EXT_IMM5(imm) ((static_cast<int>(imm) << 27) >> 27)
