diff --git a/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h b/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h
index 068b1b91dd7..59481e75ccd 100644
--- a/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h
+++ b/src/wasm/baseline/riscv/liftoff-assembler-riscv-inl.h
@@ -7,6 +7,7 @@
 
 #include "src/heap/memory-chunk.h"
 #include "src/wasm/baseline/liftoff-assembler.h"
+#include "src/wasm/baseline/parallel-move-inl.h"
 #include "src/wasm/object-access.h"
 #include "src/wasm/wasm-objects.h"
 
