diff --git a/src/builtins/mips64/builtins-mips64.cc b/src/builtins/mips64/builtins-mips64.cc
index f234373c117..c4760b15d7a 100644
--- a/src/builtins/mips64/builtins-mips64.cc
+++ b/src/builtins/mips64/builtins-mips64.cc
@@ -1494,12 +1494,6 @@ void Builtins::Generate_InterpreterPushArgsThenFastConstructFunction(
   FrameScope scope(masm, StackFrame::MANUAL);
   __ EnterFrame(StackFrame::FAST_CONSTRUCT);
 
-  if (v8_flags.debug_code) {
-    // Check that FrameScope pushed the context on to the stack already.
-    __ LoadReceiver(a2);
-    __ Check(eq, AbortReason::kUnexpectedValue, a2, Operand(cp));
-  }
-
   // Implicit receiver stored in the construct frame.
   __ LoadRoot(a2, RootIndex::kTheHoleValue);
   __ Push(cp, a2);
-- 
2.35.1

