diff --git a/src/regexp/arm/regexp-macro-assembler-arm.h b/src/regexp/arm/regexp-macro-assembler-arm.h
index b97d25eb5a7..44be0d920b6 100644
--- a/src/regexp/arm/regexp-macro-assembler-arm.h
+++ b/src/regexp/arm/regexp-macro-assembler-arm.h
@@ -143,6 +143,9 @@ class V8_EXPORT_PRIVATE RegExpMacroAssemblerARM
 
   static constexpr int kBacktrackConstantPoolSize = 4;
 
+  void CallCFunctionFromIrregexpCode(ExternalReference function,
+                                     int num_arguments);
+
   // Check whether preemption has been requested.
   void CheckPreemption();
 
