diff --git a/src/codegen/x64/macro-assembler-x64.h b/src/codegen/x64/macro-assembler-x64.h
index 30c01b2928e..5ed1caa11c2 100644
--- a/src/codegen/x64/macro-assembler-x64.h
+++ b/src/codegen/x64/macro-assembler-x64.h
@@ -359,6 +359,9 @@ class V8_EXPORT_PRIVATE MacroAssembler
   void LoadMap(Register destination, Register object);
   void LoadCompressedMap(Register destination, Register object);
 
+  void LoadFeedbackVector(Register dst, Register closure, Label* fbv_undef,
+                          Label::Distance distance);
+
   void Move(Register dst, intptr_t x) {
     if (x == 0) {
       xorl(dst, dst);
