diff --git a/src/maglev/arm/maglev-assembler-arm-inl.h b/src/maglev/arm/maglev-assembler-arm-inl.h
index 45055596b5f..9f1d333326e 100644
--- a/src/maglev/arm/maglev-assembler-arm-inl.h
+++ b/src/maglev/arm/maglev-assembler-arm-inl.h
@@ -616,6 +616,11 @@ inline void MaglevAssembler::CompareFloat64(DoubleRegister src1,
   vcmp(src1, src2);
 }
 
+inline void MaglevAssembler::PrepareCallCFunction(int num_reg_arguments,
+                                                  int num_double_registers) {
+  MacroAssembler::PrepareCallCFunction(num_reg_arguments, num_double_registers);
+}
+
 inline void MaglevAssembler::CallSelf() {
   DCHECK(code_gen_state()->entry_label()->is_bound());
   bl(code_gen_state()->entry_label());
