diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 0437a086761..93771e01b2d 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -652,6 +652,12 @@ inline void MaglevAssembler::CompareFloat64(DoubleRegister src1,
   Ucomisd(src1, src2);
 }
 
+inline void MaglevAssembler::PrepareCallCFunction(int num_reg_arguments,
+                                                  int num_double_registers) {
+  MacroAssembler::PrepareCallCFunction(num_reg_arguments +
+                                       num_double_registers);
+}
+
 inline void MaglevAssembler::CallSelf() {
   DCHECK(allow_call());
   DCHECK(code_gen_state()->entry_label()->is_bound());
