diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 82f7a63f300..9a905ddcb6c 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -1331,6 +1331,21 @@ void StringLength::GenerateCode(MaglevAssembler* masm,
   __ movl(ToRegister(result()), FieldOperand(object, String::kLengthOffset));
 }
 
+void FunctionLength::SetValueLocationConstraints() {
+  UseRegister(object_input());
+  DefineAsRegister(this);
+}
+void FunctionLength::GenerateCode(MaglevAssembler* masm,
+                                  const ProcessingState& state) {
+  Register object = ToRegister(object_input());
+  __ AssertFunction(object);
+  __ LoadTaggedPointerField(
+      kScratchRegister,
+      FieldOperand(object, JSFunction::kSharedFunctionInfoOffset));
+  __ movl(ToRegister(result()),
+          FieldOperand(kScratchRegister, SharedFunctionInfo::kLengthOffset));
+}
+
 void Int32AddWithOverflow::SetValueLocationConstraints() {
   UseRegister(left_input());
   UseRegister(right_input());
diff --git a/test/mjsunit/compiler/function-length.js b/test/mjsunit/compiler/function-length.js
new file mode 100644
index 00000000000..19d93efa895
