diff --git a/src/maglev/arm64/maglev-ir-arm64.cc b/src/maglev/arm64/maglev-ir-arm64.cc
index 756a5084944..5dae3247961 100644
--- a/src/maglev/arm64/maglev-ir-arm64.cc
+++ b/src/maglev/arm64/maglev-ir-arm64.cc
@@ -2453,6 +2453,22 @@ void StringLength::GenerateCode(MaglevAssembler* masm,
          FieldMemOperand(object, String::kLengthOffset));
 }
 
+void FunctionLength::SetValueLocationConstraints() {
+  UseRegister(object_input());
+  DefineAsRegister(this);
+}
+void FunctionLength::GenerateCode(MaglevAssembler* masm,
+                                  const ProcessingState& state) {
+  Register object = ToRegister(object_input());
+  __ AssertFunction(object);
+  UseScratchRegisterScope temps(masm);
+  Register shared = temps.AcquireX();
+  __ LoadTaggedPointerField(
+      shared, FieldMemOperand(object, JSFunction::kSharedFunctionInfoOffset));
+  __ Ldr(ToRegister(result()).W(),
+         FieldMemOperand(shared, SharedFunctionInfo::kLengthOffset));
+}
+
 void TestUndetectable::SetValueLocationConstraints() {
   UseRegister(value());
   DefineAsRegister(this);
