diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index bd94d5e9b1d..ee65afec42c 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -2204,8 +2204,9 @@ Operand MacroAssembler::EntryFromBuiltinAsOperand(Builtin builtin) {
 Operand MacroAssembler::EntryFromBuiltinIndexAsOperand(Register builtin_index) {
   if (SmiValuesAre32Bits()) {
     // The builtin_index register contains the builtin index as a Smi.
-    SmiUntagUnsigned(builtin_index);
-    return Operand(kRootRegister, builtin_index, times_system_pointer_size,
+    Move(kScratchRegister, builtin_index);  // Callee checks for equality.
+    SmiUntagUnsigned(kScratchRegister);
+    return Operand(kRootRegister, kScratchRegister, times_system_pointer_size,
                    IsolateData::builtin_entry_table_offset());
   } else {
     DCHECK(SmiValuesAre31Bits());
