diff --git a/src/builtins/riscv64/builtins-riscv64.cc b/src/builtins/riscv64/builtins-riscv64.cc
index 6324d30b88e..ac3dcec486c 100644
--- a/src/builtins/riscv64/builtins-riscv64.cc
+++ b/src/builtins/riscv64/builtins-riscv64.cc
@@ -1838,13 +1838,6 @@ void Builtins::Generate_NotifyDeoptimized(MacroAssembler* masm) {
   __ Ret();
 }
 
-void Builtins::Generate_TailCallOptimizedCodeSlot(MacroAssembler* masm) {
-  Register optimized_code_entry = kJavaScriptCallCodeStartRegister;
-  UseScratchRegisterScope temps(masm);
-  temps.Include(t4, t0);
-  TailCallOptimizedCodeSlot(masm, optimized_code_entry, temps.Acquire(),
-                            temps.Acquire());
-}
 namespace {
 
 void Generate_OSREntry(MacroAssembler* masm, Register entry_address,
