diff --git a/test/cctest/test-helper-riscv64.h b/test/cctest/test-helper-riscv64.h
index 6bd2c996e5..a6914e20c2 100644
--- a/test/cctest/test-helper-riscv64.h
+++ b/test/cctest/test-helper-riscv64.h
@@ -67,7 +67,7 @@ OUTPUT_T GenAndRunTest(INPUT_T input0, Func test_generator) {
       typename std::conditional<sizeof(INPUT_T) == 4, int32_t,
                                 int64_t>::type>::type;
 
-  auto f = GeneratedCode<OINT_T(IINT_T)>::FromCode(*code);
+  auto f = GeneratedCode<OINT_T(IINT_T)>::FromCode(isolate, *code);
 
   auto res = f.Call(base::bit_cast<IINT_T>(input0));
   return base::bit_cast<OUTPUT_T>(res);
@@ -114,7 +114,7 @@ OUTPUT_T GenAndRunTest(INPUT_T input0, INPUT_T input1, Func test_generator) {
       std::is_integral<INPUT_T>::value, INPUT_T,
       typename std::conditional<sizeof(INPUT_T) == 4, int32_t,
                                 int64_t>::type>::type;
-  auto f = GeneratedCode<OINT_T(IINT_T, IINT_T)>::FromCode(*code);
+  auto f = GeneratedCode<OINT_T(IINT_T, IINT_T)>::FromCode(isolate, *code);
 
   auto res =
       f.Call(base::bit_cast<IINT_T>(input0), base::bit_cast<IINT_T>(input1));
@@ -165,7 +165,8 @@ OUTPUT_T GenAndRunTest(INPUT_T input0, INPUT_T input1, INPUT_T input2,
       std::is_integral<INPUT_T>::value, INPUT_T,
       typename std::conditional<sizeof(INPUT_T) == 4, int32_t,
                                 int64_t>::type>::type;
-  auto f = GeneratedCode<OINT_T(IINT_T, IINT_T, IINT_T)>::FromCode(*code);
+  auto f =
+      GeneratedCode<OINT_T(IINT_T, IINT_T, IINT_T)>::FromCode(isolate, *code);
 
   auto res =
       f.Call(base::bit_cast<IINT_T>(input0), base::bit_cast<IINT_T>(input1),
@@ -211,7 +212,8 @@ void GenAndRunTestForLoadStore(T value, Func test_generator) {
   Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 
-  auto f = GeneratedCode<INT_T(void* base, INT_T val)>::FromCode(*code);
+  auto f =
+      GeneratedCode<INT_T(void* base, INT_T val)>::FromCode(isolate, *code);
 
   int64_t tmp = 0;
   auto res = f.Call(&tmp, base::bit_cast<INT_T>(value));
@@ -258,7 +260,8 @@ void GenAndRunTestForLRSC(T value, Func test_generator) {
       typename std::conditional<sizeof(T) == 4, int32_t, int64_t>::type;
 
   T tmp = 0;
-  auto f = GeneratedCode<INT_T(void* base, INT_T val)>::FromCode(*code);
+  auto f =
+      GeneratedCode<INT_T(void* base, INT_T val)>::FromCode(isolate, *code);
   auto res = f.Call(&tmp, base::bit_cast<T>(value));
   CHECK_EQ(base::bit_cast<T>(res), static_cast<T>(0));
 }
@@ -318,18 +321,19 @@ OUTPUT_T GenAndRunTestForAMO(INPUT_T input0, INPUT_T input1,
   code->Print();
 #endif
   OUTPUT_T tmp = 0;
-  auto f =
-      GeneratedCode<OUTPUT_T(void* base, INPUT_T, INPUT_T)>::FromCode(*code);
+  auto f = GeneratedCode<OUTPUT_T(void* base, INPUT_T, INPUT_T)>::FromCode(
+      isolate, *code);
   auto res = f.Call(&tmp, base::bit_cast<INPUT_T>(input0),
                     base::bit_cast<INPUT_T>(input1));
   return base::bit_cast<OUTPUT_T>(res);
 }
 
-Handle<Code> AssembleCodeImpl(Func assemble);
+Handle<Code> AssembleCodeImpl(Isolate* isolate, Func assemble);
 
 template <typename Signature>
-GeneratedCode<Signature> AssembleCode(Func assemble) {
-  return GeneratedCode<Signature>::FromCode(*AssembleCodeImpl(assemble));
+GeneratedCode<Signature> AssembleCode(Isolate* isolate, Func assemble) {
+  return GeneratedCode<Signature>::FromCode(
+      isolate, *AssembleCodeImpl(isolate, assemble));
 }
 
 template <typename T>
