diff --git a/src/codegen/x64/register-x64.h b/src/codegen/x64/register-x64.h
index f36763f2e4..0c95a43a04 100644
--- a/src/codegen/x64/register-x64.h
+++ b/src/codegen/x64/register-x64.h
@@ -100,9 +100,6 @@ constexpr RegList kCallerSaved =
 
 constexpr int kNumJSCallerSaved = 5;
 
-// Number of registers for which space is reserved in safepoints.
-constexpr int kNumSafepointRegisters = 16;
-
 #ifdef V8_TARGET_OS_WIN
 // Windows calling convention
 constexpr Register arg_reg_1 = rcx;
