diff --git a/src/maglev/arm64/maglev-assembler-arm64.cc b/src/maglev/arm64/maglev-assembler-arm64.cc
index a3814cfcfb3..fda3931b7f0 100644
--- a/src/maglev/arm64/maglev-assembler-arm64.cc
+++ b/src/maglev/arm64/maglev-assembler-arm64.cc
@@ -374,7 +374,7 @@ void MaglevAssembler::Prologue(Graph* graph) {
 
   // Tiering support.
   // TODO(jgruber): Extract to a builtin.
-  {
+  if (v8_flags.turbofan) {
     ScratchRegisterScope temps(this);
     Register flags = temps.Acquire();
     Register feedback_vector = temps.Acquire();
