diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 672ac1e19c3..5c5291dcae2 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -439,7 +439,7 @@ inline void MaglevAssembler::Move(Register dst, TaggedIndex i) {
   Mov(dst, i.ptr());
 }
 inline void MaglevAssembler::Move(Register dst, int32_t i) {
-  Mov(dst, Immediate(i));
+  Mov(dst.W(), Immediate(i));
 }
 inline void MaglevAssembler::Move(DoubleRegister dst, double n) {
   Fmov(dst, n);
