diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 25e8a58a4d6..b63ece012ce 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -325,7 +325,8 @@ inline void MaglevAssembler::Move(Register dst, Register src) {
 }
 
 inline void MaglevAssembler::Move(Register dst, int32_t i) {
-  MacroAssembler::Move(dst, Immediate(i));
+  // Move as a uint32 to avoid sign extension.
+  MacroAssembler::Move(dst, static_cast<uint32_t>(i));
 }
 
 inline void MaglevAssembler::Move(DoubleRegister dst, double n) {
