diff --git a/src/codegen/x64/interface-descriptors-x64-inl.h b/src/codegen/x64/interface-descriptors-x64-inl.h
index 87e3a984f56..488d09081fc 100644
--- a/src/codegen/x64/interface-descriptors-x64-inl.h
+++ b/src/codegen/x64/interface-descriptors-x64-inl.h
@@ -154,6 +154,17 @@ constexpr Register BaselineLeaveFrameDescriptor::WeightRegister() {
   return rcx;
 }
 
+// static
+constexpr Register
+MaglevOptimizeCodeOrTailCallOptimizedCodeSlotDescriptor::FlagsRegister() {
+  return r8;
+}
+// static
+constexpr Register MaglevOptimizeCodeOrTailCallOptimizedCodeSlotDescriptor::
+    FeedbackVectorRegister() {
+  return r9;
+}
+
 // static
 constexpr Register TypeConversionDescriptor::ArgumentRegister() { return rax; }
 
