diff --git a/src/codegen/arm/interface-descriptors-arm-inl.h b/src/codegen/arm/interface-descriptors-arm-inl.h
index 3f9481058bb..26365962bbf 100644
--- a/src/codegen/arm/interface-descriptors-arm-inl.h
+++ b/src/codegen/arm/interface-descriptors-arm-inl.h
@@ -154,6 +154,17 @@ constexpr Register TypeConversionDescriptor::ArgumentRegister() { return r0; }
 // static
 constexpr auto TypeofDescriptor::registers() { return RegisterArray(r0); }
 
+// static
+constexpr Register
+MaglevOptimizeCodeOrTailCallOptimizedCodeSlotDescriptor::FlagsRegister() {
+  return r2;
+}
+// static
+constexpr Register MaglevOptimizeCodeOrTailCallOptimizedCodeSlotDescriptor::
+    FeedbackVectorRegister() {
+  return r5;
+}
+
 // static
 constexpr auto CallTrampolineDescriptor::registers() {
   // r0 : number of arguments
