diff --git a/src/execution/mips/simulator-mips.h b/src/execution/mips/simulator-mips.h
index 16f9a1cead..2b36865609 100644
--- a/src/execution/mips/simulator-mips.h
+++ b/src/execution/mips/simulator-mips.h
@@ -404,7 +404,7 @@ class Simulator : public SimulatorBase {
   void TraceMemRd(int32_t addr, T value);
   template <typename T>
   void TraceMemWr(int32_t addr, T value);
-  EmbeddedVector<char, 128> trace_buf_;
+  base::EmbeddedVector<char, 128> trace_buf_;
 
   // Operations depending on endianness.
   // Get Double Higher / Lower word.
