diff --git a/src/maglev/arm/maglev-assembler-arm-inl.h b/src/maglev/arm/maglev-assembler-arm-inl.h
index 9f1d333326e..91136519cfc 100644
--- a/src/maglev/arm/maglev-assembler-arm-inl.h
+++ b/src/maglev/arm/maglev-assembler-arm-inl.h
@@ -233,8 +233,6 @@ inline void MaglevAssembler::DoubleToInt64Repr(Register dst,
   MAGLEV_NOT_IMPLEMENTED();
 }
 
-inline int MaglevAssembler::ToBooleanTemporaryCount() { return 1; }
-
 inline void MaglevAssembler::SmiTagInt32(Register obj, Label* fail) {
   add(obj, obj, obj, SetCC);
   if (fail != nullptr) {
