diff --git a/src/maglev/arm64/maglev-ir-arm64.cc b/src/maglev/arm64/maglev-ir-arm64.cc
index 4a9b27cf74b..6ac1d57dad2 100644
--- a/src/maglev/arm64/maglev-ir-arm64.cc
+++ b/src/maglev/arm64/maglev-ir-arm64.cc
@@ -1438,9 +1438,9 @@ void CheckJSDataViewBounds::GenerateCode(MaglevAssembler* masm,
           // TODO(v8:7700): Inline DataViewPrototypeGetByteLength or create a
           // different builtin that does not re-check the DataView object.
           SaveRegisterStateForCall save_register_state(masm, snapshot);
+          __ PushReverse(object);
           __ Mov(kContextRegister, masm->native_context().object());
           __ Mov(kJavaScriptCallArgCountRegister, 1);
-          __ Push(object);
           __ CallBuiltin(Builtin::kDataViewPrototypeGetByteLength);
         }
         __ SmiUntag(byte_length, kReturnRegister0);
