diff --git a/test/cctest/test-assembler-ia32.cc b/test/cctest/test-assembler-ia32.cc
index 423bf88dbf7..5fb548b9916 100644
--- a/test/cctest/test-assembler-ia32.cc
+++ b/test/cctest/test-assembler-ia32.cc
@@ -501,9 +501,9 @@ TEST(AssemblerIa32Extractps) {
 
   F4 f = FUNCTION_CAST<F4>(code->entry());
   uint64_t value1 = 0x1234'5678'8765'4321;
-  CHECK_EQ(0x12345678, f(uint64_to_double(value1)));
+  CHECK_EQ(0x12345678, f(base::uint64_to_double(value1)));
   uint64_t value2 = 0x8765'4321'1234'5678;
-  CHECK_EQ(static_cast<int>(0x87654321), f(uint64_to_double(value2)));
+  CHECK_EQ(static_cast<int>(0x87654321), f(base::uint64_to_double(value2)));
 }
 
 using F8 = int (*)(float x, float y);
