diff --git a/src/codegen/ia32/assembler-ia32-inl.h b/src/codegen/ia32/assembler-ia32-inl.h
index f03b032ec2f..d09b33858d0 100644
--- a/src/codegen/ia32/assembler-ia32-inl.h
+++ b/src/codegen/ia32/assembler-ia32-inl.h
@@ -166,7 +166,7 @@ void Assembler::emit(uint32_t x, RelocInfo::Mode rmode) {
   emit(x);
 }
 
-void Assembler::emit(Handle<Code> code, RelocInfo::Mode rmode) {
+void Assembler::emit(Handle<CodeT> code, RelocInfo::Mode rmode) {
   emit(code.address(), rmode);
 }
 
