diff --git a/src/codegen/arm/assembler-arm-inl.h b/src/codegen/arm/assembler-arm-inl.h
index b1cd1d5205a..3f2a1ce380a 100644
--- a/src/codegen/arm/assembler-arm-inl.h
+++ b/src/codegen/arm/assembler-arm-inl.h
@@ -156,7 +156,7 @@ void RelocInfo::WipeOut() {
   }
 }
 
-Handle<Code> Assembler::relative_code_target_object_handle_at(
+Handle<CodeT> Assembler::relative_code_target_object_handle_at(
     Address pc) const {
   Instruction* branch = Instruction::At(pc);
   int code_target_index = branch->GetBranchOffset() / kInstrSize;
