diff --git a/test/cctest/test-assembler-s390.cc b/test/cctest/test-assembler-s390.cc
index 7aa50382ff0..a69bf649881 100644
--- a/test/cctest/test-assembler-s390.cc
+++ b/test/cctest/test-assembler-s390.cc
@@ -63,7 +63,7 @@ TEST(0) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -102,7 +102,7 @@ TEST(1) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -153,7 +153,7 @@ TEST(2) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -208,7 +208,7 @@ TEST(3) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -249,8 +249,8 @@ TEST(4) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code = isolate->factory()->NewCode(
-      desc, CodeKind::FOR_TESTING, Handle<InstructionStream>());
+  Handle<Code> code = isolate->factory()->NewCode(
+      desc, CodeKind::FOR_TESTING, Handle<Code>());
 #ifdef DEBUG
   code->Print();
 #endif
@@ -277,8 +277,8 @@ TEST(5) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code = isolate->factory()->NewCode(
-      desc, CodeKind::FOR_TESTING, Handle<InstructionStream>());
+  Handle<Code> code = isolate->factory()->NewCode(
+      desc, CodeKind::FOR_TESTING, Handle<Code>());
 #ifdef DEBUG
   code->Print();
 #endif
@@ -311,8 +311,8 @@ TEST(6) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code = isolate->factory()->NewCode(
-      desc, CodeKind::FOR_TESTING, Handle<InstructionStream>());
+  Handle<Code> code = isolate->factory()->NewCode(
+      desc, CodeKind::FOR_TESTING, Handle<Code>());
 #ifdef DEBUG
   code->Print();
 #endif
@@ -343,8 +343,8 @@ TEST(7) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code = isolate->factory()->NewCode(
-      desc, CodeKind::FOR_TESTING, Handle<InstructionStream>());
+  Handle<Code> code = isolate->factory()->NewCode(
+      desc, CodeKind::FOR_TESTING, Handle<Code>());
 #ifdef DEBUG
   code->Print();
 #endif
@@ -374,8 +374,8 @@ TEST(8) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code = isolate->factory()->NewCode(
-      desc, CodeKind::FOR_TESTING, Handle<InstructionStream>());
+  Handle<Code> code = isolate->factory()->NewCode(
+      desc, CodeKind::FOR_TESTING, Handle<Code>());
 #ifdef DEBUG
   code->Print();
 #endif
@@ -401,8 +401,8 @@ TEST(9) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code = isolate->factory()->NewCode(
-      desc, CodeKind::FOR_TESTING, Handle<InstructionStream>());
+  Handle<Code> code = isolate->factory()->NewCode(
+      desc, CodeKind::FOR_TESTING, Handle<Code>());
 #ifdef DEBUG
   code->Print();
 #endif
@@ -486,7 +486,7 @@ TEST(10) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -540,7 +540,7 @@ TEST(11) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -594,7 +594,7 @@ TEST(12) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -658,7 +658,7 @@ TEST(13) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -749,7 +749,7 @@ TEST(14) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -839,7 +839,7 @@ TEST(15) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -886,7 +886,7 @@ TEST(16) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -960,7 +960,7 @@ TEST(17) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
@@ -1052,7 +1052,7 @@ TEST(18) {
 
   CodeDesc desc;
   assm.GetCode(isolate, &desc);
-  Handle<InstructionStream> code =
+  Handle<Code> code =
       Factory::CodeBuilder(isolate, desc, CodeKind::FOR_TESTING).Build();
 #ifdef DEBUG
   code->Print();
-- 
2.35.1

