diff --git a/test/unittests/assembler/macro-assembler-arm-unittest.cc b/test/unittests/assembler/macro-assembler-arm-unittest.cc
index f7ec44e77fd..875d7ba4cba 100644
--- a/test/unittests/assembler/macro-assembler-arm-unittest.cc
+++ b/test/unittests/assembler/macro-assembler-arm-unittest.cc
@@ -162,11 +162,11 @@ TEST_P(MacroAssemblerTestMoveObjectAndSlot, MoveObjectAndSlot) {
 
     buffer->MakeExecutable();
     // We need an isolate here to execute in the simulator.
-    auto f = GeneratedCode<void, byte**, byte*>::FromBuffer(isolate(),
-                                                            buffer->start());
+    auto f = GeneratedCode<void, uint8_t**, uint8_t*>::FromBuffer(
+        isolate(), buffer->start());
 
-    byte* object = new byte[offset];
-    byte* result[] = {nullptr, nullptr};
+    uint8_t* object = new uint8_t[offset];
+    uint8_t* result[] = {nullptr, nullptr};
 
     f.Call(result, object);
 
