diff --git a/src/compiler/backend/riscv64/instruction-codes-riscv64.h b/src/compiler/backend/riscv64/instruction-codes-riscv64.h
index aa0d446d22b..b7f9202c4ee 100644
--- a/src/compiler/backend/riscv64/instruction-codes-riscv64.h
+++ b/src/compiler/backend/riscv64/instruction-codes-riscv64.h
@@ -235,8 +235,6 @@ namespace compiler {
   V(RiscvF32x4Abs)                          \
   V(RiscvF32x4Neg)                          \
   V(RiscvF32x4Sqrt)                         \
-  V(RiscvF32x4RecipApprox)                  \
-  V(RiscvF32x4RecipSqrtApprox)              \
   V(RiscvF32x4Qfma)                         \
   V(RiscvF32x4Qfms)                         \
   V(RiscvF64x2Qfma)                         \
