diff --git a/src/compiler/backend/mips64/instruction-codes-mips64.h b/src/compiler/backend/mips64/instruction-codes-mips64.h
index a99726bae5c..5e05046feba 100644
--- a/src/compiler/backend/mips64/instruction-codes-mips64.h
+++ b/src/compiler/backend/mips64/instruction-codes-mips64.h
@@ -239,8 +239,6 @@ namespace compiler {
   V(Mips64F32x4Abs)                   \
   V(Mips64F32x4Neg)                   \
   V(Mips64F32x4Sqrt)                  \
-  V(Mips64F32x4RecipApprox)           \
-  V(Mips64F32x4RecipSqrtApprox)       \
   V(Mips64F32x4Add)                   \
   V(Mips64F32x4Sub)                   \
   V(Mips64F32x4Mul)                   \
