diff --git a/src/compiler/backend/loong64/instruction-codes-loong64.h b/src/compiler/backend/loong64/instruction-codes-loong64.h
index 0cbd163059b..67d0517bc41 100644
--- a/src/compiler/backend/loong64/instruction-codes-loong64.h
+++ b/src/compiler/backend/loong64/instruction-codes-loong64.h
@@ -212,8 +212,6 @@ namespace compiler {
   V(Loong64F32x4Abs)                   \
   V(Loong64F32x4Neg)                   \
   V(Loong64F32x4Sqrt)                  \
-  V(Loong64F32x4RecipApprox)           \
-  V(Loong64F32x4RecipSqrtApprox)       \
   V(Loong64F32x4Add)                   \
   V(Loong64F32x4Sub)                   \
   V(Loong64F32x4Mul)                   \
