diff --git a/src/compiler/backend/arm/instruction-codes-arm.h b/src/compiler/backend/arm/instruction-codes-arm.h
index 24e43834f06..75f6aee953c 100644
--- a/src/compiler/backend/arm/instruction-codes-arm.h
+++ b/src/compiler/backend/arm/instruction-codes-arm.h
@@ -167,8 +167,6 @@ namespace compiler {
   V(ArmF32x4Abs)                   \
   V(ArmF32x4Neg)                   \
   V(ArmF32x4Sqrt)                  \
-  V(ArmF32x4RecipApprox)           \
-  V(ArmF32x4RecipSqrtApprox)       \
   V(ArmF32x4Add)                   \
   V(ArmF32x4Sub)                   \
   V(ArmF32x4Mul)                   \
