diff --git a/src/compiler/backend/mips/instruction-scheduler-mips.cc b/src/compiler/backend/mips/instruction-scheduler-mips.cc
index d59392b40a3..83b80e99338 100644
--- a/src/compiler/backend/mips/instruction-scheduler-mips.cc
+++ b/src/compiler/backend/mips/instruction-scheduler-mips.cc
@@ -100,8 +100,6 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kMipsF32x4Ne:
     case kMipsF32x4Neg:
     case kMipsF32x4Sqrt:
-    case kMipsF32x4RecipApprox:
-    case kMipsF32x4RecipSqrtApprox:
     case kMipsF32x4ReplaceLane:
     case kMipsF32x4SConvertI32x4:
     case kMipsF32x4Splat:
