diff --git a/src/compiler/backend/riscv64/instruction-scheduler-riscv64.cc b/src/compiler/backend/riscv64/instruction-scheduler-riscv64.cc
index 23e06507d9d..a9e2acbffa0 100644
--- a/src/compiler/backend/riscv64/instruction-scheduler-riscv64.cc
+++ b/src/compiler/backend/riscv64/instruction-scheduler-riscv64.cc
@@ -119,8 +119,6 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kRiscvF32x4Ne:
     case kRiscvF32x4Neg:
     case kRiscvF32x4Sqrt:
-    case kRiscvF32x4RecipApprox:
-    case kRiscvF32x4RecipSqrtApprox:
     case kRiscvF64x2Qfma:
     case kRiscvF64x2Qfms:
     case kRiscvF32x4Qfma:
