diff --git a/src/codegen/riscv64/assembler-riscv64.cc b/src/codegen/riscv64/assembler-riscv64.cc
index ef06240ffa..3bdbe40770 100644
--- a/src/codegen/riscv64/assembler-riscv64.cc
+++ b/src/codegen/riscv64/assembler-riscv64.cc
@@ -2758,7 +2758,9 @@ void Assembler::RelocateRelativeReference(RelocInfo::Mode rmode, Address pc,
   }
 }
 
-void Assembler::FixOnHeapReferences() { UNIMPLEMENTED(); }
+void Assembler::FixOnHeapReferences(bool update_embedded_objects) {
+  UNIMPLEMENTED();
+}
 
 void Assembler::FixOnHeapReferencesToHandles() { UNIMPLEMENTED(); }
 
