diff --git a/src/baseline/x64/baseline-assembler-x64-inl.h b/src/baseline/x64/baseline-assembler-x64-inl.h
index f05d829cb54..fff84a20569 100644
--- a/src/baseline/x64/baseline-assembler-x64-inl.h
+++ b/src/baseline/x64/baseline-assembler-x64-inl.h
@@ -410,8 +410,7 @@ void BaselineAssembler::TryLoadOptimizedOsrCode(Register scratch_and_result,
 
   // Is it marked_for_deoptimization? If yes, clear the slot.
   {
-    DCHECK(!AreAliased(scratch_and_result, kScratchRegister));
-    __ TestCodeTIsMarkedForDeoptimization(scratch_and_result, kScratchRegister);
+    __ TestCodeTIsMarkedForDeoptimization(scratch_and_result);
     __ j(equal, on_result, distance);
     __ StoreTaggedField(
         FieldOperand(feedback_vector,
