diff --git a/src/execution/mips64/frame-constants-mips64.h b/src/execution/mips64/frame-constants-mips64.h
index 76f1fe593d5..e65710d5c13 100644
--- a/src/execution/mips64/frame-constants-mips64.h
+++ b/src/execution/mips64/frame-constants-mips64.h
@@ -42,7 +42,7 @@ class WasmLiftoffSetupFrameConstants : public TypedFrameConstants {
 
   // SP-relative.
   static constexpr int kWasmInstanceOffset = 2 * kSystemPointerSize;
-  static constexpr int kFunctionIndexOffset = 1 * kSystemPointerSize;
+  static constexpr int kDeclaredFunctionIndexOffset = 1 * kSystemPointerSize;
   static constexpr int kNativeModuleOffset = 0;
 };
 
