diff --git a/src/wasm/baseline/mips64/liftoff-assembler-mips64-inl.h b/src/wasm/baseline/mips64/liftoff-assembler-mips64-inl.h
index 9df43c84b67..2742e880054 100644
--- a/src/wasm/baseline/mips64/liftoff-assembler-mips64-inl.h
+++ b/src/wasm/baseline/mips64/liftoff-assembler-mips64-inl.h
@@ -3651,10 +3651,10 @@ void LiftoffAssembler::DropStackSlotsAndRet(uint32_t num_stack_slots) {
   MacroAssembler::DropAndRet(static_cast<int>(num_stack_slots));
 }
 
-void LiftoffAssembler::CallC(const std::initializer_list<VarState> args,
-                             const LiftoffRegister* rets, ValueKind return_kind,
-                             ValueKind out_argument_kind, int stack_bytes,
-                             ExternalReference ext_ref) {
+void LiftoffAssembler::CallCWithStackBuffer(
+    const std::initializer_list<VarState> args, const LiftoffRegister* rets,
+    ValueKind return_kind, ValueKind out_argument_kind, int stack_bytes,
+    ExternalReference ext_ref) {
   Daddu(sp, sp, -stack_bytes);
 
   int arg_offset = 0;
