diff --git a/src/wasm/baseline/loong64/liftoff-assembler-loong64-inl.h b/src/wasm/baseline/loong64/liftoff-assembler-loong64-inl.h
index e4b4a407131..19736789d52 100644
--- a/src/wasm/baseline/loong64/liftoff-assembler-loong64-inl.h
+++ b/src/wasm/baseline/loong64/liftoff-assembler-loong64-inl.h
@@ -3123,10 +3123,10 @@ void LiftoffAssembler::DropStackSlotsAndRet(uint32_t num_stack_slots) {
   Ret();
 }
 
-void LiftoffAssembler::CallC(const std::initializer_list<VarState> args,
-                             const LiftoffRegister* rets, ValueKind return_kind,
-                             ValueKind out_argument_kind, int stack_bytes,
-                             ExternalReference ext_ref) {
+void LiftoffAssembler::CallCWithStackBuffer(
+    const std::initializer_list<VarState> args, const LiftoffRegister* rets,
+    ValueKind return_kind, ValueKind out_argument_kind, int stack_bytes,
+    ExternalReference ext_ref) {
   addi_d(sp, sp, -stack_bytes);
 
   int arg_offset = 0;
