diff --git a/test/cctest/test-assembler-arm64.cc b/test/cctest/test-assembler-arm64.cc
index 829b124fd62..05181d6542f 100644
--- a/test/cctest/test-assembler-arm64.cc
+++ b/test/cctest/test-assembler-arm64.cc
@@ -217,7 +217,7 @@ static void InitializeVM() {
   {                                                            \
     /* Reset the scope and thus make the buffer executable. */ \
     rw_buffer_scope.reset();                                   \
-    auto f = GeneratedCode<void>::FromCode(isolate, *code);    \
+    auto f = GeneratedCode<void>::FromCode(*code);             \
     f.Call();                                                  \
   }
 
