diff --git a/src/compiler/backend/arm64/instruction-selector-arm64.cc b/src/compiler/backend/arm64/instruction-selector-arm64.cc
index b4ff77bebd8..fdb94c29623 100644
--- a/src/compiler/backend/arm64/instruction-selector-arm64.cc
+++ b/src/compiler/backend/arm64/instruction-selector-arm64.cc
@@ -3220,8 +3220,8 @@ void InstructionSelector::VisitWord32Equal(Node* const node) {
     }
   }
 
-  if (V8_STATIC_ROOTS_BOOL ||
-      (COMPRESS_POINTERS_BOOL && !isolate()->bootstrapper())) {
+  if (isolate() && (V8_STATIC_ROOTS_BOOL ||
+                    (COMPRESS_POINTERS_BOOL && !isolate()->bootstrapper()))) {
     Arm64OperandGenerator g(this);
     const RootsTable& roots_table = isolate()->roots_table();
     RootIndex root_index;
