diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 72a465a0eb9..955dfd690ff 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -1342,9 +1342,8 @@ void FunctionLength::GenerateCode(MaglevAssembler* masm,
   __ LoadTaggedPointerField(
       kScratchRegister,
       FieldOperand(object, JSFunction::kSharedFunctionInfoOffset));
-  __ LoadSignedField(
-      ToRegister(result()),
-      FieldMemOperand(kScratchRegister, SharedFunctionInfo::kLengthOffset), 2);
+  __ movl(ToRegister(result()),
+          FieldOperand(kScratchRegister, SharedFunctionInfo::kLengthOffset));
 }
 
 void Int32AddWithOverflow::SetValueLocationConstraints() {
-- 
2.35.1

