diff --git a/test/unittests/assembler/disasm-ia32-unittest.cc b/test/unittests/assembler/disasm-ia32-unittest.cc
index b5b8ced878f..e300b8d444c 100644
--- a/test/unittests/assembler/disasm-ia32-unittest.cc
+++ b/test/unittests/assembler/disasm-ia32-unittest.cc
@@ -978,7 +978,7 @@ TEST_F(DisasmIa320Test, DisasmIa320) {
 
   CodeDesc desc;
   assm.GetCode(isolate(), &desc);
-  Handle<Code> code =
+  Handle<InstructionStream> code =
       Factory::CodeBuilder(isolate(), desc, CodeKind::FOR_TESTING).Build();
   USE(code);
 #ifdef OBJECT_PRINT
