diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 8d91aabca47..95638508d1d 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -249,6 +249,20 @@ inline void MaglevAssembler::BindBlock(BasicBlock* block) {
   }
 }
 
+inline void MaglevAssembler::DoubleToInt64Repr(Register dst,
+                                               DoubleRegister src) {
+  Mov(dst, src, 0);
+}
+
+inline Condition MaglevAssembler::IsInt64Constant(Register reg,
+                                                  int64_t constant) {
+  UseScratchRegisterScope temps(this);
+  Register scratch = temps.AcquireX();
+  Mov(scratch, kHoleNanInt64);
+  Cmp(reg, scratch);
+  return eq;
+}
+
 inline Condition MaglevAssembler::IsRootConstant(Input input,
                                                  RootIndex root_index) {
   if (input.operand().IsRegister()) {
