diff --git a/src/maglev/x64/maglev-assembler-x64-inl.h b/src/maglev/x64/maglev-assembler-x64-inl.h
index 2c1814c7636..cd543ccbcf5 100644
--- a/src/maglev/x64/maglev-assembler-x64-inl.h
+++ b/src/maglev/x64/maglev-assembler-x64-inl.h
@@ -203,6 +203,8 @@ inline void MaglevAssembler::DoubleToInt64Repr(Register dst,
   Movq(dst, src);
 }
 
+inline int MaglevAssembler::ToBooleanTemporaryCount() { return 0; }
+
 inline void MaglevAssembler::SmiTagInt32(Register obj, Label* fail) {
   addl(obj, obj);
   if (fail != nullptr) {
-- 
2.35.1

