diff --git a/src/maglev/arm64/maglev-assembler-arm64-inl.h b/src/maglev/arm64/maglev-assembler-arm64-inl.h
index 4e44641dce9..a88c826d460 100644
--- a/src/maglev/arm64/maglev-assembler-arm64-inl.h
+++ b/src/maglev/arm64/maglev-assembler-arm64-inl.h
@@ -312,6 +312,8 @@ inline void MaglevAssembler::DoubleToInt64Repr(Register dst,
   Mov(dst, src, 0);
 }
 
+inline int MaglevAssembler::ToBooleanTemporaryCount() { return 0; }
+
 inline void MaglevAssembler::SmiTagInt32(Register obj, Label* fail) {
   Adds(obj.W(), obj.W(), obj.W());
   if (fail != nullptr) {
