diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 0c565fa65fc..42c2e9868a3 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -218,7 +218,7 @@ void AssertInt32::SetValueLocationConstraints() {
 }
 void AssertInt32::GenerateCode(MaglevAssembler* masm,
                                const ProcessingState& state) {
-  __ cmpq(ToRegister(left_input()), ToRegister(right_input()));
+  __ cmpl(ToRegister(left_input()), ToRegister(right_input()));
   __ Check(ToCondition(condition_), reason_);
 }
 
@@ -1982,15 +1982,6 @@ void CheckInt32IsSmi::GenerateCode(MaglevAssembler* masm,
   __ EmitEagerDeoptIf(overflow, DeoptimizeReason::kNotASmi, this);
 }
 
-void CheckUint32IsSmi::SetValueLocationConstraints() { UseRegister(input()); }
-void CheckUint32IsSmi::GenerateCode(MaglevAssembler* masm,
-                                    const ProcessingState& state) {
-  Register reg = ToRegister(input());
-  // Perform an unsigned comparison against Smi::kMaxValue.
-  __ cmpl(reg, Immediate(Smi::kMaxValue));
-  __ EmitEagerDeoptIf(above, DeoptimizeReason::kNotASmi, this);
-}
-
 void CheckedSmiTagInt32::SetValueLocationConstraints() {
   UseAndClobberRegister(input());
   DefineSameAsFirst(this);
-- 
2.35.1

