diff --git a/src/codegen/x64/macro-assembler-x64.cc b/src/codegen/x64/macro-assembler-x64.cc
index 77a9a4296ee..d67ebd80e99 100644
--- a/src/codegen/x64/macro-assembler-x64.cc
+++ b/src/codegen/x64/macro-assembler-x64.cc
@@ -3061,6 +3061,7 @@ void MacroAssembler::AssertCode(Register object) {
 }
 
 void MacroAssembler::AssertSmiOrHeapObjectInCompressionCage(Register object) {
+  DCHECK(PointerCompressionIsEnabled());
   if (!v8_flags.debug_code) return;
   ASM_CODE_COMMENT(this);
   Label is_smi;
