diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 1a37b546f22..a4e616d9c6d 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -714,9 +714,9 @@ void CheckedObjectToIndex::GenerateCode(MaglevAssembler* masm,
 
   // If we didn't enter the deferred block, we're a Smi.
   if (result_reg == object) {
-    __ SmiUntag(object);
+    __ SmiToInt32(result_reg);
   } else {
-    __ SmiUntag(result_reg, object);
+    __ SmiToInt32(result_reg, object);
   }
 
   __ bind(*done);
-- 
2.35.1

