diff --git a/src/maglev/arm64/maglev-ir-arm64.cc b/src/maglev/arm64/maglev-ir-arm64.cc
index 10d8ffcc145..93945f3e5da 100644
--- a/src/maglev/arm64/maglev-ir-arm64.cc
+++ b/src/maglev/arm64/maglev-ir-arm64.cc
@@ -790,9 +790,9 @@ void CheckedObjectToIndex::GenerateCode(MaglevAssembler* masm,
 
   // If we didn't enter the deferred block, we're a Smi.
   if (result_reg == object) {
-    __ SmiUntag(object);
+    __ SmiToInt32(result_reg);
   } else {
-    __ SmiUntag(result_reg, object);
+    __ SmiToInt32(result_reg, object);
   }
 
   __ bind(*done);
