diff --git a/src/codegen/arm64/macro-assembler-arm64.cc b/src/codegen/arm64/macro-assembler-arm64.cc
index 4d7b2e95c5a..f3d42154126 100644
--- a/src/codegen/arm64/macro-assembler-arm64.cc
+++ b/src/codegen/arm64/macro-assembler-arm64.cc
@@ -1695,6 +1695,7 @@ void MacroAssembler::AssertBoundFunction(Register object) {
 }
 
 void MacroAssembler::AssertSmiOrHeapObjectInCompressionCage(Register object) {
+  DCHECK(PointerCompressionIsEnabled());
   if (!v8_flags.debug_code) return;
   ASM_CODE_COMMENT(this);
   Label is_smi;
