diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 3ff525d8fc7..21303058a34 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -1591,8 +1591,6 @@ void CheckInt32IsSmi::GenerateCode(MaglevAssembler* masm,
   Register reg = ToRegister(input());
   __ movl(kScratchRegister, reg);
   __ addl(kScratchRegister, kScratchRegister);
-  DCHECK_REGLIST_EMPTY(RegList{reg} &
-                       GetGeneralRegistersUsedAsInputs(eager_deopt_info()));
   __ EmitEagerDeoptIf(overflow, DeoptimizeReason::kNotASmi, this);
 }
 
-- 
2.35.1

