diff --git a/src/builtins/ia32/builtins-ia32.cc b/src/builtins/ia32/builtins-ia32.cc
index 68ca26c348d..58124f8e6a1 100644
--- a/src/builtins/ia32/builtins-ia32.cc
+++ b/src/builtins/ia32/builtins-ia32.cc
@@ -1797,7 +1797,11 @@ void Builtins::Generate_BaselineOutOfLinePrologue(MacroAssembler* masm) {
     Register bytecode_array = scratch;
     __ movd(bytecode_array, saved_bytecode_array);
     __ Push(bytecode_array);
-    __ Push(Immediate(0));  // Unused slot.
+
+    // Baseline code frames store the feedback vector where interpreter would
+    // store the bytecode offset.
+    // TODO(victorgomes): The first push should actually be a free slot.
+    __ Push(saved_feedback_vector, scratch);
     __ Push(saved_feedback_vector, scratch);
   }
 
