diff --git a/src/builtins/x64/builtins-x64.cc b/src/builtins/x64/builtins-x64.cc
index 2e6a6e9ad33..dac8b21e976 100644
--- a/src/builtins/x64/builtins-x64.cc
+++ b/src/builtins/x64/builtins-x64.cc
@@ -1765,7 +1765,11 @@ void Builtins::Generate_BaselineOutOfLinePrologue(MacroAssembler* masm) {
       Register bytecode_array = descriptor.GetRegisterParameter(
           BaselineOutOfLinePrologueDescriptor::kInterpreterBytecodeArray);
       __ Push(bytecode_array);
-      __ Push(Immediate(0));  // Unused slot.
+
+      // Baseline code frames store the feedback vector where interpreter would
+      // store the bytecode offset.
+      // TODO(victorgomes): The first push should actually be a free slot.
+      __ Push(feedback_vector);
       __ Push(feedback_vector);
     }
 
