diff --git a/src/builtins/arm64/builtins-arm64.cc b/src/builtins/arm64/builtins-arm64.cc
index de8ca837326..fe641d9459b 100644
--- a/src/builtins/arm64/builtins-arm64.cc
+++ b/src/builtins/arm64/builtins-arm64.cc
@@ -2230,8 +2230,10 @@ void Builtins::Generate_MaglevFunctionEntryStackCheck(MacroAssembler* masm,
     FrameScope scope(masm, StackFrame::INTERNAL);
     __ AssertSmi(x0);
     if (save_new_target) {
-      __ AssertSmiOrHeapObjectInCompressionCage(
-          kJavaScriptCallNewTargetRegister);
+      if (PointerCompressionIsEnabled()) {
+        __ AssertSmiOrHeapObjectInCompressionCage(
+            kJavaScriptCallNewTargetRegister);
+      }
       __ Push(kJavaScriptCallNewTargetRegister, padreg);
     }
     __ PushArgument(x0);
