diff --git a/src/compiler/backend/mips/instruction-scheduler-mips.cc b/src/compiler/backend/mips/instruction-scheduler-mips.cc
index c6a5617d007..d246006cc88 100644
--- a/src/compiler/backend/mips/instruction-scheduler-mips.cc
+++ b/src/compiler/backend/mips/instruction-scheduler-mips.cc
@@ -284,10 +284,10 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kMipsS16x8InterleaveRight:
     case kMipsS16x8PackEven:
     case kMipsS16x8PackOdd:
-    case kMipsV64x2AllTrue:
-    case kMipsV32x4AllTrue:
-    case kMipsV16x8AllTrue:
-    case kMipsV8x16AllTrue:
+    case kMipsI64x2AllTrue:
+    case kMipsI32x4AllTrue:
+    case kMipsI16x8AllTrue:
+    case kMipsI8x16AllTrue:
     case kMipsV128AnyTrue:
     case kMipsS32x4InterleaveEven:
     case kMipsS32x4InterleaveLeft:
