diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 27a720a73f..84b32d19cc 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -2342,8 +2342,8 @@ void Return::GenerateCode(MaglevAssembler* masm, const ProcessingState& state) {
 
   __ bind(&drop_dynamic_arg_size);
   // Drop receiver + arguments according to dynamic arguments size.
-  __ DropArguments(actual_params_size, r9, TurboAssembler::kCountIsInteger,
-                   TurboAssembler::kCountIncludesReceiver);
+  __ DropArguments(actual_params_size, r9, MacroAssembler::kCountIsInteger,
+                   MacroAssembler::kCountIncludesReceiver);
   __ Ret();
 }
 
