diff --git a/src/execution/mips/simulator-mips.cc b/src/execution/mips/simulator-mips.cc
index 7aa5e00fbe9..cc195178b1a 100644
--- a/src/execution/mips/simulator-mips.cc
+++ b/src/execution/mips/simulator-mips.cc
@@ -2245,7 +2245,7 @@ void Simulator::SoftwareInterrupt() {
     int32_t arg17 = stack_pointer[17];
     int32_t arg18 = stack_pointer[18];
     int32_t arg19 = stack_pointer[19];
-    STATIC_ASSERT(kMaxCParameters == 20);
+    static_assert(kMaxCParameters == 20);
 
     bool fp_call =
         (redirection->type() == ExternalReference::BUILTIN_FP_FP_CALL) ||
