diff --git a/src/maglev/x64/maglev-ir-x64.cc b/src/maglev/x64/maglev-ir-x64.cc
index 4bff8f282f9..9fc6018a1be 100644
--- a/src/maglev/x64/maglev-ir-x64.cc
+++ b/src/maglev/x64/maglev-ir-x64.cc
@@ -2379,7 +2379,11 @@ void BranchIfFloat64Compare::GenerateCode(MaglevAssembler* masm,
   DoubleRegister left = ToDoubleRegister(left_input());
   DoubleRegister right = ToDoubleRegister(right_input());
   __ Ucomisd(left, right);
-  __ j(parity_even, if_false()->label());
+  if (jump_mode_if_nan_ == JumpModeIfNaN::kJumpToTrue) {
+    __ j(parity_even, if_true()->label());
+  } else {
+    __ j(parity_even, if_false()->label());
+  }
   __ Branch(ConditionForFloat64(operation_), if_true(), if_false(),
             state.next_block());
 }
diff --git a/test/mjsunit/maglev/regress/regress-crbug-1421375.js b/test/mjsunit/maglev/regress/regress-crbug-1421375.js
new file mode 100644
index 00000000000..271ef30624a
