diff --git a/src/compiler/backend/loong64/code-generator-loong64.cc b/src/compiler/backend/loong64/code-generator-loong64.cc
index 6fc5f1bf22f..dc07238322e 100644
--- a/src/compiler/backend/loong64/code-generator-loong64.cc
+++ b/src/compiler/backend/loong64/code-generator-loong64.cc
@@ -269,7 +269,7 @@ class WasmProtectedInstructionTrap final : public WasmOutOfLineTrap {
 
   void Generate() override {
     DCHECK(v8_flags.wasm_bounds_checks && !v8_flags.wasm_enforce_bounds_checks);
-    gen_->AddProtectedInstructionLanding(pc_, __ pc_offset());
+    gen_->RecordProtectedInstruction(pc_, __ pc_offset());
     GenerateWithTrapId(trap_id_);
   }
 
