diff --git a/src/codegen/riscv/assembler-riscv.h b/src/codegen/riscv/assembler-riscv.h
index 452c0b16eaf..f383cbf9212 100644
--- a/src/codegen/riscv/assembler-riscv.h
+++ b/src/codegen/riscv/assembler-riscv.h
@@ -63,9 +63,9 @@
 namespace v8 {
 namespace internal {
 
-#define DEBUG_PRINTF(...) \
-  if (FLAG_riscv_debug) { \
-    printf(__VA_ARGS__);  \
+#define DEBUG_PRINTF(...)     \
+  if (v8_flags.riscv_debug) { \
+    printf(__VA_ARGS__);      \
   }
 
 class SafepointTableBuilder;
