diff --git a/src/codegen/ppc/assembler-ppc.h b/src/codegen/ppc/assembler-ppc.h
index 837a415997a..6563287d0a8 100644
--- a/src/codegen/ppc/assembler-ppc.h
+++ b/src/codegen/ppc/assembler-ppc.h
@@ -310,7 +310,7 @@ class Assembler : public AssemblerBase {
   static constexpr int kTaggedLoadInstructions = 1;
 #endif
   static constexpr int kMovInstructions =
-      FLAG_enable_embedded_constant_pool.value()
+      v8_flags.enable_embedded_constant_pool.value()
           ? kMovInstructionsConstantPool
           : kMovInstructionsNoConstantPool;
 
