diff --git a/src/codegen/mips64/macro-assembler-mips64.cc b/src/codegen/mips64/macro-assembler-mips64.cc
index 784d08ca01c..b74659e918d 100644
--- a/src/codegen/mips64/macro-assembler-mips64.cc
+++ b/src/codegen/mips64/macro-assembler-mips64.cc
@@ -2142,7 +2142,7 @@ void TurboAssembler::Neg_d(FPURegister fd, FPURegister fs) {
     neg_d(fd, fs);  // In delay slot.
     bind(&is_nan);
     dmfc1(scratch1, fs);
-    li(scratch2, base::Double::kSignMask);
+    li(scratch2, Double::kSignMask);
     Xor(scratch1, scratch1, scratch2);
     dmtc1(scratch1, fd);
     bind(&done);
