diff --git a/src/builtins/x64/builtins-x64.cc b/src/builtins/x64/builtins-x64.cc
index 734019ecd8a..505d26d9897 100644
--- a/src/builtins/x64/builtins-x64.cc
+++ b/src/builtins/x64/builtins-x64.cc
@@ -2897,10 +2897,8 @@ void Builtins::Generate_MaglevFunctionEntryStackCheck(MacroAssembler* masm,
     FrameScope scope(masm, StackFrame::INTERNAL);
     __ AssertSmi(rax);
     if (save_new_target) {
-      if (PointerCompressionIsEnabled()) {
-        __ AssertSmiOrHeapObjectInCompressionCage(
-            kJavaScriptCallNewTargetRegister);
-      }
+      __ AssertSmiOrHeapObjectInCompressionCage(
+          kJavaScriptCallNewTargetRegister);
       __ Push(kJavaScriptCallNewTargetRegister);
     }
     __ Push(rax);
