diff --git a/src/compiler/backend/arm64/instruction-scheduler-arm64.cc b/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
index 46f390e5513..a0c25795c76 100644
--- a/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
+++ b/src/compiler/backend/arm64/instruction-scheduler-arm64.cc
@@ -343,6 +343,7 @@ int InstructionScheduler::GetTargetInstructionFlags(
     case kArm64StrPair:
     case kArm64StrCompressTagged:
     case kArm64StlrCompressTagged:
+    case kArm64StrIndirectPointer:
     case kArm64StrEncodeSandboxedPointer:
     case kArm64DmbIsh:
     case kArm64DsbIsb:
